---
title: "Georgakopoulous-soares and Parada et. al 2019"
output:
  html_notebook: default
  html_document: default
  pdf_document: default
---




# Non-B DNA structures across splice-sites


To investigate if non-B DNA structures can affect splicing, we explore the distribution of different motifs that are associated with Non-B DNA structures. We calculated the number of these motifs that can be found across splice sites (.num files) and then we calculated the positional enrichment of these across splice sites. 



```{r}
library(readr)
library(data.table)
library(ggplot2)
library(plyr)
library(cowplot)
library(Hmisc)


```


On this report we compile all the data analysis that lead us to get the final figures. For this, we processed several  a

```{r}



read_dist_table <- function(path){
  
dist_table <- data.table(read_delim(path, 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE))

dist_table <- dist_table[, 2:2001]
dist_table <- data.table(as.data.frame(t(dist_table)))
colnames(dist_table) <- c("Position", "Occurrences")

dist_table[,median:=median(Occurrences)]
dist_table[, Enrrichment:=Occurrences/median]
dist_table[, Position:=Position-1]

return(dist_table)  
}


```




```{r}

plot_density <- function(up_plus, up_minus, down_plus, down_minus){

  
  up_TOTAL <-  merge(up_plus, up_minus, by="Position")
  up_TOTAL[,Occurrences:=Occurrences.x+Occurrences.y]
  up_TOTAL[,median:=median(Occurrences)]
  up_TOTAL[, Enrrichment:=Occurrences/median]
  up_TOTAL[, Position:=Position-1]
  
  
  down_TOTAL <-  merge(down_plus, down_minus, by="Position")
  down_TOTAL[,Occurrences:=Occurrences.x+Occurrences.y]
  down_TOTAL[,median:=median(Occurrences)]
  down_TOTAL[, Enrrichment:=Occurrences/median]
  down_TOTAL[, Position:=Position-1]
  
  up_TOTAL[ ,exon_pos:="Upstream"]
  down_TOTAL[ ,exon_pos:="Downstream"]
  
  TOTAL <- rbind(up_TOTAL, down_TOTAL)
  
  TOTAL$exon_pos <-  factor(TOTAL$exon_pos, levels=c("Upstream", "Downstream" )) 
  
  p <- ggplot(TOTAL)+
    geom_line(aes(x=Position,y=Enrrichment)) +
    facet_grid( . ~ exon_pos ) +
    theme_bw()
  
  show(p)
  
  TOTAL
  
  

}
```



```{r}


plot_density_binomial <- function(up_plus, up_minus, down_plus, down_minus, observations, sig){

  
  up_TOTAL <-  merge(up_plus, up_minus, by="Position")
  up_TOTAL[,Occurrences:=Occurrences.x+Occurrences.y]
  up_TOTAL[,Occurrences:=Occurrences.x+Occurrences.y]
  

  
  
  up_TOTAL <- cbind(up_TOTAL, up_TOTAL[, binconf(Occurrences, observations, alpha=sig) ])

  up_TOTAL[,median:=median(PointEst)]
  up_TOTAL[, Enrrichment:=PointEst/median]
  up_TOTAL[, Enrrichment_l:=Lower/median]
  up_TOTAL[, Enrrichment_u:=Upper/median]
  up_TOTAL[, Position:=Position-1]

  
  
  down_TOTAL <-  merge(down_plus, down_minus, by="Position")
  down_TOTAL[,Occurrences:=Occurrences.x+Occurrences.y]

  
  down_TOTAL <- cbind(down_TOTAL, down_TOTAL[, binconf(Occurrences, observations, alpha=sig) ])

  down_TOTAL[,median:=median(PointEst)]
  down_TOTAL[, Enrrichment:=PointEst/median]
  down_TOTAL[, Enrrichment_l:=Lower/median]
  down_TOTAL[, Enrrichment_u:=Upper/median]
  down_TOTAL[, Position:=Position-1]  
  
  
  up_TOTAL[ ,exon_pos:="Upstream"]
  down_TOTAL[ ,exon_pos:="Downstream"]
  
  TOTAL <- rbind(up_TOTAL, down_TOTAL)
  
  TOTAL$exon_pos <-  factor(TOTAL$exon_pos, levels=c("Upstream", "Downstream" )) 
  
  p <- ggplot(TOTAL)+
    geom_line(aes(x=Position,y=Enrrichment)) +
    geom_ribbon(aes(ymin=Enrrichment_l, ymax=Enrrichment_u, x=Position), alpha=0.3 )+
    facet_grid( . ~ exon_pos ) +
    theme_bw()
  
  #show(p)
  
  return(TOTAL) 
  
}
```



## G-Quadruplexes

```{r, message=FALSE, error=FALSE, warning=FALSE}

G4.up_plus <- read_dist_table("./All_non_Bs/G4.exon.up_plus.list.out.num")
G4.down_plus <- read_dist_table("./All_non_Bs/G4.exon.down_plus.list.out.num")
G4.up_minus <- read_dist_table("./All_non_Bs/G4.exon.up_minus.list.out.num")
G4.down_minus <- read_dist_table("./All_non_Bs/G4.exon.down_minus.list.out.num")
G4.up_minus[,Position:=Position*-1]
G4.down_minus[,Position:=Position*-1]

G4.TOTAL <- plot_density(G4.up_plus, G4.up_minus, G4.down_plus, G4.down_minus) 

```


## Directed repeats


```{r, message=FALSE, error=FALSE, warning=FALSE}

DR.up_plus <- read_dist_table("./All_non_Bs/DR.exon.up_plus.list.out.num")
DR.down_plus <- read_dist_table("./All_non_Bs/DR.exon.down_plus.list.out.num")
DR.up_minus <- read_dist_table("./All_non_Bs/DR.exon.up_minus.list.out.num")
DR.down_minus <- read_dist_table("./All_non_Bs/DR.exon.down_minus.list.out.num")
DR.up_minus[,Position:=Position*-1]
DR.down_minus[,Position:=Position*-1]


DR.TOTAL <- plot_density(DR.up_plus, DR.up_minus, DR.down_plus, DR.down_minus)

```


## H-DNA


```{r, message=FALSE, error=FALSE, warning=FALSE}

H_DNA.up_plus <- read_dist_table("./All_non_Bs/H_DNA.exon.down_minus.list.out.num")
H_DNA.down_plus <- read_dist_table("./All_non_Bs/H_DNA.exon.down_plus.list.out.num")
H_DNA.up_minus <- read_dist_table("./All_non_Bs/H_DNA.exon.up_minus.list.out.num")
H_DNA.down_minus <- read_dist_table("./All_non_Bs/H_DNA.exon.down_minus.list.out.num")
H_DNA.up_minus[,Position:=Position*-1]
H_DNA.down_minus[,Position:=Position*-1]


H_DNA.TOTAL <- plot_density(H_DNA.up_plus, H_DNA.up_minus, H_DNA.down_plus, H_DNA.down_minus)

```


## Inverted repeat


```{r, message=FALSE, error=FALSE, warning=FALSE}

IR.up_plus <- read_dist_table("./All_non_Bs/IR.exon.down_minus.list.out.num")
IR.down_plus <- read_dist_table("./All_non_Bs/IR.exon.down_plus.list.out.num")
IR.up_minus <- read_dist_table("./All_non_Bs/IR.exon.up_minus.list.out.num")
IR.down_minus <- read_dist_table("./All_non_Bs/IR.exon.down_minus.list.out.num")
IR.up_minus[,Position:=Position*-1]
IR.down_minus[,Position:=Position*-1]


IR.TOTAL <- plot_density(IR.up_plus, IR.up_minus, IR.down_plus, IR.down_minus)

```


## Mirror repeat

```{r, message=FALSE, error=FALSE, warning=FALSE}

MR.up_plus <- read_dist_table("./All_non_Bs/MR.exon.down_minus.list.out.num")
MR.down_plus <- read_dist_table("./All_non_Bs/MR.exon.down_plus.list.out.num")
MR.up_minus <- read_dist_table("./All_non_Bs/MR.exon.up_minus.list.out.num")
MR.down_minus <- read_dist_table("./All_non_Bs/MR.exon.down_minus.list.out.num")
MR.up_minus[,Position:=Position*-1]
MR.down_minus[,Position:=Position*-1]


MR.TOTAL <- plot_density(MR.up_plus, MR.up_minus, MR.down_plus, MR.down_minus)

```


## Short tandem repeat

```{r, message=FALSE, error=FALSE, warning=FALSE}

STR.up_plus <- read_dist_table("./All_non_Bs/STR.exon.down_minus.list.out.num")
STR.down_plus <- read_dist_table("./All_non_Bs/STR.exon.down_plus.list.out.num")
STR.up_minus <- read_dist_table("./All_non_Bs/STR.exon.up_minus.list.out.num")
STR.down_minus <- read_dist_table("./All_non_Bs/STR.exon.down_minus.list.out.num")
STR.up_minus[,Position:=Position*-1]
STR.down_minus[,Position:=Position*-1]


STR.TOTAL <- plot_density(STR.up_plus, STR.up_minus, STR.down_plus, STR.down_minus)

```



## Z-DNA


```{r, message=FALSE, error=FALSE, warning=FALSE}

Z_DNA.up_plus <- read_dist_table("./All_non_Bs/Z_DNA.exon.down_minus.list.out.num")
Z_DNA.down_plus <- read_dist_table("./All_non_Bs/Z_DNA.exon.down_plus.list.out.num")
Z_DNA.up_minus <- read_dist_table("./All_non_Bs/Z_DNA.exon.up_minus.list.out.num")
Z_DNA.down_minus <- read_dist_table("./All_non_Bs/Z_DNA.exon.down_minus.list.out.num")
Z_DNA.up_minus[,Position:=Position*-1]
Z_DNA.down_minus[,Position:=Position*-1]


Z_DNA.TOTAL <- plot_density(Z_DNA.up_plus, Z_DNA.up_minus, Z_DNA.down_plus, Z_DNA.down_minus)

```


## All Non-B DNA motifs 


```{r, fig.height=7, fig.width=5}
G4.TOTAL[, Non_B:="G4"]
DR.TOTAL[, Non_B:="DR"]
H_DNA.TOTAL[, Non_B:="H DNA"]
IR.TOTAL[, Non_B:="IR"]
MR.TOTAL[, Non_B:="MR"]
STR.TOTAL[, Non_B:="STR"]
Z_DNA.TOTAL[, Non_B:="Z DNA"]

Non_B.TOTAL <- rbind(G4.TOTAL, DR.TOTAL, H_DNA.TOTAL, IR.TOTAL, MR.TOTAL, STR.TOTAL, Z_DNA.TOTAL)

Fig1.A <- ggplot(Non_B.TOTAL)+
    geom_line(aes(x=Position,y=Enrrichment)) +
    xlim(c(-300,300)) +
    facet_grid( Non_B ~ exon_pos ) +
  theme_bw()

Fig1.A

```




```{r}

Non_B.TOTAL.Enrrichment_Peak <- Non_B.TOTAL[ , .(Enrrichment_Peak=max(Enrrichment)) , by = c("Non_B", "exon_pos" )]



write.table(Non_B.TOTAL.Enrrichment_Peak, file = "./Tables/Non_B.TOTAL.Enrrichment_Peak.txt", append = FALSE, quote = FALSE, sep = "\t",
            eol = "\n", na = "NA", dec = ".", row.names = FALSE,
            col.names = TRUE, qmethod = c("escape", "double"),
            fileEncoding = "")

```


# Distribution of non-B DNA motifs across weak and strong splice sites


We calculated the distribution of non-B DNA motifs across quartiles of splice strength, where Q1 are the weakest splice sites and Q4 are the strongest.  


```{r}

Merge_Qs <- function(Qs.list, window_len, strand ) {
  
  Total_Qs <- data.table(rbindlist(Qs.list))
  Q_names <- seq(nrow(Total_Qs)/window_len)
  Total_Qs[, Q:=rep(Q_names, each=window_len)]
  
  if(strand=="-"){
    
    Total_Qs[,Position:=Position*-1]
  }
  
  Total_Qs
}


```


```{r}
merge_plus_minus <- function(plus, minus){

  TOTAL <- merge(plus, minus, by=c("Position", "Q") )
  TOTAL[,Occurrences:=Occurrences.x+Occurrences.y]
  TOTAL[,median:=median(Occurrences),  by=c("Q")]
  TOTAL[, Enrrichment:=Occurrences/median]
  TOTAL[, Position:=Position-1]
  
}
```


```{r}
Merge_up_down <- function(Qs.up, Qs.down){

  Qs.total <- rbind(Qs.up, Qs.down)
  
  half_n <- nrow(Qs.total)/2
  Qs.total[, exon_pos:=rep(c("Upstream", "Downstream"), each=half_n) ]
  
  
  Qs.total$exon_pos <-  factor(Qs.total$exon_pos, levels=c("Upstream", "Downstream" ))
  Qs.total$Q <-  factor(Qs.total$Q)
  
  Qs.total
  
}
```



```{r}


get_total_Qs <- function(Qs.plus.up.list, Qs.minus.list, Qs.plus.down.list, Qs.minus.down.list, window_len){

  Qs.plus.up <- Merge_Qs(Qs.plus.up.list, window_len, strand="+")
  Qs.minus.up <- Merge_Qs(Qs.minus.list, window_len, strand="-")
  Qs.up <- merge_plus_minus(Qs.plus.up, Qs.minus.up)
  
  Qs.plus.down <- Merge_Qs(Qs.plus.down.list, window_len, strand="+")
  Qs.minus.down<- Merge_Qs(Qs.minus.down.list, window_len, strand="-")
  Qs.down <- merge_plus_minus(Qs.plus.down, Qs.minus.down)
  
  Qs.total <- Merge_up_down(Qs.up, Qs.down)
  
  Qs.total
  
}
  
```



```{r}


G4.Q1_up_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_plus.q1.bed.G4.list.out.num")
G4.Q2_up_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_plus.q2.bed.G4.list.out.num")
G4.Q3_up_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_plus.q3.bed.G4.list.out.num")
G4.Q4_up_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_plus.q4.bed.G4.list.out.num")

G4.Q1_down_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_plus.q1.bed.G4.list.out.num")
G4.Q2_down_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_plus.q2.bed.G4.list.out.num")
G4.Q3_down_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_plus.q3.bed.G4.list.out.num")
G4.Q4_down_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_plus.q4.bed.G4.list.out.num")

G4.Q1_up_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_minus.q1.bed.G4.list.out.num")
G4.Q2_up_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_minus.q2.bed.G4.list.out.num")
G4.Q3_up_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_minus.q3.bed.G4.list.out.num")
G4.Q4_up_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_minus.q4.bed.G4.list.out.num")

G4.Q1_down_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_minus.q1.bed.G4.list.out.num")
G4.Q2_down_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_minus.q2.bed.G4.list.out.num")
G4.Q3_down_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_minus.q3.bed.G4.list.out.num")
G4.Q4_down_minus <- read_dist_table(".//All_non_Bs/q1_q4/exon.down_minus.q4.bed.G4.list.out.num")

G4.Qs.plus.up.list <- list(G4.Q1_up_plus, G4.Q2_up_plus, G4.Q3_up_plus, G4.Q4_up_plus)
G4.Qs.minus.list <- list(G4.Q1_up_minus, G4.Q2_up_minus, G4.Q3_up_minus, G4.Q4_up_minus)
G4.Qs.plus.down.list <- list(G4.Q1_down_plus, G4.Q2_down_plus, G4.Q3_down_plus, G4.Q4_down_plus)
G4.Qs.minus.down.list <- list(G4.Q1_down_minus, G4.Q2_down_minus, G4.Q3_down_minus, G4.Q4_down_minus)
G4.window_len = 2000

G4.Qs.total <- get_total_Qs(G4.Qs.plus.up.list, G4.Qs.minus.list, G4.Qs.plus.down.list, G4.Qs.minus.down.list, G4.window_len)

G4.Qs.total$Q <- mapvalues(G4.Qs.total$Q,  from =c(1:4), to =c("Q1", "Q2", "Q3", "Q4"))

ggplot(G4.Qs.total)+
  geom_line(aes(x=Position,y=Enrrichment, colour=Q)) +
  xlim(c(-150,150)) +
  facet_grid( . ~ exon_pos ) +
  theme(text = element_text(size=15)) +
  theme_bw()+labs(colour = "Splice site stregth quartile") +
   theme(legend.position = "top", legend.direction = "horizontal")

```


We calculated the binomial confidence intervals given the total number of exon in each quartile


```{r}

hg19.nexons <- 123433 + 123936  #Number of exons found in each side

G4.Qs.total.binomial <- cbind(G4.Qs.total,  G4.Qs.total[, binconf(Occurrences, hg19.nexons/4) ])
G4.Qs.total.binomial[ , median:=median(PointEst), by=c("exon_pos", "Q")]
G4.Qs.total.binomial[, `:=`(Enrrichment=PointEst/median, Enrrichment_l=Lower/median, Enrrichment_u=Upper/median)]


ggplot(G4.Qs.total.binomial)+
  geom_line(aes(x=Position,y=Enrrichment, colour=Q)) +
  geom_ribbon(aes(x=Position,y=Enrrichment, fill=Q, ymin=Enrrichment_l, ymax=Enrrichment_u), alpha=0.3) +
  xlim(c(-150,150)) +
  facet_grid( . ~ exon_pos ) +
  theme(text = element_text(size=15)) +
  theme_bw()+labs(colour = "Splice site stregth quartile") +
   theme(legend.position = "top", legend.direction = "horizontal")


```


Then we explored the enrichment in other non-B DNA motifs


```{r, message=FALSE, error=FALSE, warning=FALSE}


DR.Q1_up_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_plus.q1.bed.DRs.list.out.num")
DR.Q2_up_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_plus.q2.bed.DRs.list.out.num")
DR.Q3_up_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_plus.q3.bed.DRs.list.out.num")
DR.Q4_up_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_plus.q4.bed.DRs.list.out.num")

DR.Q1_down_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_plus.q1.bed.DRs.list.out.num")
DR.Q2_down_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_plus.q2.bed.DRs.list.out.num")
DR.Q3_down_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_plus.q3.bed.DRs.list.out.num")
DR.Q4_down_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_plus.q4.bed.DRs.list.out.num")

DR.Q1_up_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_minus.q1.bed.DRs.list.out.num")
DR.Q2_up_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_minus.q2.bed.DRs.list.out.num")
DR.Q3_up_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_minus.q3.bed.DRs.list.out.num")
DR.Q4_up_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_minus.q4.bed.DRs.list.out.num")

DR.Q1_down_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_minus.q1.bed.DRs.list.out.num")
DR.Q2_down_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_minus.q2.bed.DRs.list.out.num")
DR.Q3_down_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_minus.q3.bed.DRs.list.out.num")
DR.Q4_down_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_minus.q4.bed.DRs.list.out.num")

DR.Qs.plus.up.list <- list(DR.Q1_up_plus, DR.Q2_up_plus, DR.Q3_up_plus, DR.Q4_up_plus)
DR.Qs.minus.list <- list(DR.Q1_up_minus, DR.Q2_up_minus, DR.Q3_up_minus, DR.Q4_up_minus)
DR.Qs.plus.down.list <- list(DR.Q1_down_plus, DR.Q2_down_plus, DR.Q3_down_plus, DR.Q4_down_plus)
DR.Qs.minus.down.list <- list(DR.Q1_down_minus, DR.Q2_down_minus, DR.Q3_down_minus, DR.Q4_down_minus)
DR.window_len = 2000

DR.Qs.total <- get_total_Qs(DR.Qs.plus.up.list, DR.Qs.minus.list, DR.Qs.plus.down.list, DR.Qs.minus.down.list, DR.window_len)


  

```





```{r, message=FALSE, error=FALSE, warning=FALSE}


H_DNA.Q1_up_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_plus.q1.bed.H_DNA.list.out.num")
H_DNA.Q2_up_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_plus.q2.bed.H_DNA.list.out.num")
H_DNA.Q3_up_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_plus.q3.bed.H_DNA.list.out.num")
H_DNA.Q4_up_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_plus.q4.bed.H_DNA.list.out.num")

H_DNA.Q1_down_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_plus.q1.bed.H_DNA.list.out.num")
H_DNA.Q2_down_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_plus.q2.bed.H_DNA.list.out.num")
H_DNA.Q3_down_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_plus.q3.bed.H_DNA.list.out.num")
H_DNA.Q4_down_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_plus.q4.bed.H_DNA.list.out.num")

H_DNA.Q1_up_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_minus.q1.bed.H_DNA.list.out.num")
H_DNA.Q2_up_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_minus.q2.bed.H_DNA.list.out.num")
H_DNA.Q3_up_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_minus.q3.bed.H_DNA.list.out.num")
H_DNA.Q4_up_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_minus.q4.bed.H_DNA.list.out.num")

H_DNA.Q1_down_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_minus.q1.bed.H_DNA.list.out.num")
H_DNA.Q2_down_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_minus.q2.bed.H_DNA.list.out.num")
H_DNA.Q3_down_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_minus.q3.bed.H_DNA.list.out.num")
H_DNA.Q4_down_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_minus.q4.bed.H_DNA.list.out.num")

H_DNA.Qs.plus.up.list <- list(H_DNA.Q1_up_plus, H_DNA.Q2_up_plus, H_DNA.Q3_up_plus, H_DNA.Q4_up_plus)
H_DNA.Qs.minus.list <- list(H_DNA.Q1_up_minus, H_DNA.Q2_up_minus, H_DNA.Q3_up_minus, H_DNA.Q4_up_minus)
H_DNA.Qs.plus.down.list <- list(H_DNA.Q1_down_plus, H_DNA.Q2_down_plus, H_DNA.Q3_down_plus, H_DNA.Q4_down_plus)
H_DNA.Qs.minus.down.list <- list(H_DNA.Q1_down_minus, H_DNA.Q2_down_minus, H_DNA.Q3_down_minus, H_DNA.Q4_down_minus)
H_DNA.window_len = 2000

H_DNA.Qs.total <- get_total_Qs(H_DNA.Qs.plus.up.list, H_DNA.Qs.minus.list, H_DNA.Qs.plus.down.list, H_DNA.Qs.minus.down.list, H_DNA.window_len)


```



```{r, message=FALSE, error=FALSE, warning=FALSE}


IR.Q1_up_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_plus.q1.bed.IRs.list.out.num")
IR.Q2_up_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_plus.q2.bed.IRs.list.out.num")
IR.Q3_up_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_plus.q3.bed.IRs.list.out.num")
IR.Q4_up_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_plus.q4.bed.IRs.list.out.num")

IR.Q1_down_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_plus.q1.bed.IRs.list.out.num")
IR.Q2_down_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_plus.q2.bed.IRs.list.out.num")
IR.Q3_down_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_plus.q3.bed.IRs.list.out.num")
IR.Q4_down_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_plus.q4.bed.IRs.list.out.num")

IR.Q1_up_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_minus.q1.bed.IRs.list.out.num")
IR.Q2_up_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_minus.q2.bed.IRs.list.out.num")
IR.Q3_up_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_minus.q3.bed.IRs.list.out.num")
IR.Q4_up_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_minus.q4.bed.IRs.list.out.num")

IR.Q1_down_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_minus.q1.bed.IRs.list.out.num")
IR.Q2_down_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_minus.q2.bed.IRs.list.out.num")
IR.Q3_down_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_minus.q3.bed.IRs.list.out.num")
IR.Q4_down_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_minus.q4.bed.IRs.list.out.num")

IR.Qs.plus.up.list <- list(IR.Q1_up_plus, IR.Q2_up_plus, IR.Q3_up_plus, IR.Q4_up_plus)
IR.Qs.minus.list <- list(IR.Q1_up_minus, IR.Q2_up_minus, IR.Q3_up_minus, IR.Q4_up_minus)
IR.Qs.plus.down.list <- list(IR.Q1_down_plus, IR.Q2_down_plus, IR.Q3_down_plus, IR.Q4_down_plus)
IR.Qs.minus.down.list <- list(IR.Q1_down_minus, IR.Q2_down_minus, IR.Q3_down_minus, IR.Q4_down_minus)
IR.window_len = 2000

IR.Qs.total <- get_total_Qs(IR.Qs.plus.up.list, IR.Qs.minus.list, IR.Qs.plus.down.list, IR.Qs.minus.down.list, IR.window_len)


```



```{r, message=FALSE, error=FALSE, warning=FALSE}


MR.Q1_up_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_plus.q1.bed.MRs.list.out.num")
MR.Q2_up_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_plus.q2.bed.MRs.list.out.num")
MR.Q3_up_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_plus.q3.bed.MRs.list.out.num")
MR.Q4_up_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_plus.q4.bed.MRs.list.out.num")

MR.Q1_down_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_plus.q1.bed.MRs.list.out.num")
MR.Q2_down_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_plus.q2.bed.MRs.list.out.num")
MR.Q3_down_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_plus.q3.bed.MRs.list.out.num")
MR.Q4_down_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_plus.q4.bed.MRs.list.out.num")

MR.Q1_up_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_minus.q1.bed.MRs.list.out.num")
MR.Q2_up_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_minus.q2.bed.MRs.list.out.num")
MR.Q3_up_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_minus.q3.bed.MRs.list.out.num")
MR.Q4_up_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_minus.q4.bed.MRs.list.out.num")

MR.Q1_down_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_minus.q1.bed.MRs.list.out.num")
MR.Q2_down_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_minus.q2.bed.MRs.list.out.num")
MR.Q3_down_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_minus.q3.bed.MRs.list.out.num")
MR.Q4_down_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_minus.q4.bed.MRs.list.out.num")

MR.Qs.plus.up.list <- list(MR.Q1_up_plus, MR.Q2_up_plus, MR.Q3_up_plus, MR.Q4_up_plus)
MR.Qs.minus.list <- list(MR.Q1_up_minus, MR.Q2_up_minus, MR.Q3_up_minus, MR.Q4_up_minus)
MR.Qs.plus.down.list <- list(MR.Q1_down_plus, MR.Q2_down_plus, MR.Q3_down_plus, MR.Q4_down_plus)
MR.Qs.minus.down.list <- list(MR.Q1_down_minus, MR.Q2_down_minus, MR.Q3_down_minus, MR.Q4_down_minus)
MR.window_len = 2000

MR.Qs.total <- get_total_Qs(MR.Qs.plus.up.list, MR.Qs.minus.list, MR.Qs.plus.down.list, MR.Qs.minus.down.list, MR.window_len)


```




```{r, message=FALSE, error=FALSE, warning=FALSE}


STR.Q1_up_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_plus.q1.bed.STRs.list.out.num")
STR.Q2_up_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_plus.q2.bed.STRs.list.out.num")
STR.Q3_up_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_plus.q3.bed.STRs.list.out.num")
STR.Q4_up_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_plus.q4.bed.STRs.list.out.num")

STR.Q1_down_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_plus.q1.bed.STRs.list.out.num")
STR.Q2_down_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_plus.q2.bed.STRs.list.out.num")
STR.Q3_down_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_plus.q3.bed.STRs.list.out.num")
STR.Q4_down_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_plus.q4.bed.STRs.list.out.num")

STR.Q1_up_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_minus.q1.bed.STRs.list.out.num")
STR.Q2_up_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_minus.q2.bed.STRs.list.out.num")
STR.Q3_up_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_minus.q3.bed.STRs.list.out.num")
STR.Q4_up_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_minus.q4.bed.STRs.list.out.num")

STR.Q1_down_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_minus.q1.bed.STRs.list.out.num")
STR.Q2_down_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_minus.q2.bed.STRs.list.out.num")
STR.Q3_down_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_minus.q3.bed.STRs.list.out.num")
STR.Q4_down_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_minus.q4.bed.STRs.list.out.num")

STR.Qs.plus.up.list <- list(STR.Q1_up_plus, STR.Q2_up_plus, STR.Q3_up_plus, STR.Q4_up_plus)
STR.Qs.minus.list <- list(STR.Q1_up_minus, STR.Q2_up_minus, STR.Q3_up_minus, STR.Q4_up_minus)
STR.Qs.plus.down.list <- list(STR.Q1_down_plus, STR.Q2_down_plus, STR.Q3_down_plus, STR.Q4_down_plus)
STR.Qs.minus.down.list <- list(STR.Q1_down_minus, STR.Q2_down_minus, STR.Q3_down_minus, STR.Q4_down_minus)
STR.window_len = 2000

STR.Qs.total <- get_total_Qs(STR.Qs.plus.up.list, STR.Qs.minus.list, STR.Qs.plus.down.list, STR.Qs.minus.down.list, STR.window_len)

```




```{r, message=FALSE, error=FALSE, warning=FALSE}


Z_DNA.Q1_up_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_plus.q1.bed.Z_DNA.list.out.num")
Z_DNA.Q2_up_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_plus.q2.bed.Z_DNA.list.out.num")
Z_DNA.Q3_up_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_plus.q3.bed.Z_DNA.list.out.num")
Z_DNA.Q4_up_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_plus.q4.bed.Z_DNA.list.out.num")

Z_DNA.Q1_down_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_plus.q1.bed.Z_DNA.list.out.num")
Z_DNA.Q2_down_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_plus.q2.bed.Z_DNA.list.out.num")
Z_DNA.Q3_down_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_plus.q3.bed.Z_DNA.list.out.num")
Z_DNA.Q4_down_plus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_plus.q4.bed.Z_DNA.list.out.num")

Z_DNA.Q1_up_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_minus.q1.bed.Z_DNA.list.out.num")
Z_DNA.Q2_up_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_minus.q2.bed.Z_DNA.list.out.num")
Z_DNA.Q3_up_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_minus.q3.bed.Z_DNA.list.out.num")
Z_DNA.Q4_up_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.up_minus.q4.bed.Z_DNA.list.out.num")

Z_DNA.Q1_down_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_minus.q1.bed.Z_DNA.list.out.num")
Z_DNA.Q2_down_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_minus.q2.bed.Z_DNA.list.out.num")
Z_DNA.Q3_down_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_minus.q3.bed.Z_DNA.list.out.num")
Z_DNA.Q4_down_minus <- read_dist_table("./All_non_Bs/q1_q4/exon.down_minus.q4.bed.Z_DNA.list.out.num")

Z_DNA.Qs.plus.up.list <- list(Z_DNA.Q1_up_plus, Z_DNA.Q2_up_plus, Z_DNA.Q3_up_plus, Z_DNA.Q4_up_plus)
Z_DNA.Qs.minus.list <- list(Z_DNA.Q1_up_minus, Z_DNA.Q2_up_minus, Z_DNA.Q3_up_minus, Z_DNA.Q4_up_minus)
Z_DNA.Qs.plus.down.list <- list(Z_DNA.Q1_down_plus, Z_DNA.Q2_down_plus, Z_DNA.Q3_down_plus, Z_DNA.Q4_down_plus)
Z_DNA.Qs.minus.down.list <- list(Z_DNA.Q1_down_minus, Z_DNA.Q2_down_minus, Z_DNA.Q3_down_minus, Z_DNA.Q4_down_minus)
Z_DNA.window_len = 2000

Z_DNA.Qs.total <- get_total_Qs(Z_DNA.Qs.plus.up.list, Z_DNA.Qs.minus.list, Z_DNA.Qs.plus.down.list, Z_DNA.Qs.minus.down.list, Z_DNA.window_len)

```




```{r, fig.height=10, fig.width=7}
G4.Qs.total[, Non_B:="G4"]
DR.Qs.total[, Non_B:="DR"]
H_DNA.Qs.total[, Non_B:="H-DNA"]
IR.Qs.total[, Non_B:="IR"]
MR.Qs.total[, Non_B:="MR"]
STR.Qs.total[, Non_B:="STR"]
Z_DNA.Qs.total[, Non_B:="Z-DNA"]



Non_B.Qs.TOTAL <- rbind(G4.Qs.total, DR.Qs.total, H_DNA.Qs.total, IR.Qs.total, MR.Qs.total, STR.Qs.total, Z_DNA.Qs.total)

Non_B.Qs.TOTAL$Q <- mapvalues(Non_B.Qs.TOTAL$Q,  from =c(1:4), to =c("Q1", "Q2", "Q3", "Q4"))



ggplot(Non_B.Qs.TOTAL)+
  geom_line(aes(x=Position,y=Enrrichment, colour=Q)) +
  xlim(c(-150,150)) +
  facet_grid( Non_B ~ exon_pos ) +
  labs(colour = "Splice site stregth quartile") +
  theme_bw()  + 
  theme(legend.position = "top", legend.direction = "horizontal")


```





```{r}
Non_B.Qs.TOTAL.Enrrichment_Peak <- Non_B.Qs.TOTAL[  , .(Enrrichment_Peak=max(Enrrichment)) , by= c("Non_B", "exon_pos", "Q")]

write.table(Non_B.Qs.TOTAL.Enrrichment_Peak, file = "./Tables/Non_B.Qs.TOTAL.Enrrichment_Peak.txt", append = FALSE, quote = FALSE, sep = "\t",
            eol = "\n", na = "NA", dec = ".", row.names = FALSE,
            col.names = TRUE, qmethod = c("escape", "double"),
            fileEncoding = "")
```







# Template - non template


To investigate if the distribution of non-B DNA motifs is biased by transcription direction, we generate `.num` files where the counts were separated by template and non templated strands:


* Template = plus_minus, minus_plus
* Non-template = minus_minus, plus_plus




```{r, message=FALSE, error=FALSE, warning=FALSE}


G4.template.Q1_up_plus <- read_dist_table("./template_non_template/exon.up_plus.q1.bed.G4s_minus_plus.list.final.num")
G4.template.Q2_up_plus <- read_dist_table("./template_non_template/exon.up_plus.q2.bed.G4s_minus_plus.list.final.num")
G4.template.Q3_up_plus <- read_dist_table("./template_non_template/exon.up_plus.q3.bed.G4s_minus_plus.list.final.num")
G4.template.Q4_up_plus <- read_dist_table("./template_non_template/exon.up_plus.q4.bed.G4s_minus_plus.list.final.num")

G4.template.Q1_up_minus <- read_dist_table("./template_non_template/exon.up_minus.q1.bed.G4s_plus_minus.list.final.num")
G4.template.Q2_up_minus <- read_dist_table("./template_non_template/exon.up_minus.q2.bed.G4s_plus_minus.list.final.num")
G4.template.Q3_up_minus <- read_dist_table("./template_non_template/exon.up_minus.q3.bed.G4s_plus_minus.list.final.num")
G4.template.Q4_up_minus <- read_dist_table("./template_non_template/exon.up_minus.q4.bed.G4s_plus_minus.list.final.num")

G4.template.Q1_down_plus <- read_dist_table("./template_non_template/exon.down_plus.q1.bed.G4s_minus_plus.list.final.num")
G4.template.Q2_down_plus <- read_dist_table("./template_non_template/exon.down_plus.q2.bed.G4s_minus_plus.list.final.num")
G4.template.Q3_down_plus <- read_dist_table("./template_non_template/exon.down_plus.q3.bed.G4s_minus_plus.list.final.num")
G4.template.Q4_down_plus <- read_dist_table("./template_non_template/exon.down_plus.q4.bed.G4s_minus_plus.list.final.num")

G4.template.Q1_down_minus <- read_dist_table("./template_non_template/exon.down_minus.q1.bed.G4s_plus_minus.list.final.num")
G4.template.Q2_down_minus <- read_dist_table("./template_non_template/exon.down_minus.q2.bed.G4s_plus_minus.list.final.num")
G4.template.Q3_down_minus <- read_dist_table("./template_non_template/exon.down_minus.q3.bed.G4s_plus_minus.list.final.num")
G4.template.Q4_down_minus <- read_dist_table("./template_non_template/exon.down_minus.q4.bed.G4s_plus_minus.list.final.num")


G4.non_template.Q1_up_plus <- read_dist_table("./template_non_template/exon.up_plus.q1.bed.G4s_plus_plus.list.final.num")
G4.non_template.Q2_up_plus <- read_dist_table("./template_non_template/exon.up_plus.q2.bed.G4s_plus_plus.list.final.num")
G4.non_template.Q3_up_plus <- read_dist_table("./template_non_template/exon.up_plus.q3.bed.G4s_plus_plus.list.final.num")
G4.non_template.Q4_up_plus <- read_dist_table("./template_non_template/exon.up_plus.q4.bed.G4s_plus_plus.list.final.num")

G4.non_template.Q1_up_minus <- read_dist_table("./template_non_template/exon.up_minus.q1.bed.G4s_minus_minus.list.final.num")
G4.non_template.Q2_up_minus <- read_dist_table("./template_non_template/exon.up_minus.q2.bed.G4s_minus_minus.list.final.num")
G4.non_template.Q3_up_minus <- read_dist_table("./template_non_template/exon.up_minus.q3.bed.G4s_minus_minus.list.final.num")
G4.non_template.Q4_up_minus <- read_dist_table("./template_non_template/exon.up_minus.q4.bed.G4s_minus_minus.list.final.num")

G4.non_template.Q1_down_plus <- read_dist_table("./template_non_template/exon.down_plus.q1.bed.G4s_plus_plus.list.final.num")
G4.non_template.Q2_down_plus <- read_dist_table("./template_non_template/exon.down_plus.q2.bed.G4s_plus_plus.list.final.num")
G4.non_template.Q3_down_plus <- read_dist_table("./template_non_template/exon.down_plus.q3.bed.G4s_plus_plus.list.final.num")
G4.non_template.Q4_down_plus <- read_dist_table("./template_non_template/exon.down_plus.q4.bed.G4s_plus_plus.list.final.num")

G4.non_template.Q1_down_minus <- read_dist_table("./template_non_template/exon.down_minus.q1.bed.G4s_minus_minus.list.final.num")
G4.non_template.Q2_down_minus <- read_dist_table("./template_non_template/exon.down_minus.q2.bed.G4s_minus_minus.list.final.num")
G4.non_template.Q3_down_minus <- read_dist_table("./template_non_template/exon.down_minus.q3.bed.G4s_minus_minus.list.final.num")
G4.non_template.Q4_down_minus <- read_dist_table("./template_non_template/exon.down_minus.q4.bed.G4s_minus_minus.list.final.num")


```



```{r}
G4.template.Qs.plus.up.list <- list(G4.template.Q1_up_plus, G4.template.Q2_up_plus, G4.template.Q3_up_plus, G4.template.Q4_up_plus)
G4.template.Qs.minus.list <- list(G4.template.Q1_up_minus, G4.template.Q2_up_minus, G4.template.Q3_up_minus, G4.template.Q4_up_minus)
G4.template.Qs.plus.down.list <- list(G4.template.Q1_down_plus, G4.template.Q2_down_plus, G4.template.Q3_down_plus, G4.template.Q4_down_plus)
G4.template.Qs.minus.down.list <- list(G4.template.Q1_down_minus, G4.template.Q2_down_minus, G4.template.Q3_down_minus, G4.template.Q4_down_minus)
G4.template.window_len = 2000

G4.template.Qs.total <- get_total_Qs(G4.template.Qs.plus.up.list, G4.template.Qs.minus.list, G4.template.Qs.plus.down.list, G4.template.Qs.minus.down.list, G4.template.window_len)
```



```{r}
G4.non_template.Qs.plus.up.list <- list(G4.non_template.Q1_up_plus, G4.non_template.Q2_up_plus, G4.non_template.Q3_up_plus, G4.non_template.Q4_up_plus)
G4.non_template.Qs.minus.list <- list(G4.non_template.Q1_up_minus, G4.non_template.Q2_up_minus, G4.non_template.Q3_up_minus, G4.non_template.Q4_up_minus)
G4.non_template.Qs.plus.down.list <- list(G4.non_template.Q1_down_plus, G4.non_template.Q2_down_plus, G4.non_template.Q3_down_plus, G4.non_template.Q4_down_plus)
G4.non_template.Qs.minus.down.list <- list(G4.non_template.Q1_down_minus, G4.non_template.Q2_down_minus, G4.non_template.Q3_down_minus, G4.non_template.Q4_down_minus)
G4.non_template.window_len = 2000

G4.non_template.Qs.total <- get_total_Qs(G4.non_template.Qs.plus.up.list, G4.non_template.Qs.minus.list, G4.non_template.Qs.plus.down.list, G4.non_template.Qs.minus.down.list, G4.template.window_len)
```


```{r}

G4.template.Qs.total
G4.non_template.Qs.total


G4.template_non_template.Qs.total <- rbind(G4.template.Qs.total, G4.non_template.Qs.total)
  
half_n <- nrow(G4.template_non_template.Qs.total)/2
G4.template_non_template.Qs.total[, Strand:=rep(c("Template", "Non-template"), each=half_n) ]
  
  
  
  
G4.template_non_template.Qs.total$Strand <-  factor(G4.template_non_template.Qs.total$Strand, levels=c("Template", "Non-template" ))

  
```




```{r}



G4.template_non_template.Qs.total$Q <- mapvalues(G4.template_non_template.Qs.total$Q,  from =c(1:4), to =c("Q1", "Q2", "Q3", "Q4"))





G4.template_non_template.Qs.total.binomial <- cbind(G4.template_non_template.Qs.total,  G4.template_non_template.Qs.total[, binconf(Occurrences, hg19.nexons/8) ])
G4.template_non_template.Qs.total.binomial[, median:=NULL]

G4.template_non_template.Qs.total.binomial[ , median:=median(PointEst), by=c("exon_pos", "Q", "Strand")]
G4.template_non_template.Qs.total.binomial[, `:=`(Enrrichment=PointEst/median, Enrrichment_l=Lower/median, Enrrichment_u=Upper/median)]


Fig2.A <- ggplot(G4.template_non_template.Qs.total.binomial)+
  geom_line(aes(x=Position,y=Enrrichment, colour=Q)) +
   geom_ribbon(aes(x=Position, fill=Q, ymin=Enrrichment_l, ymax=Enrrichment_u), alpha=0.3) +
  xlim(c(-150,150)) +
  facet_grid( Strand ~ exon_pos ) +
  labs(colour = "Splice site stregth quartile") +
  theme_bw()  + 
  theme(legend.position = "top", legend.direction = "horizontal")

Fig2.A

```




```{r}

G4.template_non_template.total <- G4.template_non_template.Qs.total[ , .(Occurrences=sum(Occurrences)) , by=c("Strand", "exon_pos", "Position")]

G4.template_non_template.total[ , median:=median(Occurrences) , by=c("Strand", "exon_pos")]

G4.template_non_template.total[ , Enrrichment:=Occurrences/median , by=c("Strand", "exon_pos", "Position")]


ggplot(G4.template_non_template.total)+
  geom_line(aes(x=Position,y=Enrrichment)) +
  xlim(c(-150,150)) +
  facet_grid( Strand ~ exon_pos ) +
  labs(colour = "Splice site stregth quartile") +
  theme_bw()  + 
  theme(legend.position = "top", legend.direction = "horizontal")



```


```{r}

G4.template_non_template.total_Peak <- G4.template_non_template.total[  , .(Enrrichment_Peak=max(Enrrichment)) , by= c("Strand", "exon_pos")]


write.table(G4.template_non_template.total_Peak, file = "./Tables/G4.template_non_template.total_Peak.txt", append = FALSE, quote = FALSE, sep = "\t",
            eol = "\n", na = "NA", dec = ".", row.names = FALSE,
            col.names = TRUE, qmethod = c("escape", "double"),
            fileEncoding = "")
```



```{r}


G4.template_non_template.Qs.total.Enrrichment_Peak <- G4.template_non_template.Qs.total[  , .(Enrrichment_Peak=max(Enrrichment)) , by= c( "exon_pos", "Q", "Strand" )]

write.table(G4.template_non_template.Qs.total.Enrrichment_Peak, file = "./Tables/G4.template_non_template.Qs.total.Enrrichment_Peak.txt", append = FALSE, quote = FALSE, sep = "\t",
            eol = "\n", na = "NA", dec = ".", row.names = FALSE,
            col.names = TRUE, qmethod = c("escape", "double"),
            fileEncoding = "")

```


# G4 RUNS

Here we explored the enrichment of G-quadruplexes with different G-run length


```{r,  message=FALSE, error=FALSE, warning=FALSE}

G1_up_plus <- read_dist_table("./G4_runs/exon.up_plus.bed.Single_G_runs.list.out.num")
G2_up_plus <- read_dist_table("./G4_runs/exon.up_plus.bed.Twice_G_runs.list.out.num")
G3_up_plus <- read_dist_table("./G4_runs/exon.up_plus.bed.Three_G_runs.list.out.num")
G4_up_plus <- read_dist_table("./G4_runs/exon.up_plus.bed.Four_G_runs.list.out.num")
G5_up_plus <- read_dist_table("./G4_runs/exon.up_plus.bed.Five_G_runs.list.out.num")
G6_up_plus <- read_dist_table("./G4_runs/exon.up_plus.bed.Six_G_runs.list.out.num")

G1_up_minus <- read_dist_table("./G4_runs/exon.up_minus.bed.Single_G_runs.list.out.num")
G2_up_minus <- read_dist_table("./G4_runs/exon.up_minus.bed.Twice_G_runs.list.out.num")
G3_up_minus <- read_dist_table("./G4_runs/exon.up_minus.bed.Three_G_runs.list.out.num")
G4_up_minus <- read_dist_table("./G4_runs/exon.up_minus.bed.Four_G_runs.list.out.num")
G5_up_minus <- read_dist_table("./G4_runs/exon.up_minus.bed.Five_G_runs.list.out.num")
G6_up_minus <- read_dist_table("./G4_runs/exon.up_minus.bed.Six_G_runs.list.out.num")

G1_down_plus <- read_dist_table("./G4_runs/exon.down_plus.bed.Single_G_runs.list.out.num")
G2_down_plus <- read_dist_table("./G4_runs/exon.down_plus.bed.Twice_G_runs.list.out.num")
G3_down_plus <- read_dist_table("./G4_runs/exon.down_plus.bed.Three_G_runs.list.out.num")
G4_down_plus <- read_dist_table("./G4_runs/exon.down_plus.bed.Four_G_runs.list.out.num")
G5_down_plus <- read_dist_table("./G4_runs/exon.down_plus.bed.Five_G_runs.list.out.num")
G6_down_plus <- read_dist_table("./G4_runs/exon.down_plus.bed.Six_G_runs.list.out.num")

G1_down_minus <- read_dist_table("./G4_runs/exon.down_minus.bed.Single_G_runs.list.out.num")
G2_down_minus <- read_dist_table("./G4_runs/exon.down_minus.bed.Twice_G_runs.list.out.num")
G3_down_minus <- read_dist_table("./G4_runs/exon.down_minus.bed.Three_G_runs.list.out.num")
G4_down_minus <- read_dist_table("./G4_runs/exon.down_minus.bed.Four_G_runs.list.out.num")  
G5_down_minus <- read_dist_table("./G4_runs/exon.down_minus.bed.Five_G_runs.list.out.num")
G6_down_minus <- read_dist_table("./G4_runs/exon.down_minus.bed.Six_G_runs.list.out.num")
```



```{r}
Gruns.up_plus.list <- list(G1_up_plus, G2_up_plus, G3_up_plus, G4_up_plus, G5_up_plus, G6_up_plus)
Gruns.up_minus.list <- list(G1_up_minus, G2_up_minus, G3_up_minus, G4_up_minus, G5_up_minus, G6_up_minus)

Gruns.down_plus.list <- list(G1_down_plus, G2_down_plus, G3_down_plus, G4_down_plus, G5_down_plus, G6_down_plus)
Gruns.down_minus.list <- list(G1_down_minus, G2_down_minus, G3_down_minus, G4_down_minus, G5_down_minus, G6_down_minus)


Gruns.window_len = 2000

Gruns.total <- get_total_Qs(Gruns.up_plus.list, Gruns.up_minus.list, Gruns.down_plus.list, Gruns.down_minus.list, Gruns.window_len)

Gruns.total[, G_run_length:=Q]
```



```{r}


Gruns.total.binomial <- cbind(Gruns.total,  Gruns.total[, binconf(Occurrences, hg19.nexons) ])
Gruns.total.binomial[, median:=NULL]

Gruns.total.binomial[ , median:=median(PointEst), by=c("exon_pos", "Q" )]
Gruns.total.binomial[, `:=`(Enrrichment=PointEst/median, Enrrichment_l=Lower/median, Enrrichment_u=Upper/median)]


Fig2.B <- ggplot(Gruns.total.binomial)+
  geom_line(aes(x=Position,y=Enrrichment, colour=G_run_length)) +
  geom_ribbon(aes(x=Position, fill=Q, ymin=Enrrichment_l, ymax=Enrrichment_u), alpha=0.3) +
  xlim(c(-100,100)) +
  facet_grid( . ~ exon_pos ) +
  theme(text = element_text(size=15)) +
  theme_bw() +
  labs(colour = "Number of consecutive G-runs") +
  theme(legend.position = "top", legend.direction = "horizontal")


Fig2.B
```





```{r}

Gruns.total.Enrrichment_Peak <- Gruns.total[  , .(Enrrichment_Peak=max(Enrrichment)) , by= c( "exon_pos", "G_run_length")]

write.table(Gruns.total.Enrrichment_Peak, file = "./Tables/Gruns.total.Enrrichment_Peak.txt", append = FALSE, quote = FALSE, sep = "\t",
            eol = "\n", na = "NA", dec = ".", row.names = FALSE,
            col.names = TRUE, qmethod = c("escape", "double"),
            fileEncoding = "")
```







## G4 enrichment across evolution




We analyzed the G4 enrichment across splice sites from different species



```{r,  message=FALSE, error=FALSE, warning=FALSE}

susScr11.up_plus <- read_dist_table("./Evolution/susScr11.exon.up_plus.bed.score.susScr11.txt..list.out.num")
susScr11.down_plus <- read_dist_table("./Evolution/susScr11.exon.down_plus.bed.score.susScr11.txt..list.out.num")
susScr11.up_minus <- read_dist_table("./Evolution/susScr11.exon.up_minus.bed.score.susScr11.txt..list.out.num")
susScr11.down_minus <- read_dist_table("./Evolution/susScr11.exon.down_minus.bed.score.susScr11.txt..list.out.num")
susScr11.up_minus[,Position:=Position*-1]
susScr11.down_minus[,Position:=Position*-1]

susScr11.TOTAL <- plot_density(susScr11.up_plus, susScr11.up_minus, susScr11.down_plus, susScr11.down_minus)


```





```{r,  message=FALSE, error=FALSE, warning=FALSE}

TAIR10.up_plus <- read_dist_table("./Evolution/TAIR10.exon.up_plus.bed.score.TAIR10.txt..list.out.num")
TAIR10.down_plus <- read_dist_table("./Evolution/TAIR10.exon.down_plus.bed.score.TAIR10.txt..list.out.num")
TAIR10.up_minus <- read_dist_table("./Evolution/TAIR10.exon.up_minus.bed.score.TAIR10.txt..list.out.num")
TAIR10.down_minus <- read_dist_table("./Evolution/TAIR10.exon.down_minus.bed.score.TAIR10.txt..list.out.num")
TAIR10.up_minus[,Position:=Position*-1]
TAIR10.down_minus[,Position:=Position*-1]

TAIR10.TOTAL <- plot_density(TAIR10.up_plus, TAIR10.up_minus, TAIR10.down_plus, TAIR10.down_minus)


```






```{r,  message=FALSE, error=FALSE, warning=FALSE}

anoCar2.up_plus <- read_dist_table("./Evolution/anoCar2.exon.up_plus.bed.score.anoCar2.txt..list.out.num")
anoCar2.down_plus <- read_dist_table("./Evolution/anoCar2.exon.down_plus.bed.score.anoCar2.txt..list.out.num")
anoCar2.up_minus <- read_dist_table("./Evolution/anoCar2.exon.up_minus.bed.score.anoCar2.txt..list.out.num")
anoCar2.down_minus <- read_dist_table("./Evolution/anoCar2.exon.down_minus.bed.score.anoCar2.txt..list.out.num")
anoCar2.up_minus[,Position:=Position*-1]
anoCar2.down_minus[,Position:=Position*-1]

anoCar2.TOTAL <- plot_density(anoCar2.up_plus, anoCar2.up_minus, anoCar2.down_plus, anoCar2.down_minus)


```



```{r,  message=FALSE, error=FALSE, warning=FALSE}

xenTro9.up_plus <- read_dist_table("./Evolution/xenTro9.exon.up_plus.bed.score.xenTro9.txt..list.out.num")
xenTro9.down_plus <- read_dist_table("./Evolution/xenTro9.exon.down_plus.bed.score.xenTro9.txt..list.out.num")
xenTro9.up_minus <- read_dist_table("./Evolution/xenTro9.exon.up_minus.bed.score.xenTro9.txt..list.out.num")
xenTro9.down_minus <- read_dist_table("./Evolution/xenTro9.exon.down_minus.bed.score.xenTro9.txt..list.out.num")
xenTro9.up_minus[,Position:=Position*-1]
xenTro9.down_minus[,Position:=Position*-1]

xenTro9.TOTAL <- plot_density(xenTro9.up_plus, xenTro9.up_minus, xenTro9.down_plus, xenTro9.down_minus)


```




```{r,  message=FALSE, error=FALSE, warning=FALSE}

ce10.up_plus <- read_dist_table("./Evolution/ce10.exon.up_plus.bed.score.ce10.txt..list.out.num")
ce10.down_plus <- read_dist_table("./Evolution/ce10.exon.down_plus.bed.score.ce10.txt..list.out.num")
ce10.up_minus <- read_dist_table("./Evolution/ce10.exon.up_minus.bed.score.ce10.txt..list.out.num")
ce10.down_minus <- read_dist_table("./Evolution/ce10.exon.down_minus.bed.score.ce10.txt..list.out.num")
ce10.up_minus[,Position:=Position*-1]
ce10.down_minus[,Position:=Position*-1]

ce10.TOTAL <- plot_density(ce10.up_plus, ce10.up_minus, ce10.down_plus, ce10.down_minus)


```


```{r, message=FALSE, error=FALSE, warning=FALSE}

danRer11.up_plus <- read_dist_table("./Evolution/danRer11.exon.up_plus.bed.score.danRer11.txt..list.out.num")
danRer11.down_plus <- read_dist_table("./Evolution/danRer11.exon.down_plus.bed.score.danRer11.txt..list.out.num")
danRer11.up_minus <- read_dist_table("./Evolution/danRer11.exon.up_minus.bed.score.danRer11.txt..list.out.num")
danRer11.down_minus <- read_dist_table("./Evolution/danRer11.exon.down_minus.bed.score.danRer11.txt..list.out.num")
danRer11.up_minus[,Position:=Position*-1]
danRer11.down_minus[,Position:=Position*-1]

danRer11.TOTAL <- plot_density(danRer11.up_plus, danRer11.up_minus, danRer11.down_plus, danRer11.down_minus)


```

```{r,  message=FALSE, error=FALSE, warning=FALSE}

mm10.up_plus <- read_dist_table("./Evolution/mm10.exon.up_plus.bed.score.mm10.txt..list.out.num")
mm10.down_plus <- read_dist_table("./Evolution/mm10.exon.down_plus.bed.score.mm10.txt..list.out.num")
mm10.up_minus <- read_dist_table("./Evolution/mm10.exon.up_minus.bed.score.mm10.txt..list.out.num")
mm10.down_minus <- read_dist_table("./Evolution/mm10.exon.down_minus.bed.score.mm10.txt..list.out.num")
mm10.up_minus[,Position:=Position*-1]
mm10.down_minus[,Position:=Position*-1]

mm10.TOTAL <- plot_density(mm10.up_plus, mm10.up_minus, mm10.down_plus, mm10.down_minus)


```


```{r,  message=FALSE, error=FALSE, warning=FALSE}

hg19.up_plus <- read_dist_table("./Evolution/hg19.exon.up_plus.bed.score.hg19.txt..list.out.num")
hg19.down_plus <- read_dist_table("./Evolution/hg19.exon.down_plus.bed.score.hg19.txt..list.out.num")
hg19.up_minus <- read_dist_table("./Evolution/hg19.exon.up_minus.bed.score.hg19.txt..list.out.num")
hg19.down_minus <- read_dist_table("./Evolution/hg19.exon.down_minus.bed.score.hg19.txt..list.out.num")
hg19.up_minus[,Position:=Position*-1]
hg19.down_minus[,Position:=Position*-1]

hg19.TOTAL <- plot_density(hg19.up_plus, hg19.up_minus, hg19.down_plus, hg19.down_minus)


```




```{r,  message=FALSE, error=FALSE, warning=FALSE}

dm6.up_plus <- read_dist_table("./Evolution/dm6.exon.up_plus.bed.score.dm6.txt..list.out.num")
dm6.down_plus <- read_dist_table("./Evolution/dm6.exon.down_plus.bed.score.dm6.txt..list.out.num")
dm6.up_minus <- read_dist_table("./Evolution/dm6.exon.up_minus.bed.score.dm6.txt..list.out.num")
dm6.down_minus <- read_dist_table("./Evolution/dm6.exon.down_minus.bed.score.dm6.txt..list.out.num")
dm6.up_minus[,Position:=Position*-1]
dm6.down_minus[,Position:=Position*-1]

dm6.TOTAL <- plot_density(dm6.up_plus, dm6.up_minus, dm6.down_plus, dm6.down_minus)


```



```{r, message=FALSE, error=FALSE, warning=FALSE}

galGal5.up_plus <- read_dist_table("./Evolution/galGal5.exon.up_plus.bed.score.galGal5.txt..list.out.num")
galGal5.down_plus <- read_dist_table("./Evolution/galGal5.exon.down_plus.bed.score.galGal5.txt..list.out.num")
galGal5.up_minus <- read_dist_table("./Evolution/galGal5.exon.up_minus.bed.score.galGal5.txt..list.out.num")
galGal5.down_minus <- read_dist_table("./Evolution/galGal5.exon.down_minus.bed.score.galGal5.txt..list.out.num")
galGal5.up_minus[,Position:=Position*-1]
galGal5.down_minus[,Position:=Position*-1]

galGal5.TOTAL <- plot_density(galGal5.up_plus, galGal5.up_minus, galGal5.down_plus, galGal5.down_minus)


```





```{r, message=FALSE, error=FALSE, warning=FALSE}

sacCer.up_plus <- read_dist_table("./Evolution/sacCer3.exon.up_plus.bed.score.sacCer3.txt..list.out.num")
sacCer.down_plus <- read_dist_table("./Evolution/sacCer3.exon.down_plus.bed.score.sacCer3.txt..list.out.num")
sacCer.up_minus <- read_dist_table("./Evolution/sacCer3.exon.up_minus.bed.score.sacCer3.txt..list.out.num")
sacCer.down_minus <- read_dist_table("./Evolution/sacCer3.exon.down_minus.bed.score.sacCer3.txt..list.out.num")
sacCer.up_minus[,Position:=Position*-1]
sacCer.down_minus[,Position:=Position*-1]

sacCer.TOTAL <- plot_density(sacCer.up_plus, sacCer.up_minus, sacCer.down_plus, sacCer.down_minus)


```





```{r }



ce10.TOTAL[, species:="C. elegans"] 
hg19.TOTAL[, species:="H. sapiens"] 
mm10.TOTAL[, species:="M. musculus"]
danRer11.TOTAL[, species:="D. rerio"] 
dm6.TOTAL[, species:="D. melanogaster"]
galGal5.TOTAL[, species:="G. gallus"]
sacCer.TOTAL[, species:="S. cerevisiae"]


xenTro9.TOTAL[ , species:="X. tropicalis" ]
anoCar2.TOTAL[  , species:="A. carolinensis" ]
TAIR10.TOTAL[  , species:="A. thaliana" ]
susScr11.TOTAL[ , species:="S. scrofa"]

All.species.TOTAL <- rbind(hg19.TOTAL, mm10.TOTAL, galGal5.TOTAL, danRer11.TOTAL, anoCar2.TOTAL, xenTro9.TOTAL,  susScr11.TOTAL  )


#All.species.TOTAL.binomial <- cbind(All.species.TOTAL,  Gruns.total[, binconf(Occurrences, hg19.nexons) ])
#All.species.TOTAL.binomial[, median:=NULL]

#All.species.TOTAL.binomial[ , median:=median(PointEst), by=c("exon_pos", "species" )]
#All.species.TOTAL.binomial[, `:=`(Enrrichment=PointEst/median, Enrrichment_l=Lower/median, Enrrichment_u=Upper/median)]



Fig6.B <- ggplot(All.species.TOTAL)+
    geom_line(aes(x=Position,y=Enrrichment, color=species)) +
    #   geom_ribbon(aes(x=Position, fill=species, ymin=Enrrichment_l, ymax=Enrrichment_u), alpha=0.3) +
    xlim(c(-300,300)) +
    facet_grid( . ~ exon_pos ) +
  theme(plot.title = element_text(hjust = 0.5)) +
  theme_bw() +
    theme(legend.position = "top", legend.direction = "horizontal") 

Fig6.B

```

We plot other species that did not show strong enrichment by separate

```{r}



All.species.TOTAL <- rbind(hg19.TOTAL, mm10.TOTAL, galGal5.TOTAL, danRer11.TOTAL, anoCar2.TOTAL, xenTro9.TOTAL,  susScr11.TOTAL  )
Sup6.A <- ggplot(All.species.TOTAL)+
    geom_line(aes(x=Position,y=Enrrichment, color=species)) +
    #   geom_ribbon(aes(x=Position, fill=species, ymin=Enrrichment_l, ymax=Enrrichment_u), alpha=0.3) +
    xlim(c(-300,300)) +
    ggtitle( "Vertebrates") +
    facet_grid( . ~ exon_pos ) +
      labs(color="Specie") +
    theme(plot.title = element_text(hjust = 0.5)) +
    theme(legend.position = "top", legend.direction = "horizontal") 

All.species.TOTAL <- rbind(ce10.TOTAL, dm6.TOTAL, TAIR10.TOTAL  )
Sup6.B <- ggplot(All.species.TOTAL)+
    geom_line(aes(x=Position,y=Enrrichment, color=species)) +
    xlim(c(-300,300)) +
    ggtitle( "Non-vertebrates") +
      labs(color="Specie") +
    facet_grid( . ~ exon_pos ) +
  theme(plot.title = element_text(hjust = 0.5)) +
    theme(legend.position = "top", legend.direction = "horizontal") 
```



```{r, fig.height=10, fig.width=10}
plot_grid(Sup6.A,Sup6.B, labels = "AUTO", ncol = 1 )
```

# G4 and gene structure


## Flanking intron size 

Here we analized the intron size of distribution of exon which are flanked by G4 (on an 100nt window)



```{r}


intron_size_upstream_non_template  <- rbind(fread('./Intron_size/Introns.hg19_upstream_100nt_window.bed.plus_strand_genome.plus_G4s'),  
 fread('./Intron_size/Introns.hg19_upstream_100nt_window.bed.minus_strand_genome.minus_G4s'))

intron_size_downstream_non_template  <- rbind(fread('./Intron_size/Introns.hg19_downstream_100nt_window.bed.plus_strand_genome.plus_G4s'),  
 fread('./Intron_size/Introns.hg19_downstream_100nt_window.bed.minus_strand_genome.minus_G4s'))


intron_size_upstream_template  <- rbind(fread('./Intron_size/Introns.hg19_upstream_100nt_window.bed.plus_strand_genome.minus_G4s'),  
 fread('./Intron_size/Introns.hg19_upstream_100nt_window.bed.minus_strand_genome.plus_G4s'))

intron_size_downstream_template  <- rbind(fread('./Intron_size/Introns.hg19_downstream_100nt_window.bed.minus_strand_genome.plus_G4s'),  
 fread('./Intron_size/Introns.hg19_downstream_100nt_window.bed.plus_strand_genome.minus_G4s'))


intron_size_upstream_non_template[ , `:=`(exon_pos="upstream", strand="non_template") ]
intron_size_downstream_non_template[ , `:=`(exon_pos="downstream", strand="non_template") ]
intron_size_upstream_template[ , `:=`(exon_pos="upstream", strand="template") ]
intron_size_downstream_template[ , `:=`(exon_pos="downstream", strand="template") ]

intron_size_TOTAL <- rbind(intron_size_upstream_non_template, intron_size_downstream_non_template, intron_size_upstream_template, intron_size_downstream_template )

colnames(intron_size_TOTAL) <- c("chrom", "start", "end", "strand", "intron_number", "intron_size", "Transcript", "G4", "exon_pos", "Strand"  )


```



```{r}
intron_size_TOTAL[  G4>0, G4_type:="With G4" ]
intron_size_TOTAL[  G4==0, G4_type:="Without G4" ]

intron_size_TOTAL$exon_pos <- factor(intron_size_TOTAL$exon_pos, levels=c("upstream", "downstream"))

Fig3.A <- ggplot(data=intron_size_TOTAL) +
  geom_boxplot( aes(x=G4_type, y=log10(intron_size) ) ) +
  facet_grid( . ~  exon_pos ) +
  xlab("")+
  ylab("log10(Intron size)") +
  theme_bw()


Fig3.A

```


```{r}
long_introns.5ss <- fread("./Intron_size/GC_correction/long_introns.same_GC.bed.intron_5ss_.bed.closest.G4s")
short_introns.5ss <- fread("./Intron_size/GC_correction/short_introns.same_GC.bed.intron_5ss_.bed.closest.G4s")
long_introns.3ss <- fread("./Intron_size/GC_correction/long_introns.same_GC.bed.intron_3ss_.bed.closest.G4s")
short_introns.3ss <- fread("./Intron_size/GC_correction/short_introns.same_GC.bed.intron_3ss_.bed.closest.G4s")


colnames(long_introns.5ss) <- c("i.chrom", "i.start", "i.end", "i.id", "i.GC", "i.strand", "g.chrom", "g.start", "g.end", "dist")
colnames(short_introns.5ss) <- c("i.chrom", "i.start", "i.end", "i.id", "i.GC", "i.strand", "g.chrom", "g.start", "g.end", "dist")
colnames(long_introns.3ss) <- c("i.chrom", "i.start", "i.end", "i.id", "i.GC", "i.strand", "g.chrom", "g.start", "g.end", "dist")
colnames(short_introns.3ss) <- c("i.chrom", "i.start", "i.end", "i.id", "i.GC", "i.strand", "g.chrom", "g.start", "g.end", "dist")

long_introns.bed <- fread("./Intron_size/GC_correction/long_introns.same_GC.bed")
short_introns.bed <- fread("./Intron_size/GC_correction/short_introns.same_GC.bed")
long_short_correspondence <-  cbind(long_introns.bed, short_introns.bed)[ , c(4, 2, 3 , 5,10, 8, 9, 11)]

colnames(long_short_correspondence) <- c("l.id", "l.start", "l.end", "l.GC", "s.id", "s.start", "s.end", "s.GC"  )
long_short_correspondence[, s.len:=(s.end-s.start)]


long_introns.5ss[, G4.5ss:=FALSE ]
long_introns.5ss[ dist<=100 & i.strand=="+" & i.start<g.end , G4.5ss:=TRUE ]
long_introns.5ss[ dist<=100 & i.strand=="-" & i.start>g.end , G4.5ss:=TRUE ]

short_introns.5ss[, G4.5ss:=FALSE ]
short_introns.5ss[ dist<=100 & i.strand=="+" & i.start<g.end , G4.5ss:=TRUE ]
short_introns.5ss[ dist<=100 & i.strand=="-" & i.start>g.end , G4.5ss:=TRUE ]

long_short.2x2.5ss <- matrix(nrow = 2, c(nrow(long_introns.5ss[G4.5ss==FALSE]), nrow(short_introns.5ss[G4.5ss==FALSE]), nrow(long_introns.5ss[G4.5ss==TRUE]), nrow(short_introns.5ss[G4.5ss==TRUE]) ))

chisq.test(long_short.2x2.5ss)


long_introns.3ss[, G4.3ss:=FALSE ]
long_introns.3ss[ dist<=100 & i.strand=="+" & i.start<g.end , G4.3ss:=TRUE ]
long_introns.3ss[ dist<=100 & i.strand=="-" & i.start>g.end , G4.3ss:=TRUE ]

short_introns.3ss[, G4.3ss:=FALSE ]
short_introns.3ss[ dist<=100 & i.strand=="+" & i.start<g.end , G4.3ss:=TRUE ]
short_introns.3ss[ dist<=100 & i.strand=="-" & i.start>g.end , G4.3ss:=TRUE ]

long_short.2x2.3ss <- matrix(nrow = 2, c(nrow(long_introns.3ss[G4.3ss==FALSE]), nrow(short_introns.3ss[G4.3ss==FALSE]), nrow(long_introns.3ss[G4.3ss==TRUE]), nrow(short_introns.3ss[G4.3ss==TRUE]) ))

chisq.test(long_short.2x2.3ss)





long_introns <- merge( long_introns.5ss, long_introns.3ss, by=c("i.id", "i.GC" ))
short_introns <- merge( short_introns.5ss, short_introns.3ss, by=c("i.id", "i.GC" ))





long_short.2x2 <- matrix(nrow = 2, c(nrow(long_introns[(!G4.3ss & !G4.5ss), ]),
                                         nrow(short_introns[(!G4.3ss & !G4.5ss), ]),
                                         nrow(long_introns[(G4.3ss | G4.5ss), ]),
                                         nrow(short_introns[(G4.3ss | G4.5ss), ]) ))


                                         
chisq.test(long_short.2x2)


## Only considering short introns longer than 50





long_introns <- long_introns[i.id %in% long_short_correspondence[s.len>50, l.id]]
short_introns <- short_introns[i.id %in% long_short_correspondence[s.len>50, s.id]]


long_short.2x2 <- matrix(nrow = 2, c(nrow(long_introns[(!G4.3ss & !G4.5ss), ]),
                                         nrow(short_introns[(!G4.3ss & !G4.5ss), ]),
                                         nrow(long_introns[(G4.3ss | G4.5ss), ]),
                                         nrow(short_introns[(G4.3ss | G4.5ss), ]) ))


                                         
chisq.test(long_short.2x2)



short_introns[, type:="Short"]
long_introns[, type:="Long"]

long_introns$type <- factor(long_introns$type, levels=c("Short", "Long"))



long_short_introns <- rbind(long_introns, short_introns)

ggplot(long_short_introns) +
  geom_violin(aes( type, i.GC))


ggplot(long_short_introns) +
  geom_bar(aes(x=type, fill=((G4.3ss | G4.5ss) ), stat="count" ))


                                         
```



```{r}
long_introns.5ss <- fread("./Intron_size/GC_correction/500/long_introns.same_GC.bed.intron_5ss.bed.closest.G4s")
short_introns.5ss <- fread("./Intron_size/GC_correction/500/short_introns.same_GC.bed.intron_5ss.bed.closest.G4s")
long_introns.3ss <- fread("./Intron_size/GC_correction/500/long_introns.same_GC.bed.intron_3ss.bed.closest.G4s")
short_introns.3ss <- fread("./Intron_size/GC_correction/500/short_introns.same_GC.bed.intron_3ss.bed.closest.G4s")


colnames(long_introns.5ss) <- c("i.chrom", "i.start", "i.end", "i.id", "i.GC", "i.strand", "g.chrom", "g.start", "g.end", "dist")
colnames(short_introns.5ss) <- c("i.chrom", "i.start", "i.end", "i.id", "i.GC", "i.strand", "g.chrom", "g.start", "g.end", "dist")
colnames(long_introns.3ss) <- c("i.chrom", "i.start", "i.end", "i.id", "i.GC", "i.strand", "g.chrom", "g.start", "g.end", "dist")
colnames(short_introns.3ss) <- c("i.chrom", "i.start", "i.end", "i.id", "i.GC", "i.strand", "g.chrom", "g.start", "g.end", "dist")

long_introns.bed <- fread("./Intron_size/GC_correction/500/long_introns.same_GC.bed")
short_introns.bed <- fread("./Intron_size/GC_correction/500/short_introns.same_GC.bed")
long_short_correspondence <-  cbind(long_introns.bed, short_introns.bed)[ , c(4, 2, 3 , 5,10, 8, 9, 11)]

colnames(long_short_correspondence) <- c("l.id", "l.start", "l.end", "l.GC", "s.id", "s.start", "s.end", "s.GC"  )
long_short_correspondence[, s.len:=(s.end-s.start)]

long_short_correspondence_35K <- head(long_short_correspondence, 17500)




long_introns.5ss[, G4.5ss:=FALSE ]
long_introns.5ss[ dist<=100 & i.strand=="+" & i.start<g.end , G4.5ss:=TRUE ]
long_introns.5ss[ dist<=100 & i.strand=="-" & i.start>g.end , G4.5ss:=TRUE ]

short_introns.5ss[, G4.5ss:=FALSE ]
short_introns.5ss[ dist<=100 & i.strand=="+" & i.start<g.end , G4.5ss:=TRUE ]
short_introns.5ss[ dist<=100 & i.strand=="-" & i.start>g.end , G4.5ss:=TRUE ]

long_introns.3ss[, G4.3ss:=FALSE ]
long_introns.3ss[ dist<=100 & i.strand=="+" & i.start<g.end , G4.3ss:=TRUE ]
long_introns.3ss[ dist<=100 & i.strand=="-" & i.start>g.end , G4.3ss:=TRUE ]

short_introns.3ss[, G4.3ss:=FALSE ]
short_introns.3ss[ dist<=100 & i.strand=="+" & i.start<g.end , G4.3ss:=TRUE ]
short_introns.3ss[ dist<=100 & i.strand=="-" & i.start>g.end , G4.3ss:=TRUE ]


long_introns <- merge( long_introns.5ss, long_introns.3ss, by=c("i.id", "i.GC" ))
short_introns <- merge( short_introns.5ss, short_introns.3ss, by=c("i.id", "i.GC" ))


long_introns[, G4:=FALSE]
long_introns[G4.3ss | G4.5ss , G4:=TRUE]
short_introns[, G4:=FALSE]
short_introns[G4.3ss | G4.5ss , G4:=TRUE]




i.mins <- seq(50,450,50)
i.maxs <- seq(100,500,50)



i.mins <- seq(50,450,50)
i.maxs <- seq(100,500,50)


i.mins <- seq(0,400,100)
i.maxs <- seq(100,500,100)

short_long.odd_ratios <- c()
short_long.pvalues<- c()
short_long.long.fraq <- c()
short_long.short.fraq <- c()
  
short_long.GCs <- data.table()
  
for (i in seq(1, length(i.mins))) {
    
  i.min = i.mins[i]
  i.max = i.maxs[i]
    
  long_introns.int <- long_introns[i.id %in% long_short_correspondence_35K[s.len>i.min & s.len<=i.max ,  l.id]]
  short_introns.int <- short_introns[i.id %in% long_short_correspondence_35K[s.len>i.min & s.len<=i.max, s.id]]
  
  
  long_short.int.2x2 <- matrix(nrow = 2, c(nrow(long_introns.int[(!G4.3ss & !G4.5ss), ]),
                                           nrow(short_introns.int[(!G4.3ss & !G4.5ss), ]),
                                           nrow(long_introns.int[(G4.3ss | G4.5ss), ]),
                                           nrow(short_introns.int[(G4.3ss | G4.5ss), ]) ))
  
  
  odd.ratio <- (long_short.int.2x2[2,2] / long_short.int.2x2[2,1] ) / (long_short.int.2x2[1,2] / long_short.int.2x2[1,1])
  long.frac <-  (long_short.int.2x2[1,2] / long_short.int.2x2[1,1])
  short.frac <- (long_short.int.2x2[2,2] / long_short.int.2x2[2,1] )
  
  print(odd.ratio)
  print(chisq.test(long_short.int.2x2))
  
  short_long.odd_ratios <- c(short_long.odd_ratios, odd.ratio)
  short_long.pvalues <- c(short_long.pvalues, chisq.test(long_short.int.2x2)$p.value * length(i.mins))
  short_long.short.fraq  <- c(short_long.short.fraq, short.frac)
  short_long.long.fraq  <- c(short_long.long.fraq, long.frac)
  
  
  short_long.GCs <- rbind( short_long.GCs, short_introns.int[ , .(i.id, i.GC, G4, type="Short",  i.min=i.min, i.max=i.max )] )
  short_long.GCs <- rbind( short_long.GCs, long_introns.int[ , .(i.id, i.GC, G4, type="Long",  i.min=i.min, i.max=i.max )] )  
  }




short_long.res <- data.table(short_long.odd_ratios, short_long.pvalues, i.mins, i.maxs, short_long.short.fraq, short_long.long.fraq)

short_long.res.melt <- melt(short_long.res, id=c("short_long.odd_ratios","short_long.pvalues", "i.mins", "i.maxs"))



                                    
```



```{r, fig.width=15, fig.height=5}



short_long.GCs$`Intron type` <- factor(short_long.GCs$type, levels=c("Short", "Long"))

Fig3.sup.B <- ggplot(short_long.GCs) +
  geom_violin(aes(as.factor(i.min), i.GC, fill=`Intron type`)) +
  xlab("Intron size interval (nt)") +
  ylab("GC content percentage after correction") +
  scale_fill_brewer(palette = 'Blues') 

```





```{r}



short_long.res.melt[, sig:=""]
short_long.res.melt[short_long.pvalues <= 0.05 , sig:="*"]
short_long.res.melt[short_long.pvalues <= 0.005 , sig:="**"]
short_long.res.melt[short_long.pvalues <= 0.000 , sig:="***"]

sig_rows <- short_long.res.melt[variable=="short_long.short.fraq" & sig!="", which = TRUE]
sig_anno <- short_long.res.melt[variable=="short_long.short.fraq" & sig!="", sig]

Fig3.C.new <-  ggplot(short_long.res.melt, aes(as.factor(i.mins), value)) +
  geom_bar(aes( group=variable, fill=variable), position="dodge", stat="identity") +
  geom_signif( y_position=0.3, xmin=sig_rows-0.15, xmax=sig_rows+0.15, annotation=sig_anno, tip_length=0) +
  scale_fill_discrete(name = "Intron type", labels = c("Short", "Long")) +
  scale_fill_brewer(palette = 'Blues') +
  theme(legend.position = "top", legend.direction = "horizontal") 


Fig3.C.new

```








## Exon number


Here we explored how G4 enrichments change thorugh gene body


```{r, message=FALSE, error=FALSE, warning=FALSE}


exons_1.up_plus <- read_dist_table("./exon_numbers/exons_1_plus_strand_upstream.txt.score.G4.bed.list.out.num")
exons_1.down_plus <- read_dist_table("./exon_numbers/exons_1_plus_strand_downstream.txt.score.G4.bed.list.out.num")
exons_1.up_minus <- read_dist_table("./exon_numbers/exons_1_minus_strand_upstream.txt.score.G4.bed.list.out.num")
exons_1.down_minus <- read_dist_table("./exon_numbers/exons_1_minus_strand_downstream.txt.score.G4.bed.list.out.num")
exons_1.up_minus[,Position:=Position*-1]
exons_1.down_minus[,Position:=Position*-1]

exons_1.TOTAL <- plot_density(exons_1.up_plus, exons_1.up_minus, exons_1.down_plus, exons_1.down_minus) 

```



```{r, message=FALSE, error=FALSE, warning=FALSE}


exons_2.up_plus <- read_dist_table("./exon_numbers/exons_2_plus_strand_upstream.txt.score.G4.bed.list.out.num")
exons_2.down_plus <- read_dist_table("./exon_numbers/exons_2_plus_strand_downstream.txt.score.G4.bed.list.out.num")
exons_2.up_minus <- read_dist_table("./exon_numbers/exons_2_minus_strand_upstream.txt.score.G4.bed.list.out.num")
exons_2.down_minus <- read_dist_table("./exon_numbers/exons_2_minus_strand_downstream.txt.score.G4.bed.list.out.num")
exons_2.up_minus[,Position:=Position*-1]
exons_2.down_minus[,Position:=Position*-1]

exons_2.TOTAL <- plot_density(exons_2.up_plus, exons_2.up_minus, exons_2.down_plus, exons_2.down_minus) 

```




```{r, message=FALSE, error=FALSE, warning=FALSE}



exons_3.up_plus <- read_dist_table("./exon_numbers/exons_3_plus_strand_upstream.txt.score.G4.bed.list.out.num")
exons_3.down_plus <- read_dist_table("./exon_numbers/exons_3_plus_strand_downstream.txt.score.G4.bed.list.out.num")
exons_3.up_minus <- read_dist_table("./exon_numbers/exons_3_minus_strand_upstream.txt.score.G4.bed.list.out.num")
exons_3.down_minus <- read_dist_table("./exon_numbers/exons_3_minus_strand_downstream.txt.score.G4.bed.list.out.num")
exons_3.up_minus[,Position:=Position*-1]
exons_3.down_minus[,Position:=Position*-1]

exons_3.TOTAL <- plot_density(exons_3.up_plus, exons_3.up_minus, exons_3.down_plus, exons_3.down_minus) 

```





```{r, message=FALSE, error=FALSE, warning=FALSE}



exons_4.up_plus <- read_dist_table("./exon_numbers/exons_4_plus_strand_upstream.txt.score.G4.bed.list.out.num")
exons_4.down_plus <- read_dist_table("./exon_numbers/exons_4_plus_strand_downstream.txt.score.G4.bed.list.out.num")
exons_4.up_minus <- read_dist_table("./exon_numbers/exons_4_minus_strand_upstream.txt.score.G4.bed.list.out.num")
exons_4.down_minus <- read_dist_table("./exon_numbers/exons_4_minus_strand_downstream.txt.score.G4.bed.list.out.num")
exons_4.up_minus[,Position:=Position*-1]
exons_4.down_minus[,Position:=Position*-1]

exons_4.TOTAL <- plot_density(exons_4.up_plus, exons_4.up_minus, exons_4.down_plus, exons_4.down_minus) 

```






```{r, message=FALSE, error=FALSE, warning=FALSE}



exons_middle.up_plus <- read_dist_table("./exon_numbers/exons_middle_plus_strand_upstream.txt.score.G4.bed.list.out.num")
exons_middle.down_plus <- read_dist_table("./exon_numbers/exons_middle_plus_strand_downstream.txt.score.G4.bed.list.out.num")
exons_middle.up_minus <- read_dist_table("./exon_numbers/exons_middle_minus_strand_upstream.txt.score.G4.bed.list.out.num")
exons_middle.down_minus <- read_dist_table("./exon_numbers/exons_middle_minus_strand_downstream.txt.score.G4.bed.list.out.num")
exons_middle.up_minus[,Position:=Position*-1]
exons_middle.down_minus[,Position:=Position*-1]

exons_middle.TOTAL <- plot_density(exons_middle.up_plus, exons_middle.up_minus, exons_middle.down_plus, exons_middle.down_minus) 

```





```{r, message=FALSE, error=FALSE, warning=FALSE}



exons_minus4.up_plus <- read_dist_table("./exon_numbers/exons_minus4_plus_strand_upstream.txt.score.G4.bed.list.out.num")
exons_minus4.down_plus <- read_dist_table("./exon_numbers/exons_minus4_plus_strand_downstream.txt.score.G4.bed.list.out.num")
exons_minus4.up_minus <- read_dist_table("./exon_numbers/exons_minus4_minus_strand_upstream.txt.score.G4.bed.list.out.num")
exons_minus4.down_minus <- read_dist_table("./exon_numbers/exons_minus4_minus_strand_downstream.txt.score.G4.bed.list.out.num")
exons_minus4.up_minus[,Position:=Position*-1]
exons_minus4.down_minus[,Position:=Position*-1]

exons_minus4.TOTAL <- plot_density(exons_minus4.up_plus, exons_minus4.up_minus, exons_minus4.down_plus, exons_minus4.down_minus) 

```




```{r, message=FALSE, error=FALSE, warning=FALSE}



exons_minus3.up_plus <- read_dist_table("./exon_numbers/exons_minus3_plus_strand_upstream.txt.score.G4.bed.list.out.num")
exons_minus3.down_plus <- read_dist_table("./exon_numbers/exons_minus3_plus_strand_downstream.txt.score.G4.bed.list.out.num")
exons_minus3.up_minus <- read_dist_table("./exon_numbers/exons_minus3_minus_strand_upstream.txt.score.G4.bed.list.out.num")
exons_minus3.down_minus <- read_dist_table("./exon_numbers/exons_minus3_minus_strand_downstream.txt.score.G4.bed.list.out.num")
exons_minus3.up_minus[,Position:=Position*-1]
exons_minus3.down_minus[,Position:=Position*-1]

exons_minus3.TOTAL <- plot_density(exons_minus3.up_plus, exons_minus3.up_minus, exons_minus3.down_plus, exons_minus3.down_minus) 

```





```{r, message=FALSE, error=FALSE, warning=FALSE}



exons_minus2.up_plus <- read_dist_table("./exon_numbers/exons_minus2_plus_strand_upstream.txt.score.G4.bed.list.out.num")
exons_minus2.down_plus <- read_dist_table("./exon_numbers/exons_minus2_plus_strand_downstream.txt.score.G4.bed.list.out.num")
exons_minus2.up_minus <- read_dist_table("./exon_numbers/exons_minus2_minus_strand_upstream.txt.score.G4.bed.list.out.num")
exons_minus2.down_minus <- read_dist_table("./exon_numbers/exons_minus2_minus_strand_downstream.txt.score.G4.bed.list.out.num")
exons_minus2.up_minus[,Position:=Position*-1]
exons_minus2.down_minus[,Position:=Position*-1]

exons_minus2.TOTAL <- plot_density(exons_minus2.up_plus, exons_minus2.up_minus, exons_minus2.down_plus, exons_minus2.down_minus) 

```




```{r, message=FALSE, error=FALSE, warning=FALSE}



exons_minus1.up_plus <- read_dist_table("./exon_numbers/exons_minus1_plus_strand_upstream.txt.score.G4.bed.list.out.num")
exons_minus1.down_plus <- read_dist_table("./exon_numbers/exons_minus1_plus_strand_downstream.txt.score.G4.bed.list.out.num")
exons_minus1.up_minus <- read_dist_table("./exon_numbers/exons_minus1_minus_strand_upstream.txt.score.G4.bed.list.out.num")
exons_minus1.down_minus <- read_dist_table("./exon_numbers/exons_minus1_minus_strand_downstream.txt.score.G4.bed.list.out.num")
exons_minus1.up_minus[,Position:=Position*-1]
exons_minus1.down_minus[,Position:=Position*-1]

exons_minus1.TOTAL <- plot_density(exons_minus1.up_plus, exons_minus1.up_minus, exons_minus1.down_plus, exons_minus1.down_minus) 

```





```{r, fig.height=10, fig.width=15}
exons_1.TOTAL[, exon_num:="1"]
exons_2.TOTAL[, exon_num:="2"]
exons_3.TOTAL[, exon_num:="3"]
exons_4.TOTAL[, exon_num:="4"]

exons_middle.TOTAL[, exon_num:="middle"]

exons_minus4.TOTAL[, exon_num:="-4"]
exons_minus3.TOTAL[, exon_num:="-3"]
exons_minus2.TOTAL[, exon_num:="-2"]
exons_minus1.TOTAL[, exon_num:="-1"]


exon_num.TOTAL <- rbind(exons_1.TOTAL, exons_2.TOTAL, exons_3.TOTAL, exons_4.TOTAL, exons_middle.TOTAL, exons_minus4.TOTAL, exons_minus3.TOTAL, exons_minus2.TOTAL, exons_minus1.TOTAL)


exon_num.TOTAL$exon_num <- factor(exon_num.TOTAL$exon_num, levels = c("1", "2", "3", "4", "middle", "-4", "-3", "-2", "-1" ))

Fig3.sup.C <- ggplot(exon_num.TOTAL)+
    geom_line(aes(x=Position,y=Enrrichment)) +
    xlim(c(-300,300)) +
    facet_grid( exon_pos ~ exon_num   ) +
  theme_bw()

Fig3.sup.C

```





## Exon numbers by stratnds ##



```{r, message=FALSE, error=FALSE, warning=FALSE}


exons_1.up_plus.nt <- read_dist_table("./exon_numbers/strands/exons_1_plus_strand_upstream.txt.score.G4_plus.bed.list.out.num")
exons_1.down_plus.nt <- read_dist_table("./exon_numbers/strands/exons_1_plus_strand_downstream.txt.score.G4_plus.bed.list.out.num")
exons_1.up_minus.nt <- read_dist_table("./exon_numbers/strands/exons_1_minus_strand_upstream.txt.score.G4_minus.bed.list.out.num")
exons_1.down_minus.nt <- read_dist_table("./exon_numbers/strands/exons_1_minus_strand_downstream.txt.score.G4_minus.bed.list.out.num")
exons_1.up_minus.nt[,Position:=Position*-1]
exons_1.down_minus.nt[,Position:=Position*-1]

exons_1.TOTAL.nt <- plot_density(exons_1.up_plus.nt, exons_1.up_minus.nt, exons_1.down_plus.nt, exons_1.down_minus.nt) 

```





```{r, message=FALSE, error=FALSE, warning=FALSE}


exons_1.up_plus.t <- read_dist_table("./exon_numbers/strands/exons_1_plus_strand_upstream.txt.score.G4_minus.bed.list.out.num")
exons_1.down_plus.t <- read_dist_table("./exon_numbers/strands/exons_1_plus_strand_downstream.txt.score.G4_minus.bed.list.out.num")
exons_1.up_minus.t <- read_dist_table("./exon_numbers/strands/exons_1_minus_strand_upstream.txt.score.G4_plus.bed.list.out.num")
exons_1.down_minus.t <- read_dist_table("./exon_numbers/strands/exons_1_minus_strand_downstream.txt.score.G4_plus.bed.list.out.num")
exons_1.up_minus.t[,Position:=Position*-1]
exons_1.down_minus.t[,Position:=Position*-1]

exons_1.TOTAL.t <- plot_density(exons_1.up_plus.t, exons_1.up_minus.t, exons_1.down_plus.t, exons_1.down_minus.t) 


 ggplot(exons_1.TOTAL.t)+
    geom_line(aes(x=Position,y=Enrrichment)) +
    facet_grid( . ~ exon_pos ) +
    ylim(c(0,6)) +
    theme_bw()

```





```{r, message=FALSE, error=FALSE, warning=FALSE}


exons_2.up_plus.nt <- read_dist_table("./exon_numbers/strands/exons_2_plus_strand_upstream.txt.score.G4_plus.bed.list.out.num")
exons_2.down_plus.nt <- read_dist_table("./exon_numbers/strands/exons_2_plus_strand_downstream.txt.score.G4_plus.bed.list.out.num")
exons_2.up_minus.nt <- read_dist_table("./exon_numbers/strands/exons_2_minus_strand_upstream.txt.score.G4_minus.bed.list.out.num")
exons_2.down_minus.nt <- read_dist_table("./exon_numbers/strands/exons_2_minus_strand_downstream.txt.score.G4_minus.bed.list.out.num")
exons_2.up_minus.nt[,Position:=Position*-1]
exons_2.down_minus.nt[,Position:=Position*-1]

exons_2.TOTAL.nt <- plot_density(exons_2.up_plus.nt, exons_2.up_minus.nt, exons_2.down_plus.nt, exons_2.down_minus.nt) 

```





```{r, message=FALSE, error=FALSE, warning=FALSE}


exons_2.up_plus.t <- read_dist_table("./exon_numbers/strands/exons_2_plus_strand_upstream.txt.score.G4_minus.bed.list.out.num")
exons_2.down_plus.t <- read_dist_table("./exon_numbers/strands/exons_2_plus_strand_downstream.txt.score.G4_minus.bed.list.out.num")
exons_2.up_minus.t <- read_dist_table("./exon_numbers/strands/exons_2_minus_strand_upstream.txt.score.G4_plus.bed.list.out.num")
exons_2.down_minus.t <- read_dist_table("./exon_numbers/strands/exons_2_minus_strand_downstream.txt.score.G4_plus.bed.list.out.num")
exons_2.up_minus.t[,Position:=Position*-1]
exons_2.down_minus.t[,Position:=Position*-1]

exons_2.TOTAL.t <- plot_density(exons_2.up_plus.t, exons_2.up_minus.t, exons_2.down_plus.t, exons_2.down_minus.t) 


 ggplot(exons_2.TOTAL.t)+
    geom_line(aes(x=Position,y=Enrrichment)) +
    facet_grid( . ~ exon_pos ) +
    ylim(c(0,6)) +
    theme_bw()

```






```{r, message=FALSE, error=FALSE, warning=FALSE}


exons_3.up_plus.nt <- read_dist_table("./exon_numbers/strands/exons_3_plus_strand_upstream.txt.score.G4_plus.bed.list.out.num")
exons_3.down_plus.nt <- read_dist_table("./exon_numbers/strands/exons_3_plus_strand_downstream.txt.score.G4_plus.bed.list.out.num")
exons_3.up_minus.nt <- read_dist_table("./exon_numbers/strands/exons_3_minus_strand_upstream.txt.score.G4_minus.bed.list.out.num")
exons_3.down_minus.nt <- read_dist_table("./exon_numbers/strands/exons_3_minus_strand_downstream.txt.score.G4_minus.bed.list.out.num")
exons_3.up_minus.nt[,Position:=Position*-1]
exons_3.down_minus.nt[,Position:=Position*-1]

exons_3.TOTAL.nt <- plot_density(exons_3.up_plus.nt, exons_3.up_minus.nt, exons_3.down_plus.nt, exons_3.down_minus.nt) 

```





```{r, message=FALSE, error=FALSE, warning=FALSE}


exons_3.up_plus.t <- read_dist_table("./exon_numbers/strands/exons_3_plus_strand_upstream.txt.score.G4_minus.bed.list.out.num")
exons_3.down_plus.t <- read_dist_table("./exon_numbers/strands/exons_3_plus_strand_downstream.txt.score.G4_minus.bed.list.out.num")
exons_3.up_minus.t <- read_dist_table("./exon_numbers/strands/exons_3_minus_strand_upstream.txt.score.G4_plus.bed.list.out.num")
exons_3.down_minus.t <- read_dist_table("./exon_numbers/strands/exons_3_minus_strand_downstream.txt.score.G4_plus.bed.list.out.num")
exons_3.up_minus.t[,Position:=Position*-1]
exons_3.down_minus.t[,Position:=Position*-1]

exons_3.TOTAL.t <- plot_density(exons_3.up_plus.t, exons_3.up_minus.t, exons_3.down_plus.t, exons_3.down_minus.t) 


 ggplot(exons_3.TOTAL.t)+
    geom_line(aes(x=Position,y=Enrrichment)) +
    facet_grid( . ~ exon_pos ) +
    ylim(c(0,6)) +
    theme_bw()

```





```{r, message=FALSE, error=FALSE, warning=FALSE}


exons_4.up_plus.nt <- read_dist_table("./exon_numbers/strands/exons_4_plus_strand_upstream.txt.score.G4_plus.bed.list.out.num")
exons_4.down_plus.nt <- read_dist_table("./exon_numbers/strands/exons_4_plus_strand_downstream.txt.score.G4_plus.bed.list.out.num")
exons_4.up_minus.nt <- read_dist_table("./exon_numbers/strands/exons_4_minus_strand_upstream.txt.score.G4_minus.bed.list.out.num")
exons_4.down_minus.nt <- read_dist_table("./exon_numbers/strands/exons_4_minus_strand_downstream.txt.score.G4_minus.bed.list.out.num")
exons_4.up_minus.nt[,Position:=Position*-1]
exons_4.down_minus.nt[,Position:=Position*-1]

exons_4.TOTAL.nt <- plot_density(exons_4.up_plus.nt, exons_4.up_minus.nt, exons_4.down_plus.nt, exons_4.down_minus.nt) 

```



```{r, message=FALSE, error=FALSE, warning=FALSE}


exons_4.up_plus.t <- read_dist_table("./exon_numbers/strands/exons_4_plus_strand_upstream.txt.score.G4_minus.bed.list.out.num")
exons_4.down_plus.t <- read_dist_table("./exon_numbers/strands/exons_4_plus_strand_downstream.txt.score.G4_minus.bed.list.out.num")
exons_4.up_minus.t <- read_dist_table("./exon_numbers/strands/exons_4_minus_strand_upstream.txt.score.G4_plus.bed.list.out.num")
exons_4.down_minus.t <- read_dist_table("./exon_numbers/strands/exons_4_minus_strand_downstream.txt.score.G4_plus.bed.list.out.num")
exons_4.up_minus.t[,Position:=Position*-1]
exons_4.down_minus.t[,Position:=Position*-1]

exons_4.TOTAL.t <- plot_density(exons_4.up_plus.t, exons_4.up_minus.t, exons_4.down_plus.t, exons_4.down_minus.t) 


 ggplot(exons_4.TOTAL.t)+
    geom_line(aes(x=Position,y=Enrrichment)) +
    facet_grid( . ~ exon_pos ) +
    ylim(c(0,6)) +
    theme_bw()

```







```{r, message=FALSE, error=FALSE, warning=FALSE}


exons_middle.up_plus.nt <- read_dist_table("./exon_numbers/strands/exons_middle_plus_strand_upstream.txt.score.G4_plus.bed.list.out.num")
exons_middle.down_plus.nt <- read_dist_table("./exon_numbers/strands/exons_middle_plus_strand_downstream.txt.score.G4_plus.bed.list.out.num")
exons_middle.up_minus.nt <- read_dist_table("./exon_numbers/strands/exons_middle_minus_strand_upstream.txt.score.G4_minus.bed.list.out.num")
exons_middle.down_minus.nt <- read_dist_table("./exon_numbers/strands/exons_middle_minus_strand_downstream.txt.score.G4_minus.bed.list.out.num")
exons_middle.up_minus.nt[,Position:=Position*-1]
exons_middle.down_minus.nt[,Position:=Position*-1]

exons_middle.TOTAL.nt <- plot_density(exons_middle.up_plus.nt, exons_middle.up_minus.nt, exons_middle.down_plus.nt, exons_middle.down_minus.nt) 

```





```{r, message=FALSE, error=FALSE, warning=FALSE}


exons_middle.up_plus.t <- read_dist_table("./exon_numbers/strands/exons_middle_plus_strand_upstream.txt.score.G4_minus.bed.list.out.num")
exons_middle.down_plus.t <- read_dist_table("./exon_numbers/strands/exons_middle_plus_strand_downstream.txt.score.G4_minus.bed.list.out.num")
exons_middle.up_minus.t <- read_dist_table("./exon_numbers/strands/exons_middle_minus_strand_upstream.txt.score.G4_plus.bed.list.out.num")
exons_middle.down_minus.t <- read_dist_table("./exon_numbers/strands/exons_middle_minus_strand_downstream.txt.score.G4_plus.bed.list.out.num")
exons_middle.up_minus.t[,Position:=Position*-1]
exons_middle.down_minus.t[,Position:=Position*-1]

exons_middle.TOTAL.t <- plot_density(exons_middle.up_plus.t, exons_middle.up_minus.t, exons_middle.down_plus.t, exons_middle.down_minus.t) 


 ggplot(exons_middle.TOTAL.t)+
    geom_line(aes(x=Position,y=Enrrichment)) +
    facet_grid( . ~ exon_pos ) +
    ylim(c(0,6)) +
    theme_bw()

```





```{r, message=FALSE, error=FALSE, warning=FALSE}


exons_minus4.up_plus.nt <- read_dist_table("./exon_numbers/strands/exons_minus4_plus_strand_upstream.txt.score.G4_plus.bed.list.out.num")
exons_minus4.down_plus.nt <- read_dist_table("./exon_numbers/strands/exons_minus4_plus_strand_downstream.txt.score.G4_plus.bed.list.out.num")
exons_minus4.up_minus.nt <- read_dist_table("./exon_numbers/strands/exons_minus4_minus_strand_upstream.txt.score.G4_minus.bed.list.out.num")
exons_minus4.down_minus.nt <- read_dist_table("./exon_numbers/strands/exons_minus4_minus_strand_downstream.txt.score.G4_minus.bed.list.out.num")
exons_minus4.up_minus.nt[,Position:=Position*-1]
exons_minus4.down_minus.nt[,Position:=Position*-1]

exons_minus4.TOTAL.nt <- plot_density(exons_minus4.up_plus.nt, exons_minus4.up_minus.nt, exons_minus4.down_plus.nt, exons_minus4.down_minus.nt) 

```



```{r, message=FALSE, error=FALSE, warning=FALSE}


exons_minus4.up_plus.t <- read_dist_table("./exon_numbers/strands/exons_minus4_plus_strand_upstream.txt.score.G4_minus.bed.list.out.num")
exons_minus4.down_plus.t <- read_dist_table("./exon_numbers/strands/exons_minus4_plus_strand_downstream.txt.score.G4_minus.bed.list.out.num")
exons_minus4.up_minus.t <- read_dist_table("./exon_numbers/strands/exons_minus4_minus_strand_upstream.txt.score.G4_plus.bed.list.out.num")
exons_minus4.down_minus.t <- read_dist_table("./exon_numbers/strands/exons_minus4_minus_strand_downstream.txt.score.G4_plus.bed.list.out.num")
exons_minus4.up_minus.t[,Position:=Position*-1]
exons_minus4.down_minus.t[,Position:=Position*-1]

exons_minus4.TOTAL.t <- plot_density(exons_minus4.up_plus.t, exons_minus4.up_minus.t, exons_minus4.down_plus.t, exons_minus4.down_minus.t) 


 ggplot(exons_minus4.TOTAL.t)+
    geom_line(aes(x=Position,y=Enrrichment)) +
    facet_grid( . ~ exon_pos ) +
    ylim(c(0,6)) +
    theme_bw()

```





```{r, message=FALSE, error=FALSE, warning=FALSE}


exons_minus3.up_plus.nt <- read_dist_table("./exon_numbers/strands/exons_minus3_plus_strand_upstream.txt.score.G4_plus.bed.list.out.num")
exons_minus3.down_plus.nt <- read_dist_table("./exon_numbers/strands/exons_minus3_plus_strand_downstream.txt.score.G4_plus.bed.list.out.num")
exons_minus3.up_minus.nt <- read_dist_table("./exon_numbers/strands/exons_minus3_minus_strand_upstream.txt.score.G4_minus.bed.list.out.num")
exons_minus3.down_minus.nt <- read_dist_table("./exon_numbers/strands/exons_minus3_minus_strand_downstream.txt.score.G4_minus.bed.list.out.num")
exons_minus3.up_minus.nt[,Position:=Position*-1]
exons_minus3.down_minus.nt[,Position:=Position*-1]

exons_minus3.TOTAL.nt <- plot_density(exons_minus3.up_plus.nt, exons_minus3.up_minus.nt, exons_minus3.down_plus.nt, exons_minus3.down_minus.nt) 

```



```{r, message=FALSE, error=FALSE, warning=FALSE}


exons_minus3.up_plus.t <- read_dist_table("./exon_numbers/strands/exons_minus3_plus_strand_upstream.txt.score.G4_minus.bed.list.out.num")
exons_minus3.down_plus.t <- read_dist_table("./exon_numbers/strands/exons_minus3_plus_strand_downstream.txt.score.G4_minus.bed.list.out.num")
exons_minus3.up_minus.t <- read_dist_table("./exon_numbers/strands/exons_minus3_minus_strand_upstream.txt.score.G4_plus.bed.list.out.num")
exons_minus3.down_minus.t <- read_dist_table("./exon_numbers/strands/exons_minus3_minus_strand_downstream.txt.score.G4_plus.bed.list.out.num")
exons_minus3.up_minus.t[,Position:=Position*-1]
exons_minus3.down_minus.t[,Position:=Position*-1]

exons_minus3.TOTAL.t <- plot_density(exons_minus3.up_plus.t, exons_minus3.up_minus.t, exons_minus3.down_plus.t, exons_minus3.down_minus.t) 


 ggplot(exons_minus3.TOTAL.t)+
    geom_line(aes(x=Position,y=Enrrichment)) +
    facet_grid( . ~ exon_pos ) +
    ylim(c(0,6)) +
    theme_bw()

```





```{r, message=FALSE, error=FALSE, warning=FALSE}


exons_minus2.up_plus.nt <- read_dist_table("./exon_numbers/strands/exons_minus2_plus_strand_upstream.txt.score.G4_plus.bed.list.out.num")
exons_minus2.down_plus.nt <- read_dist_table("./exon_numbers/strands/exons_minus2_plus_strand_downstream.txt.score.G4_plus.bed.list.out.num")
exons_minus2.up_minus.nt <- read_dist_table("./exon_numbers/strands/exons_minus2_minus_strand_upstream.txt.score.G4_minus.bed.list.out.num")
exons_minus2.down_minus.nt <- read_dist_table("./exon_numbers/strands/exons_minus2_minus_strand_downstream.txt.score.G4_minus.bed.list.out.num")
exons_minus2.up_minus.nt[,Position:=Position*-1]
exons_minus2.down_minus.nt[,Position:=Position*-1]

exons_minus2.TOTAL.nt <- plot_density(exons_minus2.up_plus.nt, exons_minus2.up_minus.nt, exons_minus2.down_plus.nt, exons_minus2.down_minus.nt) 

```



```{r, message=FALSE, error=FALSE, warning=FALSE}


exons_minus2.up_plus.t <- read_dist_table("./exon_numbers/strands/exons_minus2_plus_strand_upstream.txt.score.G4_minus.bed.list.out.num")
exons_minus2.down_plus.t <- read_dist_table("./exon_numbers/strands/exons_minus2_plus_strand_downstream.txt.score.G4_minus.bed.list.out.num")
exons_minus2.up_minus.t <- read_dist_table("./exon_numbers/strands/exons_minus2_minus_strand_upstream.txt.score.G4_plus.bed.list.out.num")
exons_minus2.down_minus.t <- read_dist_table("./exon_numbers/strands/exons_minus2_minus_strand_downstream.txt.score.G4_plus.bed.list.out.num")
exons_minus2.up_minus.t[,Position:=Position*-1]
exons_minus2.down_minus.t[,Position:=Position*-1]

exons_minus2.TOTAL.t <- plot_density(exons_minus2.up_plus.t, exons_minus2.up_minus.t, exons_minus2.down_plus.t, exons_minus2.down_minus.t) 


 ggplot(exons_minus2.TOTAL.t)+
    geom_line(aes(x=Position,y=Enrrichment)) +
    facet_grid( . ~ exon_pos ) +
    ylim(c(0,6)) +
    theme_bw()

```






```{r, message=FALSE, error=FALSE, warning=FALSE}


exons_minus1.up_plus.nt <- read_dist_table("./exon_numbers/strands/exons_minus1_plus_strand_upstream.txt.score.G4_plus.bed.list.out.num")
exons_minus1.down_plus.nt <- read_dist_table("./exon_numbers/strands/exons_minus1_plus_strand_downstream.txt.score.G4_plus.bed.list.out.num")
exons_minus1.up_minus.nt <- read_dist_table("./exon_numbers/strands/exons_minus1_minus_strand_upstream.txt.score.G4_minus.bed.list.out.num")
exons_minus1.down_minus.nt <- read_dist_table("./exon_numbers/strands/exons_minus1_minus_strand_downstream.txt.score.G4_minus.bed.list.out.num")
exons_minus1.up_minus.nt[,Position:=Position*-1]
exons_minus1.down_minus.nt[,Position:=Position*-1]

exons_minus1.TOTAL.nt <- plot_density(exons_minus1.up_plus.nt, exons_minus1.up_minus.nt, exons_minus1.down_plus.nt, exons_minus1.down_minus.nt) 

```



```{r, message=FALSE, error=FALSE, warning=FALSE}


exons_minus1.up_plus.t <- read_dist_table("./exon_numbers/strands/exons_minus1_plus_strand_upstream.txt.score.G4_minus.bed.list.out.num")
exons_minus1.down_plus.t <- read_dist_table("./exon_numbers/strands/exons_minus1_plus_strand_downstream.txt.score.G4_minus.bed.list.out.num")
exons_minus1.up_minus.t <- read_dist_table("./exon_numbers/strands/exons_minus1_minus_strand_upstream.txt.score.G4_plus.bed.list.out.num")
exons_minus1.down_minus.t <- read_dist_table("./exon_numbers/strands/exons_minus1_minus_strand_downstream.txt.score.G4_plus.bed.list.out.num")
exons_minus1.up_minus.t[,Position:=Position*-1]
exons_minus1.down_minus.t[,Position:=Position*-1]

exons_minus1.TOTAL.t <- plot_density(exons_minus1.up_plus.t, exons_minus1.up_minus.t, exons_minus1.down_plus.t, exons_minus1.down_minus.t) 


 ggplot(exons_minus1.TOTAL.t)+
    geom_line(aes(x=Position,y=Enrrichment)) +
    facet_grid( . ~ exon_pos ) +
    ylim(c(0,6)) +
    theme_bw()

```




```{r, fig.height=10, fig.width=15}
exons_1.TOTAL.nt[, exon_num:="1"]
exons_2.TOTAL.nt[, exon_num:="2"]
exons_3.TOTAL.nt[, exon_num:="3"]
exons_4.TOTAL.nt[, exon_num:="4"]

exons_middle.TOTAL.nt[, exon_num:="middle"]

exons_minus4.TOTAL.nt[, exon_num:="-4"]
exons_minus3.TOTAL.nt[, exon_num:="-3"]
exons_minus2.TOTAL.nt[, exon_num:="-2"]
exons_minus1.TOTAL.nt[, exon_num:="-1"]


exon_num.TOTAL.nt <- rbind(exons_1.TOTAL.nt, exons_2.TOTAL.nt, exons_3.TOTAL.nt, exons_4.TOTAL.nt, exons_middle.TOTAL.nt, exons_minus4.TOTAL.nt, exons_minus3.TOTAL.nt, exons_minus2.TOTAL.nt, exons_minus1.TOTAL.nt)




exons_1.TOTAL.t[, exon_num:="1"]
exons_2.TOTAL.t[, exon_num:="2"]
exons_3.TOTAL.t[, exon_num:="3"]
exons_4.TOTAL.t[, exon_num:="4"]

exons_middle.TOTAL.t[, exon_num:="middle"]

exons_minus4.TOTAL.t[, exon_num:="-4"]
exons_minus3.TOTAL.t[, exon_num:="-3"]
exons_minus2.TOTAL.t[, exon_num:="-2"]
exons_minus1.TOTAL.t[, exon_num:="-1"]





exon_num.TOTAL.t <- rbind(exons_1.TOTAL.t, exons_2.TOTAL.t, exons_3.TOTAL.t, exons_4.TOTAL.t, exons_middle.TOTAL.t, exons_minus4.TOTAL.t, exons_minus3.TOTAL.t, exons_minus2.TOTAL.t, exons_minus1.TOTAL.t)


exon_num.TOTAL.nt[, strand:="Non_template"]
exon_num.TOTAL.t[, strand:="Template"]


exon_num.TOTAL.tnt <- rbind(exon_num.TOTAL.nt, exon_num.TOTAL.t)
exon_num.TOTAL.tnt$exon_num <- factor(exon_num.TOTAL.tnt$exon_num, levels = c("1", "2", "3", "4", "middle", "-4", "-3", "-2", "-1" ))



exon_num.TOTAL.tnt[exon_num=="middle", Total_exons:=(43925 + 43729) ]
exon_num.TOTAL.tnt[exon_num!="middle", Total_exons:=(4745 + 4569) ]


exon_num.TOTAL.tnt.binomial <- cbind(exon_num.TOTAL.tnt,  exon_num.TOTAL.tnt[, binconf(Occurrences, Total_exons) ])



exon_num.TOTAL.tnt.binomial[, median:=NULL]

exon_num.TOTAL.tnt.binomial[ , median:=median(PointEst), by=c("exon_pos", "strand", "exon_num" )]
exon_num.TOTAL.tnt.binomial[, `:=`(Enrrichment=PointEst/median, Enrrichment_l=Lower/median, Enrrichment_u=Upper/median)]


Fig3.D.new <-ggplot(exon_num.TOTAL.tnt.binomial)+
  geom_ribbon(aes(x=Position, fill=strand, ymin=Enrrichment_l, ymax=Enrrichment_u), alpha=0.3) +
    geom_line(aes(x=Position,y=Enrrichment, colour=strand)) +
    xlim(c(-300,300)) +
    facet_grid( exon_pos ~ exon_num   ) +
    theme_bw() +
    theme(legend.position = "top", legend.direction = "horizontal") 


Fig3.D.new


```




# G4 seq


Here we distribution of G4-seq experiments  



```{r, message=FALSE, error=FALSE, warning=FALSE}


G4_2019.K.Q1_up_plus <- read_dist_table("./G4_seq_2019/exon.up_plus.q1.bed.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.K.Q2_up_plus <- read_dist_table("./G4_seq_2019/exon.up_plus.q2.bed.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.K.Q3_up_plus <- read_dist_table("./G4_seq_2019/exon.up_plus.q3.bed.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.K.Q4_up_plus <- read_dist_table("./G4_seq_2019/exon.up_plus.q4.bed.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.bed.list.out.num")


G4_2019.K.Q1_down_plus <- read_dist_table("./G4_seq_2019/exon.down_plus.q1.bed.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.K.Q2_down_plus <- read_dist_table("./G4_seq_2019/exon.down_plus.q2.bed.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.K.Q3_down_plus <- read_dist_table("./G4_seq_2019/exon.down_plus.q3.bed.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.K.Q4_down_plus <- read_dist_table("./G4_seq_2019/exon.down_plus.q4.bed.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.bed.list.out.num")



G4_2019.K.Q1_up_minus <- read_dist_table("./G4_seq_2019/exon.up_minus.q1.bed.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.K.Q2_up_minus <- read_dist_table("./G4_seq_2019/exon.up_minus.q2.bed.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.K.Q3_up_minus <- read_dist_table("./G4_seq_2019/exon.up_minus.q3.bed.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.K.Q4_up_minus <- read_dist_table("./G4_seq_2019/exon.up_minus.q4.bed.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.bed.list.out.num")


G4_2019.K.Q1_down_minus <- read_dist_table("./G4_seq_2019/exon.down_minus.q1.bed.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.K.Q2_down_minus <- read_dist_table("./G4_seq_2019/exon.down_minus.q2.bed.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.K.Q3_down_minus <- read_dist_table("./G4_seq_2019/exon.down_minus.q3.bed.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.K.Q4_down_minus <- read_dist_table("./G4_seq_2019/exon.down_minus.q4.bed.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.bed.list.out.num")




G4_2019.K.Qs.plus.up.list <- list(G4_2019.K.Q1_up_plus, G4_2019.K.Q2_up_plus, G4_2019.K.Q3_up_plus, G4_2019.K.Q4_up_plus)
G4_2019.K.Qs.minus.list <- list(G4_2019.K.Q1_up_minus, G4_2019.K.Q2_up_minus, G4_2019.K.Q3_up_minus, G4_2019.K.Q4_up_minus)
G4_2019.K.Qs.plus.down.list <- list(G4_2019.K.Q1_down_plus, G4_2019.K.Q2_down_plus, G4_2019.K.Q3_down_plus, G4_2019.K.Q4_down_plus)
G4_2019.K.Qs.minus.down.list <- list(G4_2019.K.Q1_down_minus, G4_2019.K.Q2_down_minus, G4_2019.K.Q3_down_minus, G4_2019.K.Q4_down_minus)
G4_2019.K.window_len = 2000

G4_2019.K.Qs.total <- get_total_Qs(G4_2019.K.Qs.plus.up.list, G4_2019.K.Qs.minus.list, G4_2019.K.Qs.plus.down.list, G4_2019.K.Qs.minus.down.list, G4_2019.K.window_len)


ggplot(G4_2019.K.Qs.total)+
  geom_line(aes(x=Position,y=Enrrichment, colour=Q)) +
  xlim(c(-200,200)) +
  facet_grid( . ~ exon_pos ) +
  theme(text = element_text(size=15)) +
  theme_bw() + 
  labs(colour = "Splice site stregth quartile") +
  theme(legend.position = "top", legend.direction = "horizontal")

```







```{r, message=FALSE, error=FALSE, warning=FALSE}


G4_2019.PDS.Q1_up_plus <- read_dist_table("./G4_seq_2019/exon.up_plus.q1.bed.score.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.PDS.Q2_up_plus <- read_dist_table("./G4_seq_2019/exon.up_plus.q2.bed.score.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.PDS.Q3_up_plus <- read_dist_table("./G4_seq_2019/exon.up_plus.q3.bed.score.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.PDS.Q4_up_plus <- read_dist_table("./G4_seq_2019/exon.up_plus.q4.bed.score.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")


G4_2019.PDS.Q1_down_plus <- read_dist_table("./G4_seq_2019/exon.down_plus.q1.bed.score.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.PDS.Q2_down_plus <- read_dist_table("./G4_seq_2019/exon.down_plus.q2.bed.score.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.PDS.Q3_down_plus <- read_dist_table("./G4_seq_2019/exon.down_plus.q3.bed.score.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.PDS.Q4_down_plus <- read_dist_table("./G4_seq_2019/exon.down_plus.q4.bed.score.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")



G4_2019.PDS.Q1_up_minus <- read_dist_table("./G4_seq_2019/exon.up_minus.q1.bed.score.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.PDS.Q2_up_minus <- read_dist_table("./G4_seq_2019/exon.up_minus.q2.bed.score.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.PDS.Q3_up_minus <- read_dist_table("./G4_seq_2019/exon.up_minus.q3.bed.score.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.PDS.Q4_up_minus <- read_dist_table("./G4_seq_2019/exon.up_minus.q4.bed.score.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")


G4_2019.PDS.Q1_down_minus <- read_dist_table("./G4_seq_2019/exon.down_minus.q1.bed.score.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.PDS.Q2_down_minus <- read_dist_table("./G4_seq_2019/exon.down_minus.q2.bed.score.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.PDS.Q3_down_minus <- read_dist_table("./G4_seq_2019/exon.down_minus.q3.bed.score.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.PDS.Q4_down_minus <- read_dist_table("./G4_seq_2019/exon.down_minus.q4.bed.score.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")




G4_2019.PDS.Qs.plus.up.list <- list(G4_2019.PDS.Q1_up_plus, G4_2019.PDS.Q2_up_plus, G4_2019.PDS.Q3_up_plus, G4_2019.PDS.Q4_up_plus)
G4_2019.PDS.Qs.minus.list <- list(G4_2019.PDS.Q1_up_minus, G4_2019.PDS.Q2_up_minus, G4_2019.PDS.Q3_up_minus, G4_2019.PDS.Q4_up_minus)
G4_2019.PDS.Qs.plus.down.list <- list(G4_2019.PDS.Q1_down_plus, G4_2019.PDS.Q2_down_plus, G4_2019.PDS.Q3_down_plus, G4_2019.PDS.Q4_down_plus)
G4_2019.PDS.Qs.minus.down.list <- list(G4_2019.PDS.Q1_down_minus, G4_2019.PDS.Q2_down_minus, G4_2019.PDS.Q3_down_minus, G4_2019.PDS.Q4_down_minus)
G4_2019.PDS.window_len = 2000

G4_2019.PDS.Qs.total <- get_total_Qs(G4_2019.PDS.Qs.plus.up.list, G4_2019.PDS.Qs.minus.list, G4_2019.PDS.Qs.plus.down.list, G4_2019.PDS.Qs.minus.down.list, G4_2019.PDS.window_len)


ggplot(G4_2019.PDS.Qs.total)+
  geom_line(aes(x=Position,y=Enrrichment, colour=Q)) +
  xlim(c(-200,200)) +
  facet_grid( . ~ exon_pos ) +
  theme(text = element_text(size=15)) +
  theme_bw() + 
  labs(colour = "Splice site stregth quartile") +
  theme(legend.position = "top", legend.direction = "horizontal")

```





```{r, message=FALSE, error=FALSE, warning=FALSE}


G4_2019.template.PDS.Q1_up_plus <- read_dist_table("./G4_seq_2019/exon.up_plus.q1.bed.score.GSM3003540_Homo_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.template.PDS.Q2_up_plus <- read_dist_table("./G4_seq_2019/exon.up_plus.q2.bed.score.GSM3003540_Homo_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.template.PDS.Q3_up_plus <- read_dist_table("./G4_seq_2019/exon.up_plus.q3.bed.score.GSM3003540_Homo_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.template.PDS.Q4_up_plus <- read_dist_table("./G4_seq_2019/exon.up_plus.q4.bed.score.GSM3003540_Homo_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")


G4_2019.template.PDS.Q1_up_minus <- read_dist_table("./G4_seq_2019/exon.up_minus.q1.bed.score.GSM3003540_Homo_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.template.PDS.Q2_up_minus <- read_dist_table("./G4_seq_2019/exon.up_minus.q2.bed.score.GSM3003540_Homo_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.template.PDS.Q3_up_minus <- read_dist_table("./G4_seq_2019/exon.up_minus.q3.bed.score.GSM3003540_Homo_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.template.PDS.Q4_up_minus <- read_dist_table("./G4_seq_2019/exon.up_minus.q4.bed.score.GSM3003540_Homo_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.bed.list.out.num")




G4_2019.non_template.PDS.Q1_up_plus <- read_dist_table("./G4_seq_2019/exon.up_plus.q1.bed.score.GSM3003540_Homo_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.non_template.PDS.Q2_up_plus <- read_dist_table("./G4_seq_2019/exon.up_plus.q2.bed.score.GSM3003540_Homo_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.non_template.PDS.Q3_up_plus <- read_dist_table("./G4_seq_2019/exon.up_plus.q3.bed.score.GSM3003540_Homo_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.non_template.PDS.Q4_up_plus <- read_dist_table("./G4_seq_2019/exon.up_plus.q4.bed.score.GSM3003540_Homo_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.bed.list.out.num")


G4_2019.non_template.PDS.Q1_up_minus <- read_dist_table("./G4_seq_2019/exon.up_minus.q1.bed.score.GSM3003540_Homo_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.non_template.PDS.Q2_up_minus <- read_dist_table("./G4_seq_2019/exon.up_minus.q2.bed.score.GSM3003540_Homo_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.non_template.PDS.Q3_up_minus <- read_dist_table("./G4_seq_2019/exon.up_minus.q3.bed.score.GSM3003540_Homo_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.non_template.PDS.Q4_up_minus <- read_dist_table("./G4_seq_2019/exon.up_minus.q4.bed.score.GSM3003540_Homo_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")






G4_2019.template.PDS.Q1_down_plus <- read_dist_table("./G4_seq_2019/exon.down_plus.q1.bed.score.GSM3003540_Homo_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.template.PDS.Q2_down_plus <- read_dist_table("./G4_seq_2019/exon.down_plus.q2.bed.score.GSM3003540_Homo_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.template.PDS.Q3_down_plus <- read_dist_table("./G4_seq_2019/exon.down_plus.q3.bed.score.GSM3003540_Homo_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.template.PDS.Q4_down_plus <- read_dist_table("./G4_seq_2019/exon.down_plus.q4.bed.score.GSM3003540_Homo_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")


G4_2019.template.PDS.Q1_down_minus <- read_dist_table("./G4_seq_2019/exon.down_minus.q1.bed.score.GSM3003540_Homo_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.template.PDS.Q2_down_minus <- read_dist_table("./G4_seq_2019/exon.down_minus.q2.bed.score.GSM3003540_Homo_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.template.PDS.Q3_down_minus <- read_dist_table("./G4_seq_2019/exon.down_minus.q3.bed.score.GSM3003540_Homo_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.template.PDS.Q4_down_minus <- read_dist_table("./G4_seq_2019/exon.down_minus.q4.bed.score.GSM3003540_Homo_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.bed.list.out.num")




G4_2019.non_template.PDS.Q1_down_plus <- read_dist_table("./G4_seq_2019/exon.down_plus.q1.bed.score.GSM3003540_Homo_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.non_template.PDS.Q2_down_plus <- read_dist_table("./G4_seq_2019/exon.down_plus.q2.bed.score.GSM3003540_Homo_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.non_template.PDS.Q3_down_plus <- read_dist_table("./G4_seq_2019/exon.down_plus.q3.bed.score.GSM3003540_Homo_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.non_template.PDS.Q4_down_plus <- read_dist_table("./G4_seq_2019/exon.down_plus.q4.bed.score.GSM3003540_Homo_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.bed.list.out.num")


G4_2019.non_template.PDS.Q1_down_minus <- read_dist_table("./G4_seq_2019/exon.down_minus.q1.bed.score.GSM3003540_Homo_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.non_template.PDS.Q2_down_minus <- read_dist_table("./G4_seq_2019/exon.down_minus.q2.bed.score.GSM3003540_Homo_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.non_template.PDS.Q3_down_minus <- read_dist_table("./G4_seq_2019/exon.down_minus.q3.bed.score.GSM3003540_Homo_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")
G4_2019.non_template.PDS.Q4_down_minus <- read_dist_table("./G4_seq_2019/exon.down_minus.q4.bed.score.GSM3003540_Homo_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.bed.list.out.num")





```



```{r}
G4_2019.template.PDS.Qs.plus.up.list <- list(G4_2019.template.PDS.Q1_up_plus, G4_2019.template.PDS.Q2_up_plus, G4_2019.template.PDS.Q3_up_plus, G4_2019.template.PDS.Q4_up_plus)
G4_2019.template.PDS.Qs.minus.list <- list(G4_2019.template.PDS.Q1_up_minus, G4_2019.template.PDS.Q2_up_minus, G4_2019.template.PDS.Q3_up_minus, G4_2019.template.PDS.Q4_up_minus)
G4_2019.template.PDS.Qs.plus.down.list <- list(G4_2019.template.PDS.Q1_down_plus, G4_2019.template.PDS.Q2_down_plus, G4_2019.template.PDS.Q3_down_plus, G4_2019.template.PDS.Q4_down_plus)
G4_2019.template.PDS.Qs.minus.down.list <- list(G4_2019.template.PDS.Q1_down_minus, G4_2019.template.PDS.Q2_down_minus, G4_2019.template.PDS.Q3_down_minus, G4_2019.template.PDS.Q4_down_minus)
G4_2019.template.PDS.window_len = 2000

G4_2019.template.PDS.Qs.total <- get_total_Qs(G4_2019.template.PDS.Qs.plus.up.list, G4_2019.template.PDS.Qs.minus.list, G4_2019.template.PDS.Qs.plus.down.list, G4_2019.template.PDS.Qs.minus.down.list, G4_2019.template.PDS.window_len)
```



```{r}
G4_2019.non_template.PDS.Qs.plus.up.list <- list(G4_2019.non_template.PDS.Q1_up_plus, G4_2019.non_template.PDS.Q2_up_plus, G4_2019.non_template.PDS.Q3_up_plus, G4_2019.non_template.PDS.Q4_up_plus)
G4_2019.non_template.PDS.Qs.minus.list <- list(G4_2019.non_template.PDS.Q1_up_minus, G4_2019.non_template.PDS.Q2_up_minus, G4_2019.non_template.PDS.Q3_up_minus, G4_2019.non_template.PDS.Q4_up_minus)
G4_2019.non_template.PDS.Qs.plus.down.list <- list(G4_2019.non_template.PDS.Q1_down_plus, G4_2019.non_template.PDS.Q2_down_plus, G4_2019.non_template.PDS.Q3_down_plus, G4_2019.non_template.PDS.Q4_down_plus)
G4_2019.non_template.PDS.Qs.minus.down.list <- list(G4_2019.non_template.PDS.Q1_down_minus, G4_2019.non_template.PDS.Q2_down_minus, G4_2019.non_template.PDS.Q3_down_minus, G4_2019.non_template.PDS.Q4_down_minus)
G4_2019.non_template.PDS.window_len = 2000

G4_2019.non_template.PDS.Qs.total <- get_total_Qs(G4_2019.non_template.PDS.Qs.plus.up.list, G4_2019.non_template.PDS.Qs.minus.list, G4_2019.non_template.PDS.Qs.plus.down.list, G4_2019.non_template.PDS.Qs.minus.down.list, G4_2019.non_template.PDS.window_len)
```


```{r}



G4_2019.template_non_template.PDS.Qs.total <- rbind(G4_2019.template.PDS.Qs.total, G4_2019.non_template.PDS.Qs.total)
  
half_n <- nrow(G4_2019.template_non_template.PDS.Qs.total)/2
G4_2019.template_non_template.PDS.Qs.total[, Strand:=rep(c("Template", "Non-template"), each=half_n) ]
  
  
  
  
G4_2019.template_non_template.PDS.Qs.total$Strand <-  factor(G4_2019.template_non_template.PDS.Qs.total$Strand, levels=c("Template", "Non-template" ))

  
```




```{r}

G4_2019.template_non_template.PDS.Qs.total$Q <- mapvalues(G4_2019.template_non_template.PDS.Qs.total$Q,  from =c(1:4), to =c("Q1", "Q2", "Q3", "Q4"))


ggplot(G4_2019.template_non_template.PDS.Qs.total)+
  geom_line(aes(x=Position,y=Enrrichment, colour=Q)) +
  xlim(c(-150,150)) +
  facet_grid( Strand ~ exon_pos ) +
  labs(colour = "Splice site stregth quartile") +
  theme_bw()  + 
  theme(legend.position = "top", legend.direction = "horizontal")



```



```{r}
G4_2019.template_non_template.PDS.Qs.total[Strand=="Non-template" , new_Strand:="Template"]
G4_2019.template_non_template.PDS.Qs.total[Strand=="Template" , new_Strand:="Non-template"]


  
G4_2019.template_non_template.PDS.Qs.total$new_Strand <-  factor(G4_2019.template_non_template.PDS.Qs.total$new_Strand, levels=c("Template", "Non-template" ))

ggplot(G4_2019.template_non_template.PDS.Qs.total)+
  geom_line(aes(x=Position,y=Enrrichment, colour=Q)) +
  xlim(c(-150,150)) +
  facet_grid( new_Strand ~ exon_pos ) +
  labs(colour = "Splice site stregth quartile") +
  theme_bw()  + 
  theme(legend.position = "top", legend.direction = "horizontal")

```














```{r, message=FALSE, error=FALSE, warning=FALSE}


G4_2019.template.K.Q1_up_plus <- read_dist_table("./G4_seq_2019/exon.up_plus.q1.bed.score.GSM3003539_Homo_all_w15_th-1_minus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.template.K.Q2_up_plus <- read_dist_table("./G4_seq_2019/exon.up_plus.q2.bed.score.GSM3003539_Homo_all_w15_th-1_minus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.template.K.Q3_up_plus <- read_dist_table("./G4_seq_2019/exon.up_plus.q3.bed.score.GSM3003539_Homo_all_w15_th-1_minus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.template.K.Q4_up_plus <- read_dist_table("./G4_seq_2019/exon.up_plus.q4.bed.score.GSM3003539_Homo_all_w15_th-1_minus.hits.max.K.w50.25.bed.bed.list.out.num")


G4_2019.template.K.Q1_up_minus <- read_dist_table("./G4_seq_2019/exon.up_minus.q1.bed.score.GSM3003539_Homo_all_w15_th-1_plus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.template.K.Q2_up_minus <- read_dist_table("./G4_seq_2019/exon.up_minus.q2.bed.score.GSM3003539_Homo_all_w15_th-1_plus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.template.K.Q3_up_minus <- read_dist_table("./G4_seq_2019/exon.up_minus.q3.bed.score.GSM3003539_Homo_all_w15_th-1_plus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.template.K.Q4_up_minus <- read_dist_table("./G4_seq_2019/exon.up_minus.q4.bed.score.GSM3003539_Homo_all_w15_th-1_plus.hits.max.K.w50.25.bed.bed.list.out.num")




G4_2019.non_template.K.Q1_up_plus <- read_dist_table("./G4_seq_2019/exon.up_plus.q1.bed.score.GSM3003539_Homo_all_w15_th-1_plus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.non_template.K.Q2_up_plus <- read_dist_table("./G4_seq_2019/exon.up_plus.q2.bed.score.GSM3003539_Homo_all_w15_th-1_plus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.non_template.K.Q3_up_plus <- read_dist_table("./G4_seq_2019/exon.up_plus.q3.bed.score.GSM3003539_Homo_all_w15_th-1_plus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.non_template.K.Q4_up_plus <- read_dist_table("./G4_seq_2019/exon.up_plus.q4.bed.score.GSM3003539_Homo_all_w15_th-1_plus.hits.max.K.w50.25.bed.bed.list.out.num")


G4_2019.non_template.K.Q1_up_minus <- read_dist_table("./G4_seq_2019/exon.up_minus.q1.bed.score.GSM3003539_Homo_all_w15_th-1_minus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.non_template.K.Q2_up_minus <- read_dist_table("./G4_seq_2019/exon.up_minus.q2.bed.score.GSM3003539_Homo_all_w15_th-1_minus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.non_template.K.Q3_up_minus <- read_dist_table("./G4_seq_2019/exon.up_minus.q3.bed.score.GSM3003539_Homo_all_w15_th-1_minus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.non_template.K.Q4_up_minus <- read_dist_table("./G4_seq_2019/exon.up_minus.q4.bed.score.GSM3003539_Homo_all_w15_th-1_minus.hits.max.K.w50.25.bed.bed.list.out.num")






G4_2019.template.K.Q1_down_plus <- read_dist_table("./G4_seq_2019/exon.down_plus.q1.bed.score.GSM3003539_Homo_all_w15_th-1_minus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.template.K.Q2_down_plus <- read_dist_table("./G4_seq_2019/exon.down_plus.q2.bed.score.GSM3003539_Homo_all_w15_th-1_minus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.template.K.Q3_down_plus <- read_dist_table("./G4_seq_2019/exon.down_plus.q3.bed.score.GSM3003539_Homo_all_w15_th-1_minus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.template.K.Q4_down_plus <- read_dist_table("./G4_seq_2019/exon.down_plus.q4.bed.score.GSM3003539_Homo_all_w15_th-1_minus.hits.max.K.w50.25.bed.bed.list.out.num")


G4_2019.template.K.Q1_down_minus <- read_dist_table("./G4_seq_2019/exon.down_minus.q1.bed.score.GSM3003539_Homo_all_w15_th-1_plus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.template.K.Q2_down_minus <- read_dist_table("./G4_seq_2019/exon.down_minus.q2.bed.score.GSM3003539_Homo_all_w15_th-1_plus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.template.K.Q3_down_minus <- read_dist_table("./G4_seq_2019/exon.down_minus.q3.bed.score.GSM3003539_Homo_all_w15_th-1_plus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.template.K.Q4_down_minus <- read_dist_table("./G4_seq_2019/exon.down_minus.q4.bed.score.GSM3003539_Homo_all_w15_th-1_plus.hits.max.K.w50.25.bed.bed.list.out.num")




G4_2019.non_template.K.Q1_down_plus <- read_dist_table("./G4_seq_2019/exon.down_plus.q1.bed.score.GSM3003539_Homo_all_w15_th-1_plus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.non_template.K.Q2_down_plus <- read_dist_table("./G4_seq_2019/exon.down_plus.q2.bed.score.GSM3003539_Homo_all_w15_th-1_plus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.non_template.K.Q3_down_plus <- read_dist_table("./G4_seq_2019/exon.down_plus.q3.bed.score.GSM3003539_Homo_all_w15_th-1_plus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.non_template.K.Q4_down_plus <- read_dist_table("./G4_seq_2019/exon.down_plus.q4.bed.score.GSM3003539_Homo_all_w15_th-1_plus.hits.max.K.w50.25.bed.bed.list.out.num")


G4_2019.non_template.K.Q1_down_minus <- read_dist_table("./G4_seq_2019/exon.down_minus.q1.bed.score.GSM3003539_Homo_all_w15_th-1_minus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.non_template.K.Q2_down_minus <- read_dist_table("./G4_seq_2019/exon.down_minus.q2.bed.score.GSM3003539_Homo_all_w15_th-1_minus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.non_template.K.Q3_down_minus <- read_dist_table("./G4_seq_2019/exon.down_minus.q3.bed.score.GSM3003539_Homo_all_w15_th-1_minus.hits.max.K.w50.25.bed.bed.list.out.num")
G4_2019.non_template.K.Q4_down_minus <- read_dist_table("./G4_seq_2019/exon.down_minus.q4.bed.score.GSM3003539_Homo_all_w15_th-1_minus.hits.max.K.w50.25.bed.bed.list.out.num")





```



```{r}
G4_2019.template.K.Qs.plus.up.list <- list(G4_2019.template.K.Q1_up_plus, G4_2019.template.K.Q2_up_plus, G4_2019.template.K.Q3_up_plus, G4_2019.template.K.Q4_up_plus)
G4_2019.template.K.Qs.minus.list <- list(G4_2019.template.K.Q1_up_minus, G4_2019.template.K.Q2_up_minus, G4_2019.template.K.Q3_up_minus, G4_2019.template.K.Q4_up_minus)
G4_2019.template.K.Qs.plus.down.list <- list(G4_2019.template.K.Q1_down_plus, G4_2019.template.K.Q2_down_plus, G4_2019.template.K.Q3_down_plus, G4_2019.template.K.Q4_down_plus)
G4_2019.template.K.Qs.minus.down.list <- list(G4_2019.template.K.Q1_down_minus, G4_2019.template.K.Q2_down_minus, G4_2019.template.K.Q3_down_minus, G4_2019.template.K.Q4_down_minus)
G4_2019.template.K.window_len = 2000

G4_2019.template.K.Qs.total <- get_total_Qs(G4_2019.template.K.Qs.plus.up.list, G4_2019.template.K.Qs.minus.list, G4_2019.template.K.Qs.plus.down.list, G4_2019.template.K.Qs.minus.down.list, G4_2019.template.K.window_len)
```



```{r}
G4_2019.non_template.K.Qs.plus.up.list <- list(G4_2019.non_template.K.Q1_up_plus, G4_2019.non_template.K.Q2_up_plus, G4_2019.non_template.K.Q3_up_plus, G4_2019.non_template.K.Q4_up_plus)
G4_2019.non_template.K.Qs.minus.list <- list(G4_2019.non_template.K.Q1_up_minus, G4_2019.non_template.K.Q2_up_minus, G4_2019.non_template.K.Q3_up_minus, G4_2019.non_template.K.Q4_up_minus)
G4_2019.non_template.K.Qs.plus.down.list <- list(G4_2019.non_template.K.Q1_down_plus, G4_2019.non_template.K.Q2_down_plus, G4_2019.non_template.K.Q3_down_plus, G4_2019.non_template.K.Q4_down_plus)
G4_2019.non_template.K.Qs.minus.down.list <- list(G4_2019.non_template.K.Q1_down_minus, G4_2019.non_template.K.Q2_down_minus, G4_2019.non_template.K.Q3_down_minus, G4_2019.non_template.K.Q4_down_minus)
G4_2019.non_template.K.window_len = 2000

G4_2019.non_template.K.Qs.total <- get_total_Qs(G4_2019.non_template.K.Qs.plus.up.list, G4_2019.non_template.K.Qs.minus.list, G4_2019.non_template.K.Qs.plus.down.list, G4_2019.non_template.K.Qs.minus.down.list, G4_2019.non_template.K.window_len)
```


```{r}

G4_2019.template.K.Qs.total
G4_2019.non_template.K.Qs.total


G4_2019.template_non_template.K.Qs.total <- rbind(G4_2019.template.K.Qs.total, G4_2019.non_template.K.Qs.total)
  
half_n <- nrow(G4_2019.template_non_template.K.Qs.total)/2
G4_2019.template_non_template.K.Qs.total[, Strand:=rep(c("Template", "Non-template"), each=half_n) ]
  
  
  
  
G4_2019.template_non_template.K.Qs.total$Strand <-  factor(G4_2019.template_non_template.K.Qs.total$Strand, levels=c("Template", "Non-template" ))

  
```







```{r}


G4_2019.template_non_template.K.Qs.total$Q <- mapvalues(G4_2019.template_non_template.K.Qs.total$Q,  from =c(1:4), to =c("Q1", "Q2", "Q3", "Q4"))
G4_2019.template_non_template.K.Qs.total[Strand=="Non-template" , new_Strand:="Template"]
G4_2019.template_non_template.K.Qs.total[Strand=="Template" , new_Strand:="Non-template"]



G4_2019.template_non_template.K.Qs.total$new_Strand <-  factor(G4_2019.template_non_template.K.Qs.total$new_Strand, levels=c("Template", "Non-template" ))

ggplot(G4_2019.template_non_template.K.Qs.total)+
  geom_line(aes(x=Position,y=Enrrichment, colour=Q)) +
  xlim(c(-150,150)) +
  facet_grid( new_Strand ~ exon_pos ) +
  labs(colour = "Splice site stregth quartile") +
  theme_bw()  + 
  theme(legend.position = "top", legend.direction = "horizontal")


```










```{r, message=FALSE, error=FALSE, warning=FALSE}
#2015

G4_2015.Na_PDS.Q1_up_plus <- read_dist_table("./G4_seq_2015/exon.up_plus.q1.bed.score.GSE63874_Na_PDS_plus_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.Na_PDS.Q2_up_plus <- read_dist_table("./G4_seq_2015/exon.up_plus.q2.bed.score.GSE63874_Na_PDS_plus_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.Na_PDS.Q3_up_plus <- read_dist_table("./G4_seq_2015/exon.up_plus.q3.bed.score.GSE63874_Na_PDS_plus_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.Na_PDS.Q4_up_plus <- read_dist_table("./G4_seq_2015/exon.up_plus.q4.bed.score.GSE63874_Na_PDS_plus_minus_hits_intersect.bed.bed.list.out.num")


G4_2015.Na_PDS.Q1_up_minus <- read_dist_table("./G4_seq_2015/exon.up_minus.q1.bed.score.GSE63874_Na_PDS_plus_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.Na_PDS.Q2_up_minus <- read_dist_table("./G4_seq_2015/exon.up_minus.q2.bed.score.GSE63874_Na_PDS_plus_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.Na_PDS.Q3_up_minus <- read_dist_table("./G4_seq_2015/exon.up_minus.q3.bed.score.GSE63874_Na_PDS_plus_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.Na_PDS.Q4_up_minus <- read_dist_table("./G4_seq_2015/exon.up_minus.q4.bed.score.GSE63874_Na_PDS_plus_minus_hits_intersect.bed.bed.list.out.num")


G4_2015.Na_PDS.Q1_down_plus <- read_dist_table("./G4_seq_2015/exon.down_plus.q1.bed.score.GSE63874_Na_PDS_plus_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.Na_PDS.Q2_down_plus <- read_dist_table("./G4_seq_2015/exon.down_plus.q2.bed.score.GSE63874_Na_PDS_plus_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.Na_PDS.Q3_down_plus <- read_dist_table("./G4_seq_2015/exon.down_plus.q3.bed.score.GSE63874_Na_PDS_plus_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.Na_PDS.Q4_down_plus <- read_dist_table("./G4_seq_2015/exon.down_plus.q4.bed.score.GSE63874_Na_PDS_plus_minus_hits_intersect.bed.bed.list.out.num")


G4_2015.Na_PDS.Q1_down_minus <- read_dist_table("./G4_seq_2015/exon.down_minus.q1.bed.score.GSE63874_Na_PDS_plus_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.Na_PDS.Q2_down_minus <- read_dist_table("./G4_seq_2015/exon.down_minus.q2.bed.score.GSE63874_Na_PDS_plus_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.Na_PDS.Q3_down_minus <- read_dist_table("./G4_seq_2015/exon.down_minus.q3.bed.score.GSE63874_Na_PDS_plus_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.Na_PDS.Q4_down_minus <- read_dist_table("./G4_seq_2015/exon.down_minus.q4.bed.score.GSE63874_Na_PDS_plus_minus_hits_intersect.bed.bed.list.out.num")







G4_2015.Na_PDS.Qs.plus.up.list <- list(G4_2015.Na_PDS.Q1_up_plus, G4_2015.Na_PDS.Q2_up_plus, G4_2015.Na_PDS.Q3_up_plus, G4_2015.Na_PDS.Q4_up_plus)
G4_2015.Na_PDS.Qs.minus.list <- list(G4_2015.Na_PDS.Q1_up_minus, G4_2015.Na_PDS.Q2_up_minus, G4_2015.Na_PDS.Q3_up_minus, G4_2015.Na_PDS.Q4_up_minus)
G4_2015.Na_PDS.Qs.plus.down.list <- list(G4_2015.Na_PDS.Q1_down_plus, G4_2015.Na_PDS.Q2_down_plus, G4_2015.Na_PDS.Q3_down_plus, G4_2015.Na_PDS.Q4_down_plus)
G4_2015.Na_PDS.Qs.minus.down.list <- list(G4_2015.Na_PDS.Q1_down_minus, G4_2015.Na_PDS.Q2_down_minus, G4_2015.Na_PDS.Q3_down_minus, G4_2015.Na_PDS.Q4_down_minus)
G4_2015.Na_PDS.window_len = 2000

G4_2015.Na_PDS.Qs.total <- get_total_Qs(G4_2015.Na_PDS.Qs.plus.up.list, G4_2015.Na_PDS.Qs.minus.list, G4_2015.Na_PDS.Qs.plus.down.list, G4_2015.Na_PDS.Qs.minus.down.list, G4_2015.Na_PDS.window_len)


ggplot(G4_2015.Na_PDS.Qs.total)+
  geom_line(aes(x=Position,y=Enrrichment, colour=Q)) +
  xlim(c(-200,200)) +
  facet_grid( . ~ exon_pos ) +
  theme(text = element_text(size=15)) +
  theme_bw() + 
  labs(colour = "Splice site stregth quartile") +
  theme(legend.position = "top", legend.direction = "horizontal")

```







```{r, message=FALSE, error=FALSE, warning=FALSE}
G4_2015.Na_K.Q1_up_plus <- read_dist_table("./G4_seq_2015/exon.up_plus.q1.bed.score.GSE63874_Na_K_plus_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.Na_K.Q2_up_plus <- read_dist_table("./G4_seq_2015/exon.up_plus.q2.bed.score.GSE63874_Na_K_plus_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.Na_K.Q3_up_plus <- read_dist_table("./G4_seq_2015/exon.up_plus.q3.bed.score.GSE63874_Na_K_plus_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.Na_K.Q4_up_plus <- read_dist_table("./G4_seq_2015/exon.up_plus.q4.bed.score.GSE63874_Na_K_plus_minus_hits_intersect.bed.bed.list.out.num")


G4_2015.Na_K.Q1_up_minus <- read_dist_table("./G4_seq_2015/exon.up_minus.q1.bed.score.GSE63874_Na_K_plus_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.Na_K.Q2_up_minus <- read_dist_table("./G4_seq_2015/exon.up_minus.q2.bed.score.GSE63874_Na_K_plus_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.Na_K.Q3_up_minus <- read_dist_table("./G4_seq_2015/exon.up_minus.q3.bed.score.GSE63874_Na_K_plus_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.Na_K.Q4_up_minus <- read_dist_table("./G4_seq_2015/exon.up_minus.q4.bed.score.GSE63874_Na_K_plus_minus_hits_intersect.bed.bed.list.out.num")


G4_2015.Na_K.Q1_down_plus <- read_dist_table("./G4_seq_2015/exon.down_plus.q1.bed.score.GSE63874_Na_K_plus_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.Na_K.Q2_down_plus <- read_dist_table("./G4_seq_2015/exon.down_plus.q2.bed.score.GSE63874_Na_K_plus_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.Na_K.Q3_down_plus <- read_dist_table("./G4_seq_2015/exon.down_plus.q3.bed.score.GSE63874_Na_K_plus_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.Na_K.Q4_down_plus <- read_dist_table("./G4_seq_2015/exon.down_plus.q4.bed.score.GSE63874_Na_K_plus_minus_hits_intersect.bed.bed.list.out.num")


G4_2015.Na_K.Q1_down_minus <- read_dist_table("./G4_seq_2015/exon.down_minus.q1.bed.score.GSE63874_Na_K_plus_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.Na_K.Q2_down_minus <- read_dist_table("./G4_seq_2015/exon.down_minus.q2.bed.score.GSE63874_Na_K_plus_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.Na_K.Q3_down_minus <- read_dist_table("./G4_seq_2015/exon.down_minus.q3.bed.score.GSE63874_Na_K_plus_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.Na_K.Q4_down_minus <- read_dist_table("./G4_seq_2015/exon.down_minus.q4.bed.score.GSE63874_Na_K_plus_minus_hits_intersect.bed.bed.list.out.num")







G4_2015.Na_K.Qs.plus.up.list <- list(G4_2015.Na_K.Q1_up_plus, G4_2015.Na_K.Q2_up_plus, G4_2015.Na_K.Q3_up_plus, G4_2015.Na_K.Q4_up_plus)
G4_2015.Na_K.Qs.minus.list <- list(G4_2015.Na_K.Q1_up_minus, G4_2015.Na_K.Q2_up_minus, G4_2015.Na_K.Q3_up_minus, G4_2015.Na_K.Q4_up_minus)
G4_2015.Na_K.Qs.plus.down.list <- list(G4_2015.Na_K.Q1_down_plus, G4_2015.Na_K.Q2_down_plus, G4_2015.Na_K.Q3_down_plus, G4_2015.Na_K.Q4_down_plus)
G4_2015.Na_K.Qs.minus.down.list <- list(G4_2015.Na_K.Q1_down_minus, G4_2015.Na_K.Q2_down_minus, G4_2015.Na_K.Q3_down_minus, G4_2015.Na_K.Q4_down_minus)
G4_2015.Na_K.window_len = 2000

G4_2015.Na_K.Qs.total <- get_total_Qs(G4_2015.Na_K.Qs.plus.up.list, G4_2015.Na_K.Qs.minus.list, G4_2015.Na_K.Qs.plus.down.list, G4_2015.Na_K.Qs.minus.down.list, G4_2015.Na_K.window_len)


ggplot(G4_2015.Na_K.Qs.total)+
  geom_line(aes(x=Position,y=Enrrichment, colour=Q)) +
  xlim(c(-200,200)) +
  facet_grid( . ~ exon_pos ) +
  theme(text = element_text(size=15)) +
  theme_bw() + 
  labs(colour = "Splice site stregth quartile") +
  theme(legend.position = "top", legend.direction = "horizontal")

```







```{r, message=FALSE, error=FALSE, warning=FALSE}


G4_2015.template.Na_PDS.Q1_up_plus <- read_dist_table("./G4_seq_2015/exon.up_plus.q1.bed.score.GSE63874_Na_PDS_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.template.Na_PDS.Q2_up_plus <- read_dist_table("./G4_seq_2015/exon.up_plus.q2.bed.score.GSE63874_Na_PDS_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.template.Na_PDS.Q3_up_plus <- read_dist_table("./G4_seq_2015/exon.up_plus.q3.bed.score.GSE63874_Na_PDS_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.template.Na_PDS.Q4_up_plus <- read_dist_table("./G4_seq_2015/exon.up_plus.q4.bed.score.GSE63874_Na_PDS_minus_hits_intersect.bed.bed.list.out.num")


G4_2015.template.Na_PDS.Q1_up_minus <- read_dist_table("./G4_seq_2015/exon.up_minus.q1.bed.score.GSE63874_Na_PDS_plus_hits_intersect.bed.bed.list.out.num")
G4_2015.template.Na_PDS.Q2_up_minus <- read_dist_table("./G4_seq_2015/exon.up_minus.q2.bed.score.GSE63874_Na_PDS_plus_hits_intersect.bed.bed.list.out.num")
G4_2015.template.Na_PDS.Q3_up_minus <- read_dist_table("./G4_seq_2015/exon.up_minus.q3.bed.score.GSE63874_Na_PDS_plus_hits_intersect.bed.bed.list.out.num")
G4_2015.template.Na_PDS.Q4_up_minus <- read_dist_table("./G4_seq_2015/exon.up_minus.q4.bed.score.GSE63874_Na_PDS_plus_hits_intersect.bed.bed.list.out.num")



G4_2015.non_template.Na_PDS.Q1_up_plus <- read_dist_table("./G4_seq_2015/exon.up_plus.q1.bed.score.GSE63874_Na_PDS_plus_hits_intersect.bed.bed.list.out.num")
G4_2015.non_template.Na_PDS.Q2_up_plus <- read_dist_table("./G4_seq_2015/exon.up_plus.q2.bed.score.GSE63874_Na_PDS_plus_hits_intersect.bed.bed.list.out.num")
G4_2015.non_template.Na_PDS.Q3_up_plus <- read_dist_table("./G4_seq_2015/exon.up_plus.q3.bed.score.GSE63874_Na_PDS_plus_hits_intersect.bed.bed.list.out.num")
G4_2015.non_template.Na_PDS.Q4_up_plus <- read_dist_table("./G4_seq_2015/exon.up_plus.q4.bed.score.GSE63874_Na_PDS_plus_hits_intersect.bed.bed.list.out.num")


G4_2015.non_template.Na_PDS.Q1_up_minus <- read_dist_table("./G4_seq_2015/exon.up_minus.q1.bed.score.GSE63874_Na_PDS_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.non_template.Na_PDS.Q2_up_minus <- read_dist_table("./G4_seq_2015/exon.up_minus.q2.bed.score.GSE63874_Na_PDS_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.non_template.Na_PDS.Q3_up_minus <- read_dist_table("./G4_seq_2015/exon.up_minus.q3.bed.score.GSE63874_Na_PDS_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.non_template.Na_PDS.Q4_up_minus <- read_dist_table("./G4_seq_2015/exon.up_minus.q4.bed.score.GSE63874_Na_PDS_minus_hits_intersect.bed.bed.list.out.num")







G4_2015.template.Na_PDS.Q1_down_plus <- read_dist_table("./G4_seq_2015/exon.down_plus.q1.bed.score.GSE63874_Na_PDS_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.template.Na_PDS.Q2_down_plus <- read_dist_table("./G4_seq_2015/exon.down_plus.q2.bed.score.GSE63874_Na_PDS_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.template.Na_PDS.Q3_down_plus <- read_dist_table("./G4_seq_2015/exon.down_plus.q3.bed.score.GSE63874_Na_PDS_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.template.Na_PDS.Q4_down_plus <- read_dist_table("./G4_seq_2015/exon.down_plus.q4.bed.score.GSE63874_Na_PDS_minus_hits_intersect.bed.bed.list.out.num")


G4_2015.template.Na_PDS.Q1_down_minus <- read_dist_table("./G4_seq_2015/exon.down_minus.q1.bed.score.GSE63874_Na_PDS_plus_hits_intersect.bed.bed.list.out.num")
G4_2015.template.Na_PDS.Q2_down_minus <- read_dist_table("./G4_seq_2015/exon.down_minus.q2.bed.score.GSE63874_Na_PDS_plus_hits_intersect.bed.bed.list.out.num")
G4_2015.template.Na_PDS.Q3_down_minus <- read_dist_table("./G4_seq_2015/exon.down_minus.q3.bed.score.GSE63874_Na_PDS_plus_hits_intersect.bed.bed.list.out.num")
G4_2015.template.Na_PDS.Q4_down_minus <- read_dist_table("./G4_seq_2015/exon.down_minus.q4.bed.score.GSE63874_Na_PDS_plus_hits_intersect.bed.bed.list.out.num")



G4_2015.non_template.Na_PDS.Q1_down_plus <- read_dist_table("./G4_seq_2015/exon.down_plus.q1.bed.score.GSE63874_Na_PDS_plus_hits_intersect.bed.bed.list.out.num")
G4_2015.non_template.Na_PDS.Q2_down_plus <- read_dist_table("./G4_seq_2015/exon.down_plus.q2.bed.score.GSE63874_Na_PDS_plus_hits_intersect.bed.bed.list.out.num")
G4_2015.non_template.Na_PDS.Q3_down_plus <- read_dist_table("./G4_seq_2015/exon.down_plus.q3.bed.score.GSE63874_Na_PDS_plus_hits_intersect.bed.bed.list.out.num")
G4_2015.non_template.Na_PDS.Q4_down_plus <- read_dist_table("./G4_seq_2015/exon.down_plus.q4.bed.score.GSE63874_Na_PDS_plus_hits_intersect.bed.bed.list.out.num")


G4_2015.non_template.Na_PDS.Q1_down_minus <- read_dist_table("./G4_seq_2015/exon.down_minus.q1.bed.score.GSE63874_Na_PDS_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.non_template.Na_PDS.Q2_down_minus <- read_dist_table("./G4_seq_2015/exon.down_minus.q2.bed.score.GSE63874_Na_PDS_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.non_template.Na_PDS.Q3_down_minus <- read_dist_table("./G4_seq_2015/exon.down_minus.q3.bed.score.GSE63874_Na_PDS_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.non_template.Na_PDS.Q4_down_minus <- read_dist_table("./G4_seq_2015/exon.down_minus.q4.bed.score.GSE63874_Na_PDS_minus_hits_intersect.bed.bed.list.out.num")





```



```{r}
G4_2015.template.Na_PDS.Qs.plus.up.list <- list(G4_2015.template.Na_PDS.Q1_up_plus, G4_2015.template.Na_PDS.Q2_up_plus, G4_2015.template.Na_PDS.Q3_up_plus, G4_2015.template.Na_PDS.Q4_up_plus)
G4_2015.template.Na_PDS.Qs.minus.list <- list(G4_2015.template.Na_PDS.Q1_up_minus, G4_2015.template.Na_PDS.Q2_up_minus, G4_2015.template.Na_PDS.Q3_up_minus, G4_2015.template.Na_PDS.Q4_up_minus)
G4_2015.template.Na_PDS.Qs.plus.down.list <- list(G4_2015.template.Na_PDS.Q1_down_plus, G4_2015.template.Na_PDS.Q2_down_plus, G4_2015.template.Na_PDS.Q3_down_plus, G4_2015.template.Na_PDS.Q4_down_plus)
G4_2015.template.Na_PDS.Qs.minus.down.list <- list(G4_2015.template.Na_PDS.Q1_down_minus, G4_2015.template.Na_PDS.Q2_down_minus, G4_2015.template.Na_PDS.Q3_down_minus, G4_2015.template.Na_PDS.Q4_down_minus)
G4_2015.template.Na_PDS.window_len = 2000

G4_2015.template.Na_PDS.Qs.total <- get_total_Qs(G4_2015.template.Na_PDS.Qs.plus.up.list, G4_2015.template.Na_PDS.Qs.minus.list, G4_2015.template.Na_PDS.Qs.plus.down.list, G4_2015.template.Na_PDS.Qs.minus.down.list, G4_2015.template.Na_PDS.window_len)
```



```{r}
G4_2015.non_template.Na_PDS.Qs.plus.up.list <- list(G4_2015.non_template.Na_PDS.Q1_up_plus, G4_2015.non_template.Na_PDS.Q2_up_plus, G4_2015.non_template.Na_PDS.Q3_up_plus, G4_2015.non_template.Na_PDS.Q4_up_plus)
G4_2015.non_template.Na_PDS.Qs.minus.list <- list(G4_2015.non_template.Na_PDS.Q1_up_minus, G4_2015.non_template.Na_PDS.Q2_up_minus, G4_2015.non_template.Na_PDS.Q3_up_minus, G4_2015.non_template.Na_PDS.Q4_up_minus)
G4_2015.non_template.Na_PDS.Qs.plus.down.list <- list(G4_2015.non_template.Na_PDS.Q1_down_plus, G4_2015.non_template.Na_PDS.Q2_down_plus, G4_2015.non_template.Na_PDS.Q3_down_plus, G4_2015.non_template.Na_PDS.Q4_down_plus)
G4_2015.non_template.Na_PDS.Qs.minus.down.list <- list(G4_2015.non_template.Na_PDS.Q1_down_minus, G4_2015.non_template.Na_PDS.Q2_down_minus, G4_2015.non_template.Na_PDS.Q3_down_minus, G4_2015.non_template.Na_PDS.Q4_down_minus)
G4_2015.non_template.Na_PDS.window_len = 2000

G4_2015.non_template.Na_PDS.Qs.total <- get_total_Qs(G4_2015.non_template.Na_PDS.Qs.plus.up.list, G4_2015.non_template.Na_PDS.Qs.minus.list, G4_2015.non_template.Na_PDS.Qs.plus.down.list, G4_2015.non_template.Na_PDS.Qs.minus.down.list, G4_2015.non_template.Na_PDS.window_len)
```


```{r}

G4_2015.template.Na_PDS.Qs.total
G4_2015.non_template.Na_PDS.Qs.total


G4_2015.template_non_template.Na_PDS.Qs.total <- rbind(G4_2015.template.Na_PDS.Qs.total, G4_2015.non_template.Na_PDS.Qs.total)
  
half_n <- nrow(G4_2015.template_non_template.Na_PDS.Qs.total)/2
G4_2015.template_non_template.Na_PDS.Qs.total[, Strand:=rep(c("Template", "Non-template"), each=half_n) ]
  
  
  
  
G4_2015.template_non_template.Na_PDS.Qs.total$Strand <-  factor(G4_2015.template_non_template.Na_PDS.Qs.total$Strand, levels=c("Template", "Non-template" ))

  
```




```{r}


G4_2015.template_non_template.Na_PDS.Qs.total$Q <- mapvalues(G4_2015.template_non_template.Na_PDS.Qs.total$Q,  from =c(1:4), to =c("Q1", "Q2", "Q3", "Q4"))
G4_2015.template_non_template.Na_PDS.Qs.total[Strand=="Non-template" , new_Strand:="Template"]
G4_2015.template_non_template.Na_PDS.Qs.total[Strand=="Template" , new_Strand:="Non-template"]



G4_2015.template_non_template.Na_PDS.Qs.total$new_Strand <-  factor(G4_2015.template_non_template.Na_PDS.Qs.total$new_Strand, levels=c("Template", "Non-template" ))

ggplot(G4_2015.template_non_template.Na_PDS.Qs.total)+
  geom_line(aes(x=Position,y=Enrrichment, colour=Q)) +
  xlim(c(-300,300)) +
  facet_grid( new_Strand ~ exon_pos ) +
  labs(colour = "Splice site stregth quartile") +
  theme_bw()  + 
  theme(legend.position = "top", legend.direction = "horizontal")


```








```{r, message=FALSE, error=FALSE, warning=FALSE}


G4_2015.template.Na_K.Q1_up_plus <- read_dist_table("./G4_seq_2015/exon.up_plus.q1.bed.score.GSE63874_Na_K_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.template.Na_K.Q2_up_plus <- read_dist_table("./G4_seq_2015/exon.up_plus.q2.bed.score.GSE63874_Na_K_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.template.Na_K.Q3_up_plus <- read_dist_table("./G4_seq_2015/exon.up_plus.q3.bed.score.GSE63874_Na_K_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.template.Na_K.Q4_up_plus <- read_dist_table("./G4_seq_2015/exon.up_plus.q4.bed.score.GSE63874_Na_K_minus_hits_intersect.bed.bed.list.out.num")


G4_2015.template.Na_K.Q1_up_minus <- read_dist_table("./G4_seq_2015/exon.up_minus.q1.bed.score.GSE63874_Na_K_plus_hits_intersect.bed.bed.list.out.num")
G4_2015.template.Na_K.Q2_up_minus <- read_dist_table("./G4_seq_2015/exon.up_minus.q2.bed.score.GSE63874_Na_K_plus_hits_intersect.bed.bed.list.out.num")
G4_2015.template.Na_K.Q3_up_minus <- read_dist_table("./G4_seq_2015/exon.up_minus.q3.bed.score.GSE63874_Na_K_plus_hits_intersect.bed.bed.list.out.num")
G4_2015.template.Na_K.Q4_up_minus <- read_dist_table("./G4_seq_2015/exon.up_minus.q4.bed.score.GSE63874_Na_K_plus_hits_intersect.bed.bed.list.out.num")



G4_2015.non_template.Na_K.Q1_up_plus <- read_dist_table("./G4_seq_2015/exon.up_plus.q1.bed.score.GSE63874_Na_K_plus_hits_intersect.bed.bed.list.out.num")
G4_2015.non_template.Na_K.Q2_up_plus <- read_dist_table("./G4_seq_2015/exon.up_plus.q2.bed.score.GSE63874_Na_K_plus_hits_intersect.bed.bed.list.out.num")
G4_2015.non_template.Na_K.Q3_up_plus <- read_dist_table("./G4_seq_2015/exon.up_plus.q3.bed.score.GSE63874_Na_K_plus_hits_intersect.bed.bed.list.out.num")
G4_2015.non_template.Na_K.Q4_up_plus <- read_dist_table("./G4_seq_2015/exon.up_plus.q4.bed.score.GSE63874_Na_K_plus_hits_intersect.bed.bed.list.out.num")


G4_2015.non_template.Na_K.Q1_up_minus <- read_dist_table("./G4_seq_2015/exon.up_minus.q1.bed.score.GSE63874_Na_K_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.non_template.Na_K.Q2_up_minus <- read_dist_table("./G4_seq_2015/exon.up_minus.q2.bed.score.GSE63874_Na_K_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.non_template.Na_K.Q3_up_minus <- read_dist_table("./G4_seq_2015/exon.up_minus.q3.bed.score.GSE63874_Na_K_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.non_template.Na_K.Q4_up_minus <- read_dist_table("./G4_seq_2015/exon.up_minus.q4.bed.score.GSE63874_Na_K_minus_hits_intersect.bed.bed.list.out.num")





G4_2015.template.Na_K.Q1_down_plus <- read_dist_table("./G4_seq_2015/exon.down_plus.q1.bed.score.GSE63874_Na_K_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.template.Na_K.Q2_down_plus <- read_dist_table("./G4_seq_2015/exon.down_plus.q2.bed.score.GSE63874_Na_K_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.template.Na_K.Q3_down_plus <- read_dist_table("./G4_seq_2015/exon.down_plus.q3.bed.score.GSE63874_Na_K_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.template.Na_K.Q4_down_plus <- read_dist_table("./G4_seq_2015/exon.down_plus.q4.bed.score.GSE63874_Na_K_minus_hits_intersect.bed.bed.list.out.num")


G4_2015.template.Na_K.Q1_down_minus <- read_dist_table("./G4_seq_2015/exon.down_minus.q1.bed.score.GSE63874_Na_K_plus_hits_intersect.bed.bed.list.out.num")
G4_2015.template.Na_K.Q2_down_minus <- read_dist_table("./G4_seq_2015/exon.down_minus.q2.bed.score.GSE63874_Na_K_plus_hits_intersect.bed.bed.list.out.num")
G4_2015.template.Na_K.Q3_down_minus <- read_dist_table("./G4_seq_2015/exon.down_minus.q3.bed.score.GSE63874_Na_K_plus_hits_intersect.bed.bed.list.out.num")
G4_2015.template.Na_K.Q4_down_minus <- read_dist_table("./G4_seq_2015/exon.down_minus.q4.bed.score.GSE63874_Na_K_plus_hits_intersect.bed.bed.list.out.num")



G4_2015.non_template.Na_K.Q1_down_plus <- read_dist_table("./G4_seq_2015/exon.down_plus.q1.bed.score.GSE63874_Na_K_plus_hits_intersect.bed.bed.list.out.num")
G4_2015.non_template.Na_K.Q2_down_plus <- read_dist_table("./G4_seq_2015/exon.down_plus.q2.bed.score.GSE63874_Na_K_plus_hits_intersect.bed.bed.list.out.num")
G4_2015.non_template.Na_K.Q3_down_plus <- read_dist_table("./G4_seq_2015/exon.down_plus.q3.bed.score.GSE63874_Na_K_plus_hits_intersect.bed.bed.list.out.num")
G4_2015.non_template.Na_K.Q4_down_plus <- read_dist_table("./G4_seq_2015/exon.down_plus.q4.bed.score.GSE63874_Na_K_plus_hits_intersect.bed.bed.list.out.num")


G4_2015.non_template.Na_K.Q1_down_minus <- read_dist_table("./G4_seq_2015/exon.down_minus.q1.bed.score.GSE63874_Na_K_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.non_template.Na_K.Q2_down_minus <- read_dist_table("./G4_seq_2015/exon.down_minus.q2.bed.score.GSE63874_Na_K_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.non_template.Na_K.Q3_down_minus <- read_dist_table("./G4_seq_2015/exon.down_minus.q3.bed.score.GSE63874_Na_K_minus_hits_intersect.bed.bed.list.out.num")
G4_2015.non_template.Na_K.Q4_down_minus <- read_dist_table("./G4_seq_2015/exon.down_minus.q4.bed.score.GSE63874_Na_K_minus_hits_intersect.bed.bed.list.out.num")





```



```{r}
G4_2015.template.Na_K.Qs.plus.up.list <- list(G4_2015.template.Na_K.Q1_up_plus, G4_2015.template.Na_K.Q2_up_plus, G4_2015.template.Na_K.Q3_up_plus, G4_2015.template.Na_K.Q4_up_plus)
G4_2015.template.Na_K.Qs.minus.list <- list(G4_2015.template.Na_K.Q1_up_minus, G4_2015.template.Na_K.Q2_up_minus, G4_2015.template.Na_K.Q3_up_minus, G4_2015.template.Na_K.Q4_up_minus)
G4_2015.template.Na_K.Qs.plus.down.list <- list(G4_2015.template.Na_K.Q1_down_plus, G4_2015.template.Na_K.Q2_down_plus, G4_2015.template.Na_K.Q3_down_plus, G4_2015.template.Na_K.Q4_down_plus)
G4_2015.template.Na_K.Qs.minus.down.list <- list(G4_2015.template.Na_K.Q1_down_minus, G4_2015.template.Na_K.Q2_down_minus, G4_2015.template.Na_K.Q3_down_minus, G4_2015.template.Na_K.Q4_down_minus)
G4_2015.template.Na_K.window_len = 2000

G4_2015.template.Na_K.Qs.total <- get_total_Qs(G4_2015.template.Na_K.Qs.plus.up.list, G4_2015.template.Na_K.Qs.minus.list, G4_2015.template.Na_K.Qs.plus.down.list, G4_2015.template.Na_K.Qs.minus.down.list, G4_2015.template.Na_K.window_len)
```



```{r}
G4_2015.non_template.Na_K.Qs.plus.up.list <- list(G4_2015.non_template.Na_K.Q1_up_plus, G4_2015.non_template.Na_K.Q2_up_plus, G4_2015.non_template.Na_K.Q3_up_plus, G4_2015.non_template.Na_K.Q4_up_plus)
G4_2015.non_template.Na_K.Qs.minus.list <- list(G4_2015.non_template.Na_K.Q1_up_minus, G4_2015.non_template.Na_K.Q2_up_minus, G4_2015.non_template.Na_K.Q3_up_minus, G4_2015.non_template.Na_K.Q4_up_minus)
G4_2015.non_template.Na_K.Qs.plus.down.list <- list(G4_2015.non_template.Na_K.Q1_down_plus, G4_2015.non_template.Na_K.Q2_down_plus, G4_2015.non_template.Na_K.Q3_down_plus, G4_2015.non_template.Na_K.Q4_down_plus)
G4_2015.non_template.Na_K.Qs.minus.down.list <- list(G4_2015.non_template.Na_K.Q1_down_minus, G4_2015.non_template.Na_K.Q2_down_minus, G4_2015.non_template.Na_K.Q3_down_minus, G4_2015.non_template.Na_K.Q4_down_minus)
G4_2015.non_template.Na_K.window_len = 2000

G4_2015.non_template.Na_K.Qs.total <- get_total_Qs(G4_2015.non_template.Na_K.Qs.plus.up.list, G4_2015.non_template.Na_K.Qs.minus.list, G4_2015.non_template.Na_K.Qs.plus.down.list, G4_2015.non_template.Na_K.Qs.minus.down.list, G4_2015.non_template.Na_K.window_len)
```


```{r}

G4_2015.template.Na_K.Qs.total
G4_2015.non_template.Na_K.Qs.total


G4_2015.template_non_template.Na_K.Qs.total <- rbind(G4_2015.template.Na_K.Qs.total, G4_2015.non_template.Na_K.Qs.total)
  
half_n <- nrow(G4_2015.template_non_template.Na_K.Qs.total)/2
G4_2015.template_non_template.Na_K.Qs.total[, Strand:=rep(c("Template", "Non-template"), each=half_n) ]
  
  
  
  
G4_2015.template_non_template.Na_K.Qs.total$Strand <-  factor(G4_2015.template_non_template.Na_K.Qs.total$Strand, levels=c("Template", "Non-template" ))

  
```




```{r}


G4_2015.template_non_template.Na_K.Qs.total$Q <- mapvalues(G4_2015.template_non_template.Na_K.Qs.total$Q,  from =c(1:4), to =c("Q1", "Q2", "Q3", "Q4"))
G4_2015.template_non_template.Na_K.Qs.total[Strand=="Non-template" , new_Strand:="Template"]
G4_2015.template_non_template.Na_K.Qs.total[Strand=="Template" , new_Strand:="Non-template"]



G4_2015.template_non_template.Na_K.Qs.total$new_Strand <-  factor(G4_2015.template_non_template.Na_K.Qs.total$new_Strand, levels=c("Template", "Non-template" ))

ggplot(G4_2015.template_non_template.Na_K.Qs.total)+
  geom_line(aes(x=Position,y=Enrrichment, colour=Q)) +
  xlim(c(-300,300)) +
  facet_grid( new_Strand ~ exon_pos ) +
  labs(colour = "Splice site stregth quartile") +
  theme_bw()  + 
  theme(legend.position = "top", legend.direction = "horizontal")


```




# KCl RNA-seq


As K+ is known to stabilize G4, we analized (RNA-seq data)[https://www.ncbi.nlm.nih.gov/bioproject/PRJEB19451] comming from neurons after 4-hour KCl-indused depolarization. 






```{r, message=FALSE, error=FALSE, warning=FALSE}

KCL_human_upstream <- fread("../KCL/MicroExonator/Control_vs_KCL.diff.upstream_100nt.G4s"  ) 

KCL_human_upstream <- fread("../KCL/MicroExonator/Control_vs_KCL.diff.upstream_100nt.G4s"  ) #stop here
colnames(KCL_human_upstream) <- c("w.chrom", "w.start", "w.end", "w.strand", "chrom", "start", "end", "strand", "Type", "Psi_A", "Psi_B", "DeltaPsi", "Probability", "G4"  )


KCL_human_downstream<- fread("../KCL/MicroExonator/Control_vs_KCL.diff.downstream_100nt.G4s"  )
colnames(KCL_human_downstream) <-c("w.chrom", "w.start", "w.end", "w.strand", "chrom", "start", "end", "strand", "Type", "Psi_A", "Psi_B", "DeltaPsi", "Probability", "G4"  )


KCL_human <-  merge(KCL_human_upstream[, c("chrom",  "start", "end", "strand",  "Type" , "Psi_A", "Psi_B",  "DeltaPsi", "Probability", "G4") ],
                    KCL_human_downstream[, c("chrom",  "start", "end", "strand",  "Type" , "Psi_A", "Psi_B",  "DeltaPsi", "Probability", "G4") ], 
                    by=c("chrom",  "start", "end", "strand",  "Type" , "Psi_A", "Psi_B",  "DeltaPsi", "Probability") )

KCL_human[(G4.x>0 | G4.y>0 ), G4_bol:="With G-quadruplex"]
KCL_human[(G4.x==0 & G4.y==0 ), G4_bol:="Without G-quadruplex"]
KCL_human$G4_bol <- factor(KCL_human$G4_bol, levels=c("Without G-quadruplex", "With G-quadruplex") )


KCL_human[end - start > 30, exon:="exon" ]
KCL_human[end - start <= 30, exon:="microexon" ]

size.var=0.5
alpha.var=0.25

 ggplot(data=KCL_human[ Type=="CE",]) +
  geom_point( aes(x=-DeltaPsi, y=Probability), colour="grey", alpha=alpha.var, size=size.var) +
  geom_point(data=KCL_human[Type=="CE" & DeltaPsi>0.1 & Probability>0.9, ], aes(x=-DeltaPsi, y=Probability), colour="skyblue", alpha=alpha.var, size=size.var) +
   geom_point(data=KCL_human[Type=="CE" & DeltaPsi<(-0.1) & Probability>0.9, ], aes(x=-DeltaPsi, y=Probability), colour="red", alpha=alpha.var, size=size.var) +
  xlim(c(-0.8, 0.8)) +
  facet_grid( . ~ G4_bol  ) +
  xlab("DeltaPSI (Control - KCl)") +
  theme_bw()




```


To highlight NRXN2, SHANK1 and RBM10 we used biomaRt to find the gene name to Ensembl ID correspondence 


```{r, message=FALSE, error=FALSE, warning=FALSE}
library(biomaRt)

ensembl = useEnsembl(biomart="ensembl", dataset="hsapiens_gene_ensembl")







KCL_human[ Type=="CE"& DeltaPsi>0.1 & Probability>0.9,  ]


KCL_human[, Coords:=paste(chrom, paste(start, end, sep = "-"), strand, sep=":" )]


whippet.jls.exons.hg19 <-  fread("../KCL/whippet.jls.exons.tab")

KCL_human_CE <- merge(KCL_human[Type=="CE", ], whippet.jls.exons.hg19, by.x="Coords", by.y="Potential_Exon")


KCL_human_CE[ , ensembl_gene_id:=vapply(strsplit(KCL_human_CE$Gene, ".", fixed = TRUE), "[", "", 1)]

KCL_human_CE_genes <-  data.table(getBM(attributes=c('ensembl_gene_id', "wikigene_description", "hgnc_symbol"),filters = 'ensembl_gene_id', values = unique(vapply(strsplit(KCL_human_CE$Gene, ".", fixed = TRUE), "[", "", 1)) , mart = ensembl))


KCL_human_CE <-  merge(KCL_human_CE, KCL_human_CE_genes, by="ensembl_gene_id")











KCL_human_CE[ hgnc_symbol=="GRIN1",  ][ Type=="CE"& DeltaPsi>0.1 & Probability>0.85,  ]

KCL_human_CE[ hgnc_symbol=="NRXN2",  ][ Type=="CE"& DeltaPsi>0.1 & Probability>0.9,  ]
KCL_human_CE[ hgnc_symbol=="SHANK1",  ][ Type=="CE"& DeltaPsi>0.1 & Probability>0.85,  ]
KCL_human_CE[ hgnc_symbol=="RBM10",  ][ Type=="CE"& DeltaPsi>0.1 & Probability>0.9,  ]





  



```


```{r}
KCL_human_CE[ Type=="CE"& DeltaPsi>0.1 & Probability>0.9,  ]
```



### Microexons
```{r}
library(tidyr)


TOTAL_ME_stats <- function(KCL_human){

KCL_human[  , diff:="NA"]
KCL_human[ (DeltaPsi<=-0.1 & Probability>=0.9) , diff:="In"]
KCL_human[ (DeltaPsi>0.1 & Probability>=0.9) , diff:="Ex"]
  
  
KCL_human.TOTAL_and_ME <- separate(KCL_human,col = "Coord",into = c("chrom", "start_end"), sep = ":")
KCL_human.TOTAL_and_ME <- separate(KCL_human.TOTAL_and_ME,col = "start_end", into = c("start", "end"), sep = "-")


KCL_human.TOTAL_and_ME <- KCL_human.TOTAL_and_ME[Type=="CE", ]

KCL_human.TOTAL_and_ME[ ,  exon_type:="Exon"]
KCL_human.TOTAL_and_ME[((as.numeric(end)-as.numeric(start))<=30), exon_type:="Microexon" ]

table(KCL_human.TOTAL_and_ME[  exon_type=="Microexon", diff])


KCL_human.TOTAL_and_ME_stats <- KCL_human.TOTAL_and_ME[, .N, by=c("diff", "exon_type")]
KCL_human.TOTAL_and_ME_stats[, total:=sum(N), by=exon_type]
KCL_human.TOTAL_and_ME_stats[, Percent:=(N/total)*100]

return(KCL_human.TOTAL_and_ME_stats)

}


KCL_human.TOTAL_and_ME_stats <- TOTAL_ME_stats(KCL_human) 

ggplot(KCL_human.TOTAL_and_ME_stats2[diff!="NA"] ) +
  geom_bar( aes( exon_type, Percent, fill=diff), stat="identity", position="dodge"  )
```



```{r}



ME_TOTAL_chi_squared <- function(KCL_human.TOTAL_and_ME_stats){


KCL_human.TOTAL_and_ME_stats.matrix <-

matrix(cbind(
KCL_human.TOTAL_and_ME_stats[ diff!="NA", sum(N), by="exon_type" ][ exon_type=="Microexon", V1 ],
KCL_human.TOTAL_and_ME_stats[ diff!="NA", sum(N), by="exon_type" ][ exon_type=="Exon", V1 ],

KCL_human.TOTAL_and_ME_stats[ diff=="NA", sum(N), by="exon_type" ][ exon_type=="Microexon", V1 ],
KCL_human.TOTAL_and_ME_stats[ diff=="NA", sum(N), by="exon_type" ][ exon_type=="Exon", V1 ] ), nrow=2)

chi <- chisq.test(KCL_human.TOTAL_and_ME_stats.matrix)
return(chi)

}



matrix.in <-  matrix(nrow=2, c(KCL_human_stats[Type==i & diff=="In"]$WO_G4_NA, KCL_human_stats[Type==i & diff=="In"]$WO_G4, KCL_human_stats[Type==i & diff=="In"]$W_G4_NA, KCL_human_stats[Type==i & diff=="In"]$W_G4))
chisq.test(matrix.in)
```



```{r}
KCL_human.TOTAL_and_ME_stats <- TOTAL_ME_stats(KCL_human)

KCL_DIV10_Tc1.TOTAL_and_ME_stats <-  TOTAL_ME_stats(KCL_primary_cortical_neuron_DIV10_Tc1)
KCL_ESC_CD1.TOTAL_and_ME_stats <-  TOTAL_ME_stats(KCL_ESC_derived_neuron_CD1)
KCL_DIV4_CD1.TOTAL_and_ME_stats <- TOTAL_ME_stats(KCL_primary_cortical_neuron_DIV4_CD1)
KCL_DIV10_CD1.TOTAL_and_ME_stats <- TOTAL_ME_stats(KCL_primary_cortical_neuron_DIV10_CD1)


ME_TOTAL_chi_squared(KCL_human.TOTAL_and_ME_stats)
ME_TOTAL_chi_squared(KCL_DIV10_Tc1.TOTAL_and_ME_stats)
ME_TOTAL_chi_squared(KCL_ESC_CD1.TOTAL_and_ME_stats)
ME_TOTAL_chi_squared(KCL_DIV4_CD1.TOTAL_and_ME_stats)
ME_TOTAL_chi_squared(KCL_DIV10_CD1.TOTAL_and_ME_stats)


```


```{r}

KCL_human.TOTAL_and_ME_stats[, Experiment:="Human"]
KCL_DIV10_Tc1.TOTAL_and_ME_stats[, Experiment:="DIV10_Tc1"]
KCL_ESC_CD1.TOTAL_and_ME_stats[, Experiment:="ESC_CD1"]
KCL_DIV4_CD1.TOTAL_and_ME_stats[, Experiment:="DIV4_CD1"]
KCL_DIV10_CD1.TOTAL_and_ME_stats[, Experiment:="DIV10_CD1"]


KCL_TOTAL.TOTAL_and_ME_stats <- rbind(KCL_human.TOTAL_and_ME_stats,
                                      KCL_DIV10_Tc1.TOTAL_and_ME_stats,
                                      KCL_ESC_CD1.TOTAL_and_ME_stats,
                                      KCL_DIV4_CD1.TOTAL_and_ME_stats,
                                      KCL_DIV10_CD1.TOTAL_and_ME_stats)



KCL_TOTAL.TOTAL_and_ME_stats$Experiment <- factor(KCL_TOTAL.TOTAL_and_ME_stats$Experiment, levels=c("Human", "ESC_CD1", "DIV4_CD1", "DIV10_CD1", "DIV10_Tc1"))



ggplot(KCL_TOTAL.TOTAL_and_ME_stats[diff!="NA"] ) +
  geom_bar( aes( exon_type, Percent, fill=diff), stat="identity"  ) +
  facet_grid( . ~ Experiment) +
    theme(legend.position = "top", legend.direction = "horizontal", axis.text.x = element_text(angle = 30)) +
    scale_fill_brewer(palette = 'Blues') 
```




```{r}
table(KCL_human.TOTAL_and_ME[  exon_type=="Exon", diff])
```


```{r}
install.packages("epiR")
```




```{r, fig.width=5, fig.height=4}
library("ggrepel")

validation_exons <-c("chr1:110734594-110734835:+", "chr19:17731502-17731531:-", "chr11:20072835-20072879:+")

Fig5.A <- ggplot(data=KCL_human[ Type=="CE",]) +
  geom_point( aes(x=-DeltaPsi, y=Probability), colour="grey", alpha=alpha.var, size=size.var) +
  geom_point(data=KCL_human[Type=="CE" & DeltaPsi>0.1 & Probability>0.9, ], aes(x=-DeltaPsi, y=Probability), colour="skyblue", alpha=alpha.var, size=size.var) +
   geom_point(data=KCL_human[Type=="CE" & DeltaPsi<(-0.1) & Probability>0.9, ], aes(x=-DeltaPsi, y=Probability), colour="red", alpha=alpha.var, size=size.var) +
  xlim(c(-0.8, 0.8)) +
  facet_grid( . ~ G4_bol  ) +
  xlab("DeltaPSI (Control - KCl)") +
  theme_bw() +
  geom_point(data=KCL_human_CE[Coords %in% validation_exons , ], aes(x=-DeltaPsi, y=Probability), colour="blue", alpha=alpha.var, size=2) +
  geom_text_repel(data = KCL_human_CE[Coords %in% validation_exons , ],
                  colour="black", aes(x=-DeltaPsi, y=Probability),
                  nudge_y      = 3,
                  direction    = "x",
                  angle        = 90,
                  vjust        = 1,
                  segment.size = 0.2,
                  label.size = 0.05,
                  label=KCL_human_CE[Coords %in% validation_exons , hgnc_symbol]) +
  ylim(c(0.5, 1.15))

Fig5.A
```


To analyse all the Whippet node types, we calculate the odd-ratios of being differentially included given the G4 presence or absence 


```{r}


KCL_human[  , diff:="NA"]
KCL_human[ (DeltaPsi<=-0.1 & Probability>=0.9) , diff:="In"]
KCL_human[ (DeltaPsi>0.1 & Probability>=0.9) , diff:="Ex"]

KCL_human_stats <- merge(KCL_human[  G4_bol=="Without G-quadruplex",  .(WO_G4=.N) , by=c("Type", "diff")  ], 
KCL_human[  G4_bol=="With G-quadruplex",  .(W_G4=.N) , by=c("Type", "diff")  ],
by=c("Type", "diff"))



KCL_human_stats <- merge(KCL_human_stats[diff!="NA"],  KCL_human_stats[diff=="NA" , .(Type,  WO_G4_NA=WO_G4, W_G4_NA=W_G4) ], by="Type")

KCL_human_stats[, `:=`(WO_G4_fraq=WO_G4/WO_G4_NA, W_G4_fraq=W_G4/W_G4_NA)]

KCL_human_stats[ , OR := W_G4_fraq/WO_G4_fraq]




ggplot(KCL_human_stats) +
   geom_tile( aes(Type, diff, fill = log2(OR))) +
   scale_fill_gradient2(low="blue", high="red", mid = "grey", midpoint = 0, limits = c(-1.6, 1.6)) +
    theme(legend.position = "bottom", legend.direction = "horizontal") +
      theme(legend.position = "bottom", legend.direction = "horizontal") + theme(axis.text.x = element_text(angle = 90)) +labs(x = NULL, y = NULL)
```


We use chi-squared test to calculate the significance


```{r}


KCL_human_stats.chi <- data.table()

for ( i in unique(KCL_human_stats[Type!="AL"]$Type)){
  
matrix.ex <-  matrix(nrow=2, c(KCL_human_stats[Type==i & diff=="Ex"]$WO_G4_NA, KCL_human_stats[Type==i & diff=="Ex"]$WO_G4, KCL_human_stats[Type==i & diff=="Ex"]$W_G4_NA, KCL_human_stats[Type==i & diff=="Ex"]$W_G4))
matrix.ex_res <- chisq.test(matrix.ex)

matrix.in <-  matrix(nrow=2, c(KCL_human_stats[Type==i & diff=="In"]$WO_G4_NA, KCL_human_stats[Type==i & diff=="In"]$WO_G4, KCL_human_stats[Type==i & diff=="In"]$W_G4_NA, KCL_human_stats[Type==i & diff=="In"]$W_G4))
matrix.in_res <- chisq.test(matrix.in)



KCL_human_stats.chi <- rbind(KCL_human_stats.chi, cbind(i, "Ex", matrix.ex_res$p.value)) 
KCL_human_stats.chi <- rbind(KCL_human_stats.chi, cbind(i, "In", matrix.in_res$p.value)) 
                                                        
}

colnames(KCL_human_stats.chi) <- c("Type", "diff", "P_value")

ntest = nrow(KCL_human_stats.chi)

KCL_human_stats.chi[ , `:=`(P_value.Bonferroni=as.numeric(P_value)*ntest) ]

KCL_human_stats <- merge(KCL_human_stats, KCL_human_stats.chi, by=c("Type", "diff"))

```


```{r, fig.height=2.2, fig.width=5}


library(plyr)


ggplot(KCL_human_stats) +
  geom_point(aes(Type, diff, size = -log(P_value.Bonferroni), shape=P_value.Bonferroni<0.05, colour=log2(OR) ) ) +
  scale_colour_gradient2(low="blue", high="red", mid = "grey", midpoint = 0) +
  theme_bw() +
      theme(legend.position = "top", legend.direction = "horizontal") 





```



```{r}

splice_node_dot_plot <- function(upstream.path, downstream.path ) {

KCL_x_upstream <- fread(upstream.path  )
colnames(KCL_x_upstream) <- c("w.chrom", "w.start", "w.end", "w.strand", "chrom", "start", "end", "strand", "Type", "Psi_A", "Psi_B", "DeltaPsi", "Probability", "G4"  )


KCL_x_downstream<- fread(downstream.path  )
colnames(KCL_x_downstream) <-c("w.chrom", "w.start", "w.end", "w.strand", "chrom", "start", "end", "strand", "Type", "Psi_A", "Psi_B", "DeltaPsi", "Probability", "G4"  )


KCL_x <-  merge(KCL_x_upstream[, c("chrom",  "start", "end", "strand",  "Type" ,  "DeltaPsi", "Probability", "G4") ],
                    KCL_x_downstream[, c("chrom",  "start", "end", "strand",  "Type" ,  "DeltaPsi", "Probability", "G4") ], 
                    by=c("chrom",  "start", "end", "strand",  "Type" ,  "DeltaPsi", "Probability") )


#KCL_x <-  KCL_x[end-start>30 || Type!="CE", ]

KCL_x[(G4.x>0 | G4.y>0 ), G4_bol:="With G-quadruplex"]
KCL_x[(G4.x==0 & G4.y==0 ), G4_bol:="Without G-quadruplex"]
KCL_x$G4_bol <- factor(KCL_x$G4_bol, levels=c("Without G-quadruplex", "With G-quadruplex") )


KCL_x[  , diff:="NA"]
KCL_x[ (DeltaPsi<=-0.1 & Probability>=0.90) , diff:="In"]
KCL_x[ (DeltaPsi>0.1 & Probability>=0.90) , diff:="Ex"]

KCL_x_stats <- merge(KCL_x[  G4_bol=="Without G-quadruplex",  .(WO_G4=.N) , by=c("Type", "diff")  ], 
KCL_x[  G4_bol=="With G-quadruplex",  .(W_G4=.N) , by=c("Type", "diff")  ],
by=c("Type", "diff"))



KCL_x_stats <- merge(KCL_x_stats[diff!="NA"],  KCL_x_stats[diff=="NA" , .(Type,  WO_G4_NA=WO_G4, W_G4_NA=W_G4) ], by="Type")

KCL_x_stats[, `:=`(WO_G4_fraq=WO_G4/WO_G4_NA, W_G4_fraq=W_G4/W_G4_NA)]

KCL_x_stats[ , OR := W_G4_fraq/WO_G4_fraq]


KCL_x_stats.chi <- data.table()

for ( i in unique(KCL_x_stats[Type!="AL"]$Type)){
  
matrix.ex <-  matrix(nrow=2, c(KCL_x_stats[Type==i & diff=="Ex"]$WO_G4_NA, KCL_x_stats[Type==i & diff=="Ex"]$WO_G4, KCL_x_stats[Type==i & diff=="Ex"]$W_G4_NA, KCL_x_stats[Type==i & diff=="Ex"]$W_G4))
matrix.ex_res <- chisq.test(matrix.ex)

matrix.in <-  matrix(nrow=2, c(KCL_x_stats[Type==i & diff=="In"]$WO_G4_NA, KCL_x_stats[Type==i & diff=="In"]$WO_G4, KCL_x_stats[Type==i & diff=="In"]$W_G4_NA, KCL_x_stats[Type==i & diff=="In"]$W_G4))
matrix.in_res <- chisq.test(matrix.in)



KCL_x_stats.chi <- rbind(KCL_x_stats.chi, cbind(i, "Ex", matrix.ex_res$p.value)) 
KCL_x_stats.chi <- rbind(KCL_x_stats.chi, cbind(i, "In", matrix.in_res$p.value)) 
                                                        
}

colnames(KCL_x_stats.chi) <- c("Type", "diff", "P_value")

ntest = nrow(KCL_x_stats.chi)

KCL_x_stats.chi[ , `:=`(P_value.Bonferroni=as.numeric(P_value)*ntest) ]

KCL_x_stats <- merge(KCL_x_stats, KCL_x_stats.chi, by=c("Type", "diff"))

ggplot(KCL_x_stats) +
  geom_point(aes(Type, diff, size = -log(P_value.Bonferroni), shape=P_value.Bonferroni<0.05, colour=log2(OR) ) ) +
  scale_colour_gradient2(low="blue", high="red", mid = "grey", midpoint = 0) +
  theme_bw() +
      theme(legend.position = "top", legend.direction = "horizontal") 


return(KCL_x_stats)


}

```







```{r, message=FALSE, error=FALSE, warning=FALSE}

upstream.path = "../KCL/MicroExonator/ESC-derived_neuron_CD1.diff.upstream_100nt.G4s" 
downstream.path = "../KCL/MicroExonator/ESC-derived_neuron_CD1.diff.downstream_100nt.G4s" 

splice_node_dot_plot2 <- function(upstream.path, downstream.path ) {

KCL_x_upstream <- fread(upstream.path  )
colnames(KCL_x_upstream) <- c("w.chrom", "w.start", "w.end", "w.strand", "chrom", "start", "end", "strand", "Type", "Psi_A", "Psi_B", "DeltaPsi", "Probability", "G4"  )


KCL_x_downstream<- fread(downstream.path  )
colnames(KCL_x_downstream) <-c("w.chrom", "w.start", "w.end", "w.strand", "chrom", "start", "end", "strand", "Type", "Psi_A", "Psi_B", "DeltaPsi", "Probability", "G4"  )


KCL_x <-  merge(KCL_x_upstream[, c("chrom",  "start", "end", "strand",  "Type" ,  "DeltaPsi", "Probability", "G4") ],
                    KCL_x_downstream[, c("chrom",  "start", "end", "strand",  "Type" ,  "DeltaPsi", "Probability", "G4") ], 
                    by=c("chrom",  "start", "end", "strand",  "Type" ,  "DeltaPsi", "Probability") )


#KCL_x <-  KCL_x[end-start>30 || Type!="CE", ]

KCL_x[(G4.x>0 | G4.y>0 ), G4_bol:="With G-quadruplex"]
KCL_x[(G4.x==0 & G4.y==0 ), G4_bol:="Without G-quadruplex"]
KCL_x$G4_bol <- factor(KCL_x$G4_bol, levels=c("Without G-quadruplex", "With G-quadruplex") )


KCL_x[  , diff:="NA"]
KCL_x[ (DeltaPsi<=-0.1 & Probability>=0.90) , diff:="In"]
KCL_x[ (DeltaPsi>0.1 & Probability>=0.90) , diff:="Ex"]

KCL_x_stats <- merge(KCL_x[  G4_bol=="Without G-quadruplex",  .(WO_G4=.N) , by=c("Type", "diff")  ], 
KCL_x[  G4_bol=="With G-quadruplex",  .(W_G4=.N) , by=c("Type", "diff")  ],
by=c("Type", "diff"))



KCL_x_stats <- merge(KCL_x_stats[diff!="NA"],  KCL_x_stats[diff=="NA" , .(Type,  WO_G4_NA=WO_G4, W_G4_NA=W_G4) ], by="Type")

KCL_x_stats[, `:=`(WO_G4_fraq=WO_G4/WO_G4_NA, W_G4_fraq=W_G4/W_G4_NA)]

KCL_x_stats[ , OR := W_G4_fraq/WO_G4_fraq]


KCL_x_stats.chi <- data.table()

for ( i in unique(KCL_x_stats[Type!="AL"]$Type)){
  
matrix <-  matrix(nrow=2, c(KCL_x_stats[Type==i & diff=="In"]$WO_G4, KCL_x_stats[Type==i & diff=="Ex"]$WO_G4, KCL_x_stats[Type==i & diff=="In"]$W_G4, KCL_x_stats[Type==i & diff=="Ex"]$W_G4))
matrix.res <- chisq.test(matrix)





KCL_x_stats.chi <- rbind(KCL_x_stats.chi, cbind(i,  matrix.res$p.value)) 

                                                        
}

colnames(KCL_x_stats.chi) <- c("Type", "P_value")

ntest = nrow(KCL_x_stats.chi)

KCL_x_stats.chi[ , `:=`(P_value.Bonferroni=as.numeric(P_value)*ntest) ]

#KCL_x_stats <- merge(KCL_x_stats, KCL_x_stats.chi, by=c("Type", "diff"))

#ggplot(KCL_x_stats.chi) +
#  geom_point(aes(Type, diff, size = -log(P_value.Bonferroni), shape=P_value.Bonferroni<0.05, colour=log2(OR) ) ) +
#  scale_colour_gradient2(low="blue", high="red", mid = "grey", midpoint = 0) +
#  theme_bw() +
#      theme(legend.position = "top", legend.direction = "horizontal") 


return(KCL_x_stats.chi)


}

```




```{r}

diff_G4_ME_table <- function(upstream.path, downstream.path, dist_lim ) {


KCL_x_upstream <- fread(upstream.path  )
colnames(KCL_x_upstream) <- c("w.chrom", "w.start", "w.end", "w.strand", "chrom", "start", "end", "strand", "Type", "Psi_A", "Psi_B", "DeltaPsi", "Probability", "gchrom", "gstart", "gend", "gscore", "exon_dist"  )

KCL_x_upstream[ ,dist:=((gstart + (gend- gstart)/2) -w.start  ) ]
#KCL_x[(dist.x<=dist_lim | dist.y<=dist_lim ), G4_bol:="With G-quadruplex"]


KCL_x_downstream<- fread(downstream.path  )
colnames(KCL_x_downstream) <- c("w.chrom", "w.start", "w.end", "w.strand", "chrom", "start", "end", "strand", "Type", "Psi_A", "Psi_B", "DeltaPsi", "Probability", "gchrom", "gstart", "gend", "gscore", "exon_dist"  )


KCL_x_downstream[ ,dist:=((gstart + (gend- gstart)/2)) - w.start   ]

KCL_x <-  merge(KCL_x_upstream[, c("chrom",  "start", "end", "strand",  "Type" ,  "DeltaPsi", "Probability", "dist") ],
                    KCL_x_downstream[, c("chrom",  "start", "end", "strand",  "Type" ,  "DeltaPsi", "Probability", "dist") ], 
                    by=c("chrom",  "start", "end", "strand",  "Type" ,  "DeltaPsi", "Probability") )


#KCL_x <-  KCL_x[end-start>30 || Type!="CE", ]

KCL_x[ , G4_bol:="Without G-quadruplex"]

KCL_x[strand=="+" & ((dist.x >= -dist_lim &  dist.x  <= 0) | ( dist.y >=0 & dist.y<=dist_lim)) , G4_bol:="With G-quadruplex"]
KCL_x[strand=="-" & ((dist.y >= -dist_lim &  dist.y  <= 0) | ( dist.x >=0 & dist.x<=dist_lim)) , G4_bol:="With G-quadruplex"]

KCL_x$G4_bol <- factor(KCL_x$G4_bol, levels=c("Without G-quadruplex", "With G-quadruplex") )


KCL_x[  , diff:="NA"]
KCL_x[ (DeltaPsi<=-0.1 & Probability>=0.90) , diff:="In"]
KCL_x[ (DeltaPsi>0.1 & Probability>=0.90) , diff:="Ex"]


KCL_x <- KCL_x[Type=="CE"]

KCL_x[ ,  exon_class:="exon"]
KCL_x[end-start<=30,  exon_class:="microexon" ]

return(KCL_x)

}
```


```{r}

```



```{r}
upstream.path = "../Exons_G4s/human.exons.tvs.upstream.G4seq_K"
downstream.path = "../Exons_G4s/human.exons.tvs.downstream.G4seq_K"

G4_ME_table <- function(upstream.path, downstream.path, dist_lim ) {


KCL_x_upstream <- fread(upstream.path  )
colnames(KCL_x_upstream) <- c( "chrom", "sstart", "send", "sstrand", "start", "end", "strand",  "gchrom", "gstart", "gend", "gscore", "exon_dist"  )

KCL_x_upstream[ ,dist:=(sstart - (gstart + (gend- gstart)/2)) ]
#KCL_x[(dist.x<=dist_lim | dist.y<=dist_lim ), G4_bol:="With G-quadruplex"]


KCL_x_downstream<- fread(downstream.path  )
colnames(KCL_x_downstream) <- c( "chrom", "sstart", "send", "strand", "start", "end", "strand", "gchrom", "gstart", "gend", "gscore", "exon_dist"  )


KCL_x_downstream[ ,dist:=(sstart - (gstart + (gend- gstart)/2)) ]

KCL_x <-  merge(KCL_x_upstream[, c( "chrom", "start", "end", "strand", "dist") ],
                    KCL_x_downstream[, c( "chrom", "start", "end", "strand", "dist") ], 
                    by=c( "chrom", "start", "end", "strand") )


#KCL_x <-  KCL_x[end-start>30 || Type!="CE", ]

KCL_x[ , G4_bol:="Without G-quadruplex"]

KCL_x[ strand=="+" & ((dist.x >= -dist_lim &  dist.x  <= 0) | ( dist.y >=0 & dist.y<=dist_lim)) , G4_bol:="With G-quadruplex"]
KCL_x[ strand=="-" & ((dist.y >= -dist_lim &  dist.y  <= 0) | ( dist.x >=0 & dist.x<=dist_lim)) , G4_bol:="With G-quadruplex"]


KCL_x$G4_bol <- factor(KCL_x$G4_bol, levels=c("Without G-quadruplex", "With G-quadruplex") )


KCL_x[ ,  exon_class:="exon"]
KCL_x[end-start<=30,  exon_class:="microexon" ]

return(KCL_x)

}

```



```{r}

KCL_human_CE <- KCL_human[ , paste(chrom,   start,     end, sep="_" ) ]

human.exons.upstream.G4s <- fread("../Exons_G4s/human.exons.tvs.upstream.G4s")
human.exons.downstream.G4s <- fread("../Exons_G4s/human.exons.tvs.downstream.G4s")



colnames(human.exons.upstream.G4s) <- c( "chrom", "sstart", "send", "sstrand", "start", "end", "strand",  "gchrom", "gstart", "gend", "gscore", "exon_dist"  )


human.exons.upstream.G4s[ ,dist:=( (gstart + (gend- gstart)/2) - sstart   ) ]
human.exons.upstream.G4s[, exon_len:=end-start]

human.exons.upstream.G4s[ , G4_bol:="Without G-quadruplex"]
human.exons.upstream.G4s[ dist>=-100  & dist<=0 , G4_bol:="With G-quadruplex"]


human.exons.upstream.G4s[ , ID:=paste(chrom,   start,     end, sep="_" )]

human.exons.upstream.G4s[, filter:="Out"]
human.exons.upstream.G4s[ ID %in%  KCL_human_CE, ]

human.exons.upstream.G4s <- human.exons.upstream.G4s[ ID %in%  KCL_human_CE, ]

human.exons.upstream.G4s.stats <-human.exons.upstream.G4s[ exon_len<=300, .N  , by= c("exon_len", "G4_bol") ]
human.exons.upstream.G4s.stats[, Total:=sum(N), by=exon_len]
human.exons.upstream.G4s.stats[, Percentage:=N*100/Total]

human.exons.upstream.G4s$G4_bol <- factor(human.exons.upstream.G4s$G4_bol, levels=c("Without G-quadruplex", "With G-quadruplex"))

human.exons.upstream.G4s.300 <- human.exons.upstream.G4s[exon_len<=300]

human.exons.upstream.G4s.300$bin <- cut(human.exons.upstream.G4s[exon_len<=150]$exon_len, breaks=60, labels= seq(5, 300, 5), include.lowest=TRUE )

human.exons.upstream.G4s.300.stats <-human.exons.upstream.G4s.300[ exon_len<=150, .N  , by= c("bin", "G4_bol") ]
human.exons.upstream.G4s.300.stats[, Total:=sum(N), by=bin]
human.exons.upstream.G4s.300.stats[, Percentage:=N*100/Total]

labels= seq(-995, 1000, 5), include.lowest=TRUE )

ggplot(human.exons.upstream.G4s[filter=="In"]) +
  geom_bar(aes(x=exon_len, fill=G4_bol), position = "fill" ) +
  xlim(c(1, 150)) +
  geom_vline(xintercept = 30, linetype="dashed", colour="red") +
  ylim(c(0, 0.1))



ggplot(human.exons.upstream.G4s.300.stats[G4_bol=="With G-quadruplex"]) +
  geom_bar(aes(x=bin, y=Percentage), stat = "identity" ) 


```



```{r}
human.exons.downstream.G4s <- fread("../Exons_G4s/human.exons.tvs.downstream.G4seq_PDS")

colnames(human.exons.downstream.G4s) <- c( "chrom", "sstart", "send", "sstrand", "start", "end", "strand",  "gchrom", "gstart", "gend", "gscore", "exon_dist"  )


human.exons.downstream.G4s[ ,dist:=( (gstart + (gend- gstart)/2) - sstart) ]
human.exons.downstream.G4s[ sstrand=="-", dist:=(   sstart - (gstart + (gend- gstart)/2)) ]

human.exons.downstream.G4s[, exon_len:=end-start]

human.exons.downstream.G4s[ , G4_bol:="Without G-quadruplex"]
human.exons.downstream.G4s[ dist<=100  & dist>=0 , G4_bol:="With G-quadruplex"]


human.exons.downstream.G4s[, exon_class:="exon"]
human.exons.downstream.G4s[ exon_len<=30, exon_class:="microexon"]


human.exons.downstream.G4s.2x2 <- matrix(ncol = 2,
c(nrow(human.exons.downstream.G4s[ exon_class=="exon" & G4_bol=="Without G-quadruplex"]),
nrow(human.exons.downstream.G4s[ exon_class=="exon" & G4_bol=="With G-quadruplex"]),
nrow(human.exons.downstream.G4s[ exon_class=="microexon" & G4_bol=="Without G-quadruplex"]),
nrow(human.exons.downstream.G4s[ exon_class=="microexon" & G4_bol=="With G-quadruplex"]) ))


chisq.test(human.exons.downstream.G4s.2x2)






human.exons.downstream.G4s[ , ID:=paste(chrom,   start,     end, sep="_" )]

human.exons.downstream.G4s[, filter:="Out"]
human.exons.downstream.G4s[ ID %in%  KCL_human_CE,  filter:="In"]

human.exons.downstream.G4s <- human.exons.downstream.G4s[ ID %in%  KCL_human_CE, ]

human.exons.downstream.G4s.stats <-human.exons.downstream.G4s[ exon_len<=300, .N  , by= c("exon_len", "G4_bol") ]
human.exons.downstream.G4s.stats[, Total:=sum(N), by=exon_len]
human.exons.downstream.G4s.stats[, Percentage:=N*100/Total]

human.exons.downstream.G4s$G4_bol <- factor(human.exons.downstream.G4s$G4_bol, levels=c("Without G-quadruplex", "With G-quadruplex"))



ggplot(human.exons.downstream.G4s) +
  geom_bar(aes(x=exon_len, fill=G4_bol), position = "fill" ) +
  xlim(c(1, 300)) +
  geom_vline(xintercept = 30, linetype="dashed", colour="red") +
  ylim(c(0, 0.1))



ggplot(human.exons.downstream.G4s.stats[G4_bol=="With G-quadruplex"]) +
  geom_bar(aes(x=exon_len, y=Percentage), stat = "identity" ) +
  xlim(c(1, 300)) +
  geom_vline(xintercept = 30, linetype="dashed", colour="red") 

```



```{r}
ggplot(human.exons.upstream.G4s) +
  geom_bar(aes(x=exon_len, fill=G4_bol), position = "fill" ) +
  xlim(c(1, 150)) +
  geom_vline(xintercept = 30) +
  ylim(c(0.8, 1))
```

```{r}

ggplot(hg19_ME_table_G4s) + 
  geom_boxplot(aes(G4_bol, exon_len)) +
  ylim(c(0, 300))


hg19_ME_table_K[, exon_len:=end-start]

ggplot(hg19_ME_table_K) + 
  geom_boxplot(aes(G4_bol, exon_len)) +
  ylim(c(0, 300))

hg19_ME_table_G4s[, median(exon_len), by=G4_bol]
hg19_ME_table_K[, median(exon_len), by=G4_bol]

ggplot(hg19_ME_table_PDS) + 
  geom_boxplot(aes(G4_bol, exon_len)) +
  ylim(c(0, 300))

```




```{r}

human.ME_g4.stats <- data.table()

hg19_ME_table_G4s <- G4_ME_table("../Exons_G4s/human.exons.tvs.upstream.G4s" , "../Exons_G4s/human.exons.tvs.downstream.G4s", 100 )



hg19_ME_table_G4s.2x2 <- matrix(ncol = 2,
c(nrow(hg19_ME_table_G4s[ exon_class=="exon" & G4_bol=="Without G-quadruplex"]),
nrow(hg19_ME_table_G4s[ exon_class=="exon" & G4_bol=="With G-quadruplex"]),
nrow(hg19_ME_table_G4s[ exon_class=="microexon" & G4_bol=="Without G-quadruplex"]),
nrow(hg19_ME_table_G4s[ exon_class=="microexon" & G4_bol=="With G-quadruplex"]) ))

hg19_ME_table_G4s[ , exon_len:=end-start]


colnames(hg19_ME_table_G4s.2x2) <- c("exon", "microexon")
rownames(hg19_ME_table_G4s.2x2) <- c("Without G-quadruplex", "With G-quadruplex")


human.ME_g4.stats <- rbind(human.ME_g4.stats,
                           cbind(
                             "motif",
                             chisq.test( hg19_ME_table_G4s.2x2)$p.value,
                             (hg19_ME_table_G4s.2x2[ 2 , 2 ]/hg19_ME_table_G4s.2x2[ 1 , 2 ]) / (hg19_ME_table_G4s.2x2[ 2 , 1 ]/hg19_ME_table_G4s.2x2[ 1 , 1 ])
                             ))


hg19_ME_table_K <- G4_ME_table("../Exons_G4s/human.exons.tvs.upstream.G4seq_K" , "../Exons_G4s/human.exons.tvs.downstream.G4seq_K", 100 )



hg19_ME_table_K.2x2 <- matrix(ncol = 2,
c(nrow(hg19_ME_table_K[ exon_class=="exon" & G4_bol=="Without G-quadruplex"]),
nrow(hg19_ME_table_K[ exon_class=="exon" & G4_bol=="With G-quadruplex"]),
nrow(hg19_ME_table_K[ exon_class=="microexon" & G4_bol=="Without G-quadruplex"]),
nrow(hg19_ME_table_K[ exon_class=="microexon" & G4_bol=="With G-quadruplex"]) ))

chisq.test( hg19_ME_table_K.2x2)


human.ME_g4.stats <- rbind(human.ME_g4.stats,
                           cbind(
                             "K+",
                             chisq.test( hg19_ME_table_K.2x2)$p.value,
                             (hg19_ME_table_K.2x2[ 2 , 2 ]/hg19_ME_table_K.2x2[ 1 , 2 ]) / (hg19_ME_table_K.2x2[ 2 , 1 ]/hg19_ME_table_K.2x2[ 1 , 1 ])
                             ))


hg19_ME_table_PDS <- G4_ME_table("../Exons_G4s/human.exons.tvs.upstream.G4seq_PDS" , "../Exons_G4s/human.exons.tvs.downstream.G4seq_PDS", 100 )


hg19_ME_table_PDS.2x2 <- matrix(ncol = 2,
c(nrow(hg19_ME_table_PDS[ exon_class=="exon" & G4_bol=="Without G-quadruplex"]),
nrow(hg19_ME_table_PDS[ exon_class=="exon" & G4_bol=="With G-quadruplex"]),
nrow(hg19_ME_table_PDS[ exon_class=="microexon" & G4_bol=="Without G-quadruplex"]),
nrow(hg19_ME_table_PDS[ exon_class=="microexon" & G4_bol=="With G-quadruplex"]) ))

chisq.test( hg19_ME_table_PDS.2x2)

human.ME_g4.stats <- rbind(human.ME_g4.stats,
                           cbind(
                             "PDS",
                             chisq.test( hg19_ME_table_PDS.2x2)$p.value,
                             (hg19_ME_table_PDS.2x2[ 2 , 2 ]/hg19_ME_table_PDS.2x2[ 1 , 2 ]) / (hg19_ME_table_PDS.2x2[ 2 , 1 ]/hg19_ME_table_PDS.2x2[ 1 , 1 ])
                             ))

```


```{r}
colnames(human.ME_g4.stats) <- c("G4_signal", "p.value", "OR") 

human.ME_g4.stats[, P_value.Bonferroni:=as.numeric(p.value)*3]


ggplot(human.ME_g4.stats) +
  geom_point(aes(G4_signal, log2(as.numeric(OR)), size = -log10(P_value.Bonferroni), shape=P_value.Bonferroni<0.05 ) ) +
  scale_colour_gradient2(low="blue", high="red", mid = "grey", midpoint = 0) +
  theme_bw() +
      theme(legend.position = "top", legend.direction = "horizontal") 
```




```{r}
primary_cortical_neuron_DIV4_CD1_ME_table <- diff_G4_ME_table("../KCL/MicroExonator/closest/primary_cortical_neuron_DIV4_CD1.diff.upstream.G4s.fix" , "../KCL/MicroExonator/closest/primary_cortical_neuron_DIV4_CD1.diff.downstream.G4s.fix", 100 )


ESC_derived_neuron_CD1_ME_table <- diff_G4_ME_table("../KCL/MicroExonator/closest/ESC-derived_neuron_CD1.diff.upstream.G4s.fix" , "../KCL/MicroExonator/closest/ESC-derived_neuron_CD1.diff.downstream.G4s.fix", 100 )

primary_cortical_neuron_DIV10_CD1_ME_table <- diff_G4_ME_table("../KCL/MicroExonator/closest/primary_cortical_neuron_DIV10_CD1.diff.upstream.G4s.fix" , "../KCL/MicroExonator/closest/primary_cortical_neuron_DIV10_CD1.diff.downstream.G4s.fix", 100 )

primary_cortical_neuron_DIV10_Tc1_ME_table <- diff_G4_ME_table("../KCL/MicroExonator/closest/primary_cortical_neuron_DIV10_Tc1.diff.upstream.G4s.fix" , "../KCL/MicroExonator/closest/primary_cortical_neuron_DIV10_Tc1.diff.downstream.G4s.fix", 100 )




 ggplot(data=primary_cortical_neuron_DIV4_CD1_ME_table[ Type=="CE",]) +
  geom_point( aes(x=-DeltaPsi, y=Probability), colour="grey", alpha=alpha.var, size=size.var) +
  geom_point(data=KCL_human[Type=="CE" & DeltaPsi>0.1 & Probability>0.9, ], aes(x=-DeltaPsi, y=Probability), colour="skyblue", alpha=alpha.var, size=size.var) +
   geom_point(data=KCL_human[Type=="CE" & DeltaPsi<(-0.1) & Probability>0.9, ], aes(x=-DeltaPsi, y=Probability), colour="red", alpha=alpha.var, size=size.var) +
  xlim(c(-0.8, 0.8)) +
  xlab("DeltaPSI (Control - KCl)") +
  theme_bw()

```



```{r}


 ggplot(data=KCL_human[ Type=="CE",]) +
  geom_point( aes(x=-DeltaPsi, y=Probability), colour="grey", alpha=alpha.var, size=size.var) +
  geom_point(data=KCL_human[Type=="CE" & DeltaPsi>0.1 & Probability>0.9, ], aes(x=-DeltaPsi, y=Probability), colour="skyblue", alpha=alpha.var, size=size.var) +
   geom_point(data=KCL_human[Type=="CE" & DeltaPsi<(-0.1) & Probability>0.9, ], aes(x=-DeltaPsi, y=Probability), colour="red", alpha=alpha.var, size=size.var) +
  xlim(c(-0.8, 0.8)) +
  xlab("DeltaPSI (Control - KCl)") +
  theme_bw()



 ggplot(data=primary_cortical_neuron_DIV4_CD1_K_ME_table[ Type=="CE",]) +
  geom_point( aes(x=-DeltaPsi, y=Probability), colour="grey", alpha=alpha.var, size=size.var) +
  geom_point(data=KCL_human[Type=="CE" & DeltaPsi>0.1 & Probability>0.9, ], aes(x=-DeltaPsi, y=Probability), colour="skyblue", alpha=alpha.var, size=size.var) +
   geom_point(data=KCL_human[Type=="CE" & DeltaPsi<(-0.1) & Probability>0.9, ], aes(x=-DeltaPsi, y=Probability), colour="red", alpha=alpha.var, size=size.var) +
  xlim(c(-0.8, 0.8)) +
  xlab("DeltaPSI (Control - KCl)") +
  theme_bw()

 
 
 
 
KCL_human[, Experiment:="Human"] 
primary_cortical_neuron_DIV4_CD1_ME_table[, Experiment:="DIV4_CD1"]
ESC_derived_neuron_CD1_ME_table[, Experiment:="ESC_CD1"]
primary_cortical_neuron_DIV10_CD1_ME_table[, Experiment:="DIV10_CD1"]
primary_cortical_neuron_DIV10_Tc1_ME_table[, Experiment:="DIV10_Tc1"]



TOTAL.KCL_diff <- rbind(KCL_human[, c("Type", "DeltaPsi", "Probability", "Experiment")],
                        primary_cortical_neuron_DIV4_CD1_ME_table[, c("Type", "DeltaPsi", "Probability", "Experiment")],
                        ESC_derived_neuron_CD1_ME_table[, c("Type", "DeltaPsi", "Probability", "Experiment")],
                        primary_cortical_neuron_DIV10_CD1_ME_table[, c("Type" , "DeltaPsi", "Probability", "Experiment")],
                        primary_cortical_neuron_DIV10_Tc1_ME_table[, c("Type", "DeltaPsi", "Probability", "Experiment")] )




TOTAL.KCL_diff$Experiment <- factor(TOTAL.KCL_diff$Experiment, levels=c("Human", "ESC_CD1", "DIV4_CD1", "DIV10_CD1", "DIV10_Tc1"))



 ggplot(data=TOTAL.KCL_diff[ Type=="CE",]) +
  geom_point( aes(x=-DeltaPsi, y=Probability), colour="grey", alpha=alpha.var, size=size.var) +
  geom_point(data=TOTAL.KCL_diff[Type=="CE" & DeltaPsi>0.1 & Probability>0.9, ], aes(x=-DeltaPsi, y=Probability), colour="skyblue", alpha=alpha.var, size=size.var) +
   geom_point(data=TOTAL.KCL_diff[Type=="CE" & DeltaPsi<(-0.1) & Probability>0.9, ], aes(x=-DeltaPsi, y=Probability), colour="red", alpha=alpha.var, size=size.var) + 
   facet_grid(. ~ Experiment ) +
  xlim(c(-0.8, 0.8)) +
  xlab("DeltaPSI (Control - KCl)") +
  theme_bw()


KCL_human.TOTAL_and_ME_stats[, Experiment:="Human"]
KCL_DIV10_Tc1.TOTAL_and_ME_stats[, Experiment:="DIV10_Tc1"]
KCL_ESC_CD1.TOTAL_and_ME_stats[, Experiment:="ESC_CD1"]
KCL_DIV4_CD1.TOTAL_and_ME_stats[, Experiment:="DIV4_CD1"]
KCL_DIV10_CD1.TOTAL_and_ME_stats[, Experiment:="DIV10_CD1"]

```





```{r}

primary_cortical_neuron_DIV4_CD1_K_ME_table <- diff_G4_ME_table("../KCL/MicroExonator/closest/primary_cortical_neuron_DIV4_CD1.diff.upstream.G4seq_K" , "../KCL/MicroExonator/closest/primary_cortical_neuron_DIV4_CD1.diff.downstream.G4seq_K", 100 )

ESC_derived_neuron_CD1_K_ME_table <- diff_G4_ME_table("../KCL/MicroExonator/closest/ESC-derived_neuron_CD1.diff.upstream.G4seq_K" , "../KCL/MicroExonator/closest/ESC-derived_neuron_CD1.diff.downstream.G4seq_K", 100 )

primary_cortical_neuron_DIV10_CD1_K_ME_table <- diff_G4_ME_table("../KCL/MicroExonator/closest/primary_cortical_neuron_DIV10_CD1.diff.upstream.G4seq_K" , "../KCL/MicroExonator/closest/primary_cortical_neuron_DIV10_CD1.diff.downstream.G4seq_K", 100 )

primary_cortical_neuron_DIV10_Tc1_K_ME_table <- diff_G4_ME_table("../KCL/MicroExonator/closest/primary_cortical_neuron_DIV10_Tc1.diff.upstream.G4seq_K" , "../KCL/MicroExonator/closest/primary_cortical_neuron_DIV10_Tc1.diff.downstream.G4seq_K", 100 )



total_mouse_K_ME_table <- unique(rbind(primary_cortical_neuron_DIV4_CD1_K_ME_table[ , c("chrom" ,"start", "end", "strand", "Type", "G4_bol", "exon_class")],
ESC_derived_neuron_CD1_K_ME_table[ , c("chrom" ,"start", "end", "strand", "Type", "G4_bol", "exon_class")],
primary_cortical_neuron_DIV10_CD1_K_ME_table[ , c("chrom" ,"start", "end", "strand", "Type", "G4_bol", "exon_class")],
primary_cortical_neuron_DIV10_Tc1_K_ME_table[ , c("chrom" ,"start", "end", "strand", "Type", "G4_bol", "exon_class")]))


total_mouse_K_ME_table.2x2 <- matrix(ncol = 2,
c(nrow(total_mouse_K_ME_table[ exon_class=="exon" & G4_bol=="Without G-quadruplex"]),
nrow(total_mouse_K_ME_table[ exon_class=="exon" & G4_bol=="With G-quadruplex"]),
nrow(total_mouse_K_ME_table[ exon_class=="microexon" & G4_bol=="Without G-quadruplex"]),
nrow(total_mouse_K_ME_table[ exon_class=="microexon" & G4_bol=="With G-quadruplex"]) ))

chisq.test( total_mouse_K_ME_table.2x2)


colnames(total_mouse_K_ME_table.2x2) <- c("exon", "microexon")
rownames(total_mouse_K_ME_table.2x2) <- c("Without G-quadruplex", "With G-quadruplex")


primary_cortical_neuron_DIV4_CD1_PDS_ME_table <- diff_G4_ME_table("../KCL/MicroExonator/closest/primary_cortical_neuron_DIV4_CD1.diff.upstream.G4seq_PDS" , "../KCL/MicroExonator/closest/primary_cortical_neuron_DIV4_CD1.diff.downstream.G4seq_PDS", 100 )

ESC_derived_neuron_CD1_PDS_ME_table <- diff_G4_ME_table("../KCL/MicroExonator/closest/ESC-derived_neuron_CD1.diff.upstream.G4seq_PDS" , "../KCL/MicroExonator/closest/ESC-derived_neuron_CD1.diff.downstream.G4seq_PDS", 100 )

primary_cortical_neuron_DIV10_CD1_PDS_ME_table <- diff_G4_ME_table("../KCL/MicroExonator/closest/primary_cortical_neuron_DIV10_CD1.diff.upstream.G4seq_PDS" , "../KCL/MicroExonator/closest/primary_cortical_neuron_DIV10_CD1.diff.downstream.G4seq_PDS", 100 )

primary_cortical_neuron_DIV10_Tc1_PDS_ME_table <- diff_G4_ME_table("../KCL/MicroExonator/closest/primary_cortical_neuron_DIV10_Tc1.diff.upstream.G4seq_PDS" , "../KCL/MicroExonator/closest/primary_cortical_neuron_DIV10_Tc1.diff.downstream.G4seq_PDS", 100 )



total_mouse_PDS_ME_table <- unique(rbind(primary_cortical_neuron_DIV4_CD1_PDS_ME_table[ , c("chrom" ,"start", "end", "strand", "Type", "G4_bol", "exon_class")],
ESC_derived_neuron_CD1_PDS_ME_table[ , c("chrom" ,"start", "end", "strand", "Type", "G4_bol", "exon_class")],
primary_cortical_neuron_DIV10_CD1_PDS_ME_table[ , c("chrom" ,"start", "end", "strand", "Type", "G4_bol", "exon_class")],
primary_cortical_neuron_DIV10_Tc1_PDS_ME_table[ , c("chrom" ,"start", "end", "strand", "Type", "G4_bol", "exon_class")]))


total_mouse_PDS_ME_table.2x2 <- matrix(ncol = 2,
c(nrow(total_mouse_PDS_ME_table[ exon_class=="exon" & G4_bol=="Without G-quadruplex"]),
nrow(total_mouse_PDS_ME_table[ exon_class=="exon" & G4_bol=="With G-quadruplex"]),
nrow(total_mouse_PDS_ME_table[ exon_class=="microexon" & G4_bol=="Without G-quadruplex"]),
nrow(total_mouse_PDS_ME_table[ exon_class=="microexon" & G4_bol=="With G-quadruplex"]) ))

chisq.test( total_mouse_PDS_ME_table.2x2)
```


```{r}

KCL_human[, ID:=paste( chrom, strand, start, end  , sep="_")]



primary_cortical_neuron_DIV4_CD1_motif_ME_table <- diff_G4_ME_table("../KCL/MicroExonator/closest/primary_cortical_neuron_DIV4_CD1.diff.upstream.G4s.fix" , "../KCL/MicroExonator/closest/primary_cortical_neuron_DIV4_CD1.diff.downstream.G4s.fix", 100 )
ESC_derived_neuron_CD1_motif_ME_table <- diff_G4_ME_table("../KCL/MicroExonator/closest/ESC-derived_neuron_CD1.diff.upstream.G4s.fix" , "../KCL/MicroExonator/closest/ESC-derived_neuron_CD1.diff.downstream.G4s.fix", 100 )
primary_cortical_neuron_DIV10_CD1_motif_ME_table <- diff_G4_ME_table("../KCL/MicroExonator/closest/primary_cortical_neuron_DIV10_CD1.diff.upstream.G4s.fix" , "../KCL/MicroExonator/closest/primary_cortical_neuron_DIV10_CD1.diff.downstream.G4s.fix", 100 )
primary_cortical_neuron_DIV10_Tc1_motif_ME_table <- diff_G4_ME_table("../KCL/MicroExonator/closest/primary_cortical_neuron_DIV10_Tc1.diff.upstream.G4s.fix" , "../KCL/MicroExonator/closest/primary_cortical_neuron_DIV10_Tc1.diff.downstream.G4s.fix", 100 )


primary_cortical_neuron_DIV4_CD1_motif_ME_table[, ID:=paste( chrom, strand, start, end  , sep="_")]
ESC_derived_neuron_CD1_motif_ME_table[, ID:=paste( chrom, strand, start, end  , sep="_")]
primary_cortical_neuron_DIV10_CD1_motif_ME_table[, ID:=paste( chrom, strand, start, end  , sep="_")]
primary_cortical_neuron_DIV10_Tc1_motif_ME_table[, ID:=paste( chrom, strand, start, end  , sep="_")]

primary_cortical_neuron_DIV4_CD1_K_ME_table[, ID:=paste( chrom, strand, start, end  , sep="_")]
ESC_derived_neuron_CD1_K_ME_table[, ID:=paste( chrom, strand, start, end  , sep="_")]
primary_cortical_neuron_DIV10_CD1_K_ME_table[, ID:=paste( chrom, strand, start, end  , sep="_")]
primary_cortical_neuron_DIV10_Tc1_K_ME_table[, ID:=paste( chrom, strand, start, end  , sep="_")]


primary_cortical_neuron_DIV4_CD1_PDS_ME_table[, ID:=paste( chrom, strand, start, end  , sep="_")]
ESC_derived_neuron_CD1_PDS_ME_table[, ID:=paste( chrom, strand, start, end  , sep="_")]
primary_cortical_neuron_DIV10_CD1_PDS_ME_table[, ID:=paste( chrom, strand, start, end  , sep="_")]
primary_cortical_neuron_DIV10_Tc1_PDS_ME_table[, ID:=paste( chrom, strand, start, end  , sep="_")]


All_mouse <- data.table(unique(c(primary_cortical_neuron_DIV4_CD1_K_ME_table[G4_bol=="With G-quadruplex", ID],
ESC_derived_neuron_CD1_K_ME_table[G4_bol=="With G-quadruplex", ID],
primary_cortical_neuron_DIV10_CD1_K_ME_table[G4_bol=="With G-quadruplex", ID],
primary_cortical_neuron_DIV10_Tc1_K_ME_table[G4_bol=="With G-quadruplex", ID],
primary_cortical_neuron_DIV4_CD1_PDS_ME_table[G4_bol=="With G-quadruplex", ID],
ESC_derived_neuron_CD1_PDS_ME_table[G4_bol=="With G-quadruplex", ID],
primary_cortical_neuron_DIV10_CD1_PDS_ME_table[G4_bol=="With G-quadruplex", ID],
primary_cortical_neuron_DIV10_Tc1_PDS_ME_table[G4_bol=="With G-quadruplex", ID],
primary_cortical_neuron_DIV4_CD1_motif_ME_table[G4_bol=="With G-quadruplex", ID],
ESC_derived_neuron_CD1_motif_ME_table[G4_bol=="With G-quadruplex", ID],
primary_cortical_neuron_DIV10_CD1_motif_ME_table[G4_bol=="With G-quadruplex", ID],
primary_cortical_neuron_DIV10_Tc1_motif_ME_table[G4_bol=="With G-quadruplex", ID])))


colnames(All_mouse) <- "ID"





All_mouse[ ID %in% primary_cortical_neuron_DIV4_CD1_motif_ME_table[abs(DeltaPsi)>=0.1 & Probability>=0.9 & G4_bol=="With G-quadruplex", ID ], primary_cortical_neuron_DIV4_CD1_motif:=TRUE   ]
All_mouse[ ID %in% ESC_derived_neuron_CD1_motif_ME_table[abs(DeltaPsi)>=0.1 & Probability>=0.9 & G4_bol=="With G-quadruplex", ID ], ESC_derived_neuron_CD1_motif:=TRUE   ]
All_mouse[ ID %in% primary_cortical_neuron_DIV10_CD1_motif_ME_table[abs(DeltaPsi)>=0.1 & Probability>=0.9 & G4_bol=="With G-quadruplex", ID ], primary_cortical_neuron_DIV10_CD1_motif:=TRUE   ]
All_mouse[ ID %in% primary_cortical_neuron_DIV10_Tc1_motif_ME_table[abs(DeltaPsi)>=0.1 & Probability>=0.9 & G4_bol=="With G-quadruplex", ID ], primary_cortical_neuron_DIV10_Tc1_motif:=TRUE   ]

All_mouse[ ID %in% primary_cortical_neuron_DIV4_CD1_K_ME_table[abs(DeltaPsi)>=0.1 & Probability>=0.9 & G4_bol=="With G-quadruplex", ID ], primary_cortical_neuron_DIV4_CD1_K:=TRUE   ]
All_mouse[ ID %in% ESC_derived_neuron_CD1_K_ME_table[abs(DeltaPsi)>=0.1 & Probability>=0.9 & G4_bol=="With G-quadruplex", ID ], ESC_derived_neuron_CD1_K:=TRUE   ]
All_mouse[ ID %in% primary_cortical_neuron_DIV10_CD1_K_ME_table[abs(DeltaPsi)>=0.1 & Probability>=0.9 & G4_bol=="With G-quadruplex", ID ], primary_cortical_neuron_DIV10_CD1_K:=TRUE   ]
All_mouse[ ID %in% primary_cortical_neuron_DIV10_Tc1_K_ME_table[abs(DeltaPsi)>=0.1 & Probability>=0.9 & G4_bol=="With G-quadruplex", ID ], primary_cortical_neuron_DIV10_Tc1_K:=TRUE   ]


All_mouse[ ID %in% primary_cortical_neuron_DIV4_CD1_PDS_ME_table[abs(DeltaPsi)>=0.1 & Probability>=0.9 & G4_bol=="With G-quadruplex", ID ], primary_cortical_neuron_DIV4_CD1_PDS:=TRUE   ]
All_mouse[ ID %in% ESC_derived_neuron_CD1_PDS_ME_table[abs(DeltaPsi)>=0.1 & Probability>=0.9 & G4_bol=="With G-quadruplex", ID ], ESC_derived_neuron_CD1_PDS:=TRUE   ]
All_mouse[ ID %in% primary_cortical_neuron_DIV10_CD1_PDS_ME_table[abs(DeltaPsi)>=0.1 & Probability>=0.9 & G4_bol=="With G-quadruplex", ID ], primary_cortical_neuron_DIV10_CD1_PDS:=TRUE   ]
All_mouse[ ID %in% primary_cortical_neuron_DIV10_Tc1_PDS_ME_table[abs(DeltaPsi)>=0.1 & Probability>=0.9 & G4_bol=="With G-quadruplex", ID ], primary_cortical_neuron_DIV10_Tc1_PDS:=TRUE   ]


All_mouse <- cbind(All_mouse, rowSums(is.na(All_mouse)))


fwrite(All_mouse[order(V2)], "../potassium_experiments/Total_mouse.txt", na="FALSE", sep = "\t", quote=F)

       

       

human_motif_ME_table <- diff_G4_ME_table("../KCL/MicroExonator/closest/Control_vs_KCL.diff.upstream.G4s" , "../KCL/MicroExonator/closest/Control_vs_KCL.diff.downstream.G4s", 100 )   
human_K_ME_table <- diff_G4_ME_table("../KCL/MicroExonator/closest/Control_vs_KCL.diff.upstream.G4seq_K" , "../KCL/MicroExonator/closest/Control_vs_KCL.diff.downstream.G4seq_K", 100 )
human_PDS_ME_table <- diff_G4_ME_table("../KCL/MicroExonator/closest/Control_vs_KCL.diff.upstream.G4seq_PDS" , "../KCL/MicroExonator/closest/Control_vs_KCL.diff.downstream.G4seq_PDS", 100 )   
     
human_motif_ME_table[, ID:=paste( chrom, strand, start, end  , sep="_")]
human_K_ME_table[, ID:=paste( chrom, strand, start, end  , sep="_")]
human_PDS_ME_table[, ID:=paste( chrom, strand, start, end  , sep="_")]

All_human<- data.table(unique(c(         
human_motif_ME_table[G4_bol=="With G-quadruplex", ID],       
human_K_ME_table[G4_bol=="With G-quadruplex", ID],
human_PDS_ME_table[G4_bol=="With G-quadruplex", ID])))

colnames(All_human) <- "ID"



All_human[ ID %in% human_motif_ME_table[abs(DeltaPsi)>=0.1 & Probability>=0.9 & G4_bol=="With G-quadruplex", ID ], diff_motif:=TRUE   ]
All_human[ ID %in% human_K_ME_table[abs(DeltaPsi)>=0.1 & Probability>=0.9 & G4_bol=="With G-quadruplex", ID ], diff_K:=TRUE   ]
All_human[ ID %in% human_PDS_ME_table[abs(DeltaPsi)>=0.1 & Probability>=0.9 & G4_bol=="With G-quadruplex", ID ], diff_:=TRUE   ]



All_human <- cbind(All_human, rowSums(is.na(All_human)))


fwrite(All_human[order(V2)], "../potassium_experiments/Total_human.txt",  na="FALSE", sep = "\t", quote=F)





```


## Not working

```{r}
human_mouse <- fread("../potassium_experiments/human_mouse.0.2.exons")

colnames(human_mouse) <- c("hID", "mID", "score_index")


All_human.mouse <- merge(All_human, human_mouse, by.x="ID", by.y="hID")


All_human.mouse <- merge(All_human.mouse, All_mouse, by.x = "mID", by.y="ID")





All_human.mouse.diff <-  All_human.mouse[V2.y<12 & V2.x<3 ]


All_human.mouse.diff





KCL_human_CE[, ID:=paste( chrom, strand, start, end  , sep="_")]


All_human.mouse.diff.gene <- merge(All_human.mouse.diff, KCL_human_CE, by="ID")


View(All_human.mouse.diff.gene[order(V2.y, V2.x  )][, c("ID", "mID", "V2.x", "V2.y", "wikigene_description", "hgnc_symbol")] )


cat(All_human.mouse.diff.gene[, hgnc_symbol], sep="\n")

fwrite(All_human.mouse.diff.gene[order(V2.y, V2.x  )][, c("ID", "V2.x", "V2.y", "wikigene_description", "hgnc_symbol")], "../potassium_experiments/human.mouse.diff.G4.gene.txt",  na="FALSE", sep = "\t", quote=F)

```



```{r}
KCL_human[ID=""]
```



```{r}

total_mouse_PDS_ME_table[G4_bol=="With G-quadruplex", mean(exon_len) ]
total_mouse_PDS_ME_table[G4_bol=="Without G-quadruplex",  mean(exon_len) ]

total_mouse_PDS_ME_table[, exon_len:=end-start]

wilcox.test(  exon_len ~ G4_bol, data=total_mouse_PDS_ME_table )



```



```{r}
primary_cortical_neuron_DIV4_CD1_K_ME_table <- diff_G4_ME_table("../KCL/MicroExonator/closest/primary_cortical_neuron_DIV4_CD1.diff.upstream.G4seq_K" , "../KCL/MicroExonator/closest/primary_cortical_neuron_DIV4_CD1.diff.downstream.G4seq_K", 100 )


primary_cortical_neuron_DIV4_CD1_K_ME_table.2x2 <- matrix(ncol = 2,
c(nrow(primary_cortical_neuron_DIV4_CD1_K_ME_table[ exon_class=="exon" & G4_bol=="Without G-quadruplex"]),
nrow(primary_cortical_neuron_DIV4_CD1_K_ME_table[ exon_class=="exon" & G4_bol=="With G-quadruplex"]),
nrow(primary_cortical_neuron_DIV4_CD1_K_ME_table[ exon_class=="microexon" & G4_bol=="Without G-quadruplex"]),
nrow(primary_cortical_neuron_DIV4_CD1_K_ME_table[ exon_class=="microexon" & G4_bol=="With G-quadruplex"]) ))

chisq.test( primary_cortical_neuron_DIV4_CD1_K_ME_table.2x2)


primary_cortical_neuron_DIV4_CD1_PDS_ME_table <- diff_G4_ME_table("../KCL/MicroExonator/closest/primary_cortical_neuron_DIV4_CD1.diff.upstream.G4seq_PDS" , "../KCL/MicroExonator/closest/primary_cortical_neuron_DIV4_CD1.diff.downstream.G4seq_PDS", 100 )



primary_cortical_neuron_DIV4_CD1_PDS_ME_table.2x2 <- matrix(ncol = 2,
c(nrow(primary_cortical_neuron_DIV4_CD1_PDS_ME_table[ exon_class=="exon" & G4_bol=="Without G-quadruplex"]),
nrow(primary_cortical_neuron_DIV4_CD1_PDS_ME_table[ exon_class=="exon" & G4_bol=="With G-quadruplex"]),
nrow(primary_cortical_neuron_DIV4_CD1_PDS_ME_table[ exon_class=="microexon" & G4_bol=="Without G-quadruplex"]),
nrow(primary_cortical_neuron_DIV4_CD1_PDS_ME_table[ exon_class=="microexon" & G4_bol=="With G-quadruplex"]) ))

chisq.test( primary_cortical_neuron_DIV4_CD1_PDS_ME_table.2x2)



Control_vs_KCL_K_ME_table <- diff_G4_ME_table("../KCL/MicroExonator/closest/Control_vs_KCL.diff.upstream.G4seq_K" , "../KCL/MicroExonator/closest/Control_vs_KCL.diff.downstream.G4seq_K", 100 )


Control_vs_KCL_K_ME_table.2x2 <- matrix(ncol = 2,
c(nrow(Control_vs_KCL_K_ME_table[ exon_class=="exon" & G4_bol=="Without G-quadruplex"]),
nrow(Control_vs_KCL_K_ME_table[ exon_class=="exon" & G4_bol=="With G-quadruplex"]),
nrow(Control_vs_KCL_K_ME_table[ exon_class=="microexon" & G4_bol=="Without G-quadruplex"]),
nrow(Control_vs_KCL_K_ME_table[ exon_class=="microexon" & G4_bol=="With G-quadruplex"]) ))

chisq.test( Control_vs_KCL_K_ME_table.2x2)


Control_vs_KCL_PDS_ME_table <- diff_G4_ME_table("../KCL/MicroExonator/closest/Control_vs_KCL.diff.upstream.G4seq_PDS" , "../KCL/MicroExonator/closest/Control_vs_KCL.diff.downstream.G4seq_K", 100 )


Control_vs_KCL_PDS_ME_table.2x2 <- matrix(ncol = 2,
c(nrow(Control_vs_KCL_PDS_ME_table[ exon_class=="exon" & G4_bol=="Without G-quadruplex"]),
nrow(Control_vs_KCL_PDS_ME_table[ exon_class=="exon" & G4_bol=="With G-quadruplex"]),
nrow(Control_vs_KCL_PDS_ME_table[ exon_class=="microexon" & G4_bol=="Without G-quadruplex"]),
nrow(Control_vs_KCL_PDS_ME_table[ exon_class=="microexon" & G4_bol=="With G-quadruplex"]) ))

chisq.test( Control_vs_KCL_PDS_ME_table.2x2)
```


```{r}
splice_node_dot_plot3 <- function(upstream.path, downstream.path, dist_lim ) {

KCL_x_upstream <- fread(upstream.path  )
colnames(KCL_x_upstream) <- c("w.chrom", "w.start", "w.end", "w.strand", "chrom", "start", "end", "strand", "Type", "Psi_A", "Psi_B", "DeltaPsi", "Probability", "gchrom", "gstart", "gend", "gscore", "exon_dist"  )

KCL_x_upstream[ ,dist:=abs(w.start - (gstart + (gend- gstart)/2)) ]


KCL_x_downstream<- fread(downstream.path  )
colnames(KCL_x_downstream) <- c("w.chrom", "w.start", "w.end", "w.strand", "chrom", "start", "end", "strand", "Type", "Psi_A", "Psi_B", "DeltaPsi", "Probability", "gchrom", "gstart", "gend", "gscore", "exon_dist"  )


KCL_x_downstream[ ,dist:=abs(w.start - (gstart + (gend- gstart)/2)) ]

KCL_x <-  merge(KCL_x_upstream[, c("chrom",  "start", "end", "strand",  "Type" ,  "DeltaPsi", "Probability", "dist") ],
                    KCL_x_downstream[, c("chrom",  "start", "end", "strand",  "Type" ,  "DeltaPsi", "Probability", "dist") ], 
                    by=c("chrom",  "start", "end", "strand",  "Type" ,  "DeltaPsi", "Probability") )


#KCL_x <-  KCL_x[end-start>30 || Type!="CE", ]

KCL_x[(dist.x<=dist_lim | dist.y<=dist_lim ), G4_bol:="With G-quadruplex"]
KCL_x[(dist.x>dist_lim & dist.y>dist_lim ), G4_bol:="Without G-quadruplex"]
KCL_x$G4_bol <- factor(KCL_x$G4_bol, levels=c("Without G-quadruplex", "With G-quadruplex") )


KCL_x[  , diff:="NA"]
KCL_x[ (DeltaPsi<=-0.1 & Probability>=0.90) , diff:="In"]
KCL_x[ (DeltaPsi>0.1 & Probability>=0.90) , diff:="Ex"]

KCL_x_stats <- merge(KCL_x[  G4_bol=="Without G-quadruplex",  .(WO_G4=.N) , by=c("Type", "diff")  ], 
KCL_x[  G4_bol=="With G-quadruplex",  .(W_G4=.N) , by=c("Type", "diff")  ],
by=c("Type", "diff"))



KCL_x_stats <- merge(KCL_x_stats[diff!="NA"],  KCL_x_stats[diff=="NA" , .(Type,  WO_G4_NA=WO_G4, W_G4_NA=W_G4) ], by="Type")

KCL_x_stats[, `:=`(WO_G4_fraq=WO_G4/WO_G4_NA, W_G4_fraq=W_G4/W_G4_NA)]

KCL_x_stats[ , OR := W_G4_fraq/WO_G4_fraq]


KCL_x_stats.chi <- data.table()

for ( i in unique(KCL_x_stats[Type!="AL"]$Type)){
  
matrix.ex <-  matrix(nrow=2, c(KCL_x_stats[Type==i & diff=="Ex"]$WO_G4_NA, KCL_x_stats[Type==i & diff=="Ex"]$WO_G4, KCL_x_stats[Type==i & diff=="Ex"]$W_G4_NA, KCL_x_stats[Type==i & diff=="Ex"]$W_G4))
matrix.ex_res <- chisq.test(matrix.ex, correct=TRUE)

matrix.in <-  matrix(nrow=2, c(KCL_x_stats[Type==i & diff=="In"]$WO_G4_NA, KCL_x_stats[Type==i & diff=="In"]$WO_G4, KCL_x_stats[Type==i & diff=="In"]$W_G4_NA, KCL_x_stats[Type==i & diff=="In"]$W_G4))
matrix.in_res <- chisq.test(matrix.in, correct=TRUE)



KCL_x_stats.chi <- rbind(KCL_x_stats.chi, cbind(i, "Ex", matrix.ex_res$p.value)) 
KCL_x_stats.chi <- rbind(KCL_x_stats.chi, cbind(i, "In", matrix.in_res$p.value)) 
                                                        
}

colnames(KCL_x_stats.chi) <- c("Type", "diff", "P_value")

ntest = nrow(KCL_x_stats.chi)

KCL_x_stats.chi[ , `:=`(P_value.Bonferroni=as.numeric(P_value)*ntest) ]

KCL_x_stats <- merge(KCL_x_stats, KCL_x_stats.chi, by=c("Type", "diff"))

ggplot(KCL_x_stats) +
  geom_point(aes(Type, diff, size = -log(P_value.Bonferroni), shape=P_value.Bonferroni<0.05, colour=log2(OR) ) ) +
  scale_colour_gradient2(low="blue", high="red", mid = "grey", midpoint = 0) +
  theme_bw() +
      theme(legend.position = "top", legend.direction = "horizontal") 


return(KCL_x_stats)


}
```








```{r,  fig.height=4, fig.width=10}

human.dotplot <- splice_node_dot_plot2("../KCL/MicroExonator/Control_vs_KCL.diff.upstream_100nt.G4s",  "../KCL/MicroExonator/Control_vs_KCL.diff.downstream_100nt.G4s"  )



human.motif <- splice_node_dot_plot3("../KCL/MicroExonator/closest/Control_vs_KCL.diff.upstream.G4s",
                                        "../KCL/MicroExonator/closest/Control_vs_KCL.diff.downstream.G4s",
                                        100)




human.K <- splice_node_dot_plot3("../KCL/MicroExonator/closest/Control_vs_KCL.diff.upstream.G4seq_K",
                                        "../KCL/MicroExonator/closest/Control_vs_KCL.diff.downstream.G4seq_K",
                                        45)


human.PDS <- splice_node_dot_plot3("../KCL/MicroExonator/closest/Control_vs_KCL.diff.upstream.G4seq_PDS",
                                        "../KCL/MicroExonator/closest/Control_vs_KCL.diff.downstream.G4seq_PDS",
                                   45)


human.motif[ , G4_signal:="Motif"  ]
human.K[ , G4_signal:="K+"  ]
human.PDS[ , G4_signal:="PDS"  ]

human.motif_K_PDS <- rbind(human.motif, human.K, human.PDS)

human.motif_K_PDS$G4_signal <- factor(human.motif_K_PDS$G4_signal, levels = c("Motif", "K+", "PDS") )

Fig5.A1 <- ggplot(human.motif_K_PDS) +
  geom_point(aes(Type, diff, size = -log10(P_value.Bonferroni), shape=P_value.Bonferroni<0.05, colour=log2(OR) ) ) +
  scale_colour_gradient2(low="blue", high="red", mid = "grey", midpoint = 0) +
  facet_grid(G4_signal ~ .) +
  theme_bw() +
      theme(legend.position = "top", legend.direction = "horizontal")


Fig5.A1
```



```{r,  fig.height=4, fig.width=10}

supp.dotplot.A <- splice_node_dot_plot2("../KCL/MicroExonator/ESC-derived_neuron_CD1.diff.upstream_100nt.G4s",  "../KCL/MicroExonator/ESC-derived_neuron_CD1.diff.downstream_100nt.G4s"  )



ESC_derived_neuron_CD1.motif <- splice_node_dot_plot3("../KCL/MicroExonator/closest/ESC-derived_neuron_CD1.diff.upstream.G4s.fix",
                                        "../KCL/MicroExonator/closest/ESC-derived_neuron_CD1.diff.downstream.G4s.fix",
                                        100)


ESC_derived_neuron_CD1.K <- splice_node_dot_plot3("../KCL/MicroExonator/closest/ESC-derived_neuron_CD1.diff.upstream.G4seq_K",
                                        "../KCL/MicroExonator/closest/ESC-derived_neuron_CD1.diff.downstream.G4seq_K",
                                        45)


ESC_derived_neuron_CD1.PDS <- splice_node_dot_plot3("../KCL/MicroExonator/closest/ESC-derived_neuron_CD1.diff.upstream.G4seq_PDS",
                                        "../KCL/MicroExonator/closest/ESC-derived_neuron_CD1.diff.downstream.G4seq_PDS",
                                        45)




ESC_derived_neuron_CD1.motif[ , G4_signal:="Motif"  ]
ESC_derived_neuron_CD1.K[ , G4_signal:="K+"  ]
ESC_derived_neuron_CD1.PDS[ , G4_signal:="PDS"  ]

ESC_derived_neuron_CD1.motif_K_PDS <- rbind(ESC_derived_neuron_CD1.motif, ESC_derived_neuron_CD1.K, ESC_derived_neuron_CD1.PDS)

ESC_derived_neuron_CD1.motif_K_PDS$G4_signal <- factor(ESC_derived_neuron_CD1.motif_K_PDS$G4_signal, levels = c("Motif", "K+", "PDS") )

ggplot(ESC_derived_neuron_CD1.motif_K_PDS) +
  geom_point(aes(Type, diff, size = -log10(P_value.Bonferroni), shape=P_value.Bonferroni<0.05, colour=log2(OR) ) ) +
  scale_colour_gradient2(low="blue", high="red", mid = "grey", midpoint = 0) +
  facet_grid(G4_signal ~ .) +
  ylab("") +
  theme_bw() +
      theme(legend.position = "top", legend.direction = "horizontal") 

```




```{r,  fig.height=4, fig.width=10}

supp.dotplot.C <-  splice_node_dot_plot2("../KCL/MicroExonator/primary_cortical_neuron_DIV10_CD1.diff.upstream_100nt.G4s",  "../KCL/MicroExonator/primary_cortical_neuron_DIV10_CD1.diff.downstream_100nt.G4s"  )


primary_cortical_neuron_DIV10_CD1.motif <- splice_node_dot_plot3("../KCL/MicroExonator/closest/primary_cortical_neuron_DIV10_CD1.diff.upstream.G4s.fix",
                                        "../KCL/MicroExonator/closest/primary_cortical_neuron_DIV10_CD1.diff.downstream.G4s.fix",
                                        100)


primary_cortical_neuron_DIV10_CD1.K <- splice_node_dot_plot3("../KCL/MicroExonator/closest/primary_cortical_neuron_DIV10_CD1.diff.upstream.G4seq_K",
                                        "../KCL/MicroExonator/closest/primary_cortical_neuron_DIV10_CD1.diff.downstream.G4seq_K",
                                        45)


primary_cortical_neuron_DIV10_CD1.PDS <- splice_node_dot_plot3("../KCL/MicroExonator/closest/primary_cortical_neuron_DIV10_CD1.diff.upstream.G4seq_PDS",
                                        "../KCL/MicroExonator/closest/primary_cortical_neuron_DIV10_CD1.diff.downstream.G4seq_PDS",
                                        45)



primary_cortical_neuron_DIV10_CD1.motif[ , G4_signal:="Motif"  ]
primary_cortical_neuron_DIV10_CD1.K[ , G4_signal:="K+"  ]
primary_cortical_neuron_DIV10_CD1.PDS[ , G4_signal:="PDS"  ]

primary_cortical_neuron_DIV10_CD1.motif_K_PDS <- rbind(primary_cortical_neuron_DIV10_CD1.motif, primary_cortical_neuron_DIV10_CD1.K, primary_cortical_neuron_DIV10_CD1.PDS)

primary_cortical_neuron_DIV10_CD1.motif_K_PDS$G4_signal <- factor(primary_cortical_neuron_DIV10_CD1.motif_K_PDS$G4_signal, levels = c("Motif", "K+", "PDS") )

ggplot(primary_cortical_neuron_DIV10_CD1.motif_K_PDS) +
  geom_point(aes(Type, diff, size = -log10(P_value.Bonferroni), shape=P_value.Bonferroni<0.05, colour=log2(OR) ) ) +
  scale_colour_gradient2(low="blue", high="red", mid = "grey", midpoint = 0) +
  facet_grid(G4_signal ~ .) +
  ylab("") +
  theme_bw() +
      theme(legend.position = "top", legend.direction = "horizontal") 


```




```{r,  fig.height=4, fig.width=10}

supp.dotplot.D <- splice_node_dot_plot2("../KCL/MicroExonator/primary_cortical_neuron_DIV10_Tc1.diff.upstream_100nt.G4s",  "../KCL/MicroExonator/primary_cortical_neuron_DIV10_Tc1.diff.downstream_100nt.G4s"  )



primary_cortical_neuron_DIV10_Tc1.motif <- splice_node_dot_plot3("../KCL/MicroExonator/closest/primary_cortical_neuron_DIV10_Tc1.diff.upstream.G4s.fix",
                                        "../KCL/MicroExonator/closest/primary_cortical_neuron_DIV10_Tc1.diff.downstream.G4s.fix",
                                        100)

primary_cortical_neuron_DIV10_Tc1.K <- splice_node_dot_plot3("../KCL/MicroExonator/closest/primary_cortical_neuron_DIV10_Tc1.diff.upstream.G4seq_K",
                                        "../KCL/MicroExonator/closest/primary_cortical_neuron_DIV10_Tc1.diff.downstream.G4seq_K",
                                        45)


primary_cortical_neuron_DIV10_Tc1.PDS <- splice_node_dot_plot3("../KCL/MicroExonator/closest/primary_cortical_neuron_DIV10_Tc1.diff.upstream.G4seq_PDS",
                                        "../KCL/MicroExonator/closest/primary_cortical_neuron_DIV10_Tc1.diff.downstream.G4seq_PDS",
                                        45)




primary_cortical_neuron_DIV10_Tc1.motif[ , G4_signal:="Motif"  ]
primary_cortical_neuron_DIV10_Tc1.K[ , G4_signal:="K+"  ]
primary_cortical_neuron_DIV10_Tc1.PDS[ , G4_signal:="PDS"  ]

primary_cortical_neuron_DIV10_Tc1.motif_K_PDS <- rbind(primary_cortical_neuron_DIV10_Tc1.motif, primary_cortical_neuron_DIV10_Tc1.K, primary_cortical_neuron_DIV10_Tc1.PDS)

primary_cortical_neuron_DIV10_Tc1.motif_K_PDS$G4_signal <- factor(primary_cortical_neuron_DIV10_Tc1.motif_K_PDS$G4_signal, levels = c("Motif", "K+", "PDS") )

ggplot(primary_cortical_neuron_DIV10_Tc1.motif_K_PDS) +
  geom_point(aes(Type, diff, size = -log10(P_value.Bonferroni), shape=P_value.Bonferroni<0.05, colour=log2(OR) ) ) +
  scale_colour_gradient2(low="blue", high="red", mid = "grey", midpoint = 0) +
  facet_grid(G4_signal ~ .) +
  ylab("") +
  theme_bw() +
      theme(legend.position = "top", legend.direction = "horizontal") 

```






```{r,  fig.height=4, fig.width=10}

supp.dotplot.B <- splice_node_dot_plot2("../KCL/MicroExonator/primary_cortical_neuron_DIV4_CD1.diff.upstream_100nt.G4s",  "../KCL/MicroExonator/primary_cortical_neuron_DIV4_CD1.diff.downstream_100nt.G4s"  )



primary_cortical_neuron_DIV4_CD1.motif <- splice_node_dot_plot3("../KCL/MicroExonator/closest/primary_cortical_neuron_DIV4_CD1.diff.upstream.G4s.fix",
                                        "../KCL/MicroExonator/closest/primary_cortical_neuron_DIV4_CD1.diff.downstream.G4s.fix",
                                        100)



primary_cortical_neuron_DIV4_CD1.K <- splice_node_dot_plot3("../KCL/MicroExonator/closest/primary_cortical_neuron_DIV4_CD1.diff.upstream.G4seq_K",
                                        "../KCL/MicroExonator/closest/primary_cortical_neuron_DIV4_CD1.diff.downstream.G4seq_K",
                                        45)


primary_cortical_neuron_DIV4_CD1.PDS <- splice_node_dot_plot3("../KCL/MicroExonator/closest/primary_cortical_neuron_DIV4_CD1.diff.upstream.G4seq_PDS",
                                        "../KCL/MicroExonator/closest/primary_cortical_neuron_DIV4_CD1.diff.downstream.G4seq_PDS",
                                        45)




primary_cortical_neuron_DIV4_CD1.motif[ , G4_signal:="Motif"  ]
primary_cortical_neuron_DIV4_CD1.K[ , G4_signal:="K+"  ]
primary_cortical_neuron_DIV4_CD1.PDS[ , G4_signal:="PDS"  ]

primary_cortical_neuron_DIV4_CD1.motif_K_PDS <- rbind(primary_cortical_neuron_DIV4_CD1.motif, primary_cortical_neuron_DIV4_CD1.K, primary_cortical_neuron_DIV4_CD1.PDS)

primary_cortical_neuron_DIV4_CD1.motif_K_PDS$G4_signal <- factor(primary_cortical_neuron_DIV4_CD1.motif_K_PDS$G4_signal, levels = c("Motif", "K+", "PDS") )

ggplot(primary_cortical_neuron_DIV4_CD1.motif_K_PDS) +
  geom_point(aes(Type, diff, size = -log10(P_value.Bonferroni), shape=P_value.Bonferroni<0.05, colour=log2(OR) ) ) +
  scale_colour_gradient2(low="blue", high="red", mid = "grey", midpoint = 0) +
  facet_grid(G4_signal ~ .) +
  ylab("") +
  theme_bw() +
      theme(legend.position = "top", legend.direction = "horizontal") 

```



```{r}


KCL_human[(DeltaPsi<=-0.1 & Probability>0.9), Delta_type:="Included"]
KCL_human[(DeltaPsi>=0.1 & Probability>0.9), Delta_type:="Skiped"]
KCL_human[is.na(Delta_type), Delta_type:="None"]
KCL_human_stats <-  KCL_human[Type=="CE" , .(count=.N), by=c("Delta_type", "exon", "G4_bol") ] 
KCL_human_stats[, total:=sum(count), by="G4_bol"]
KCL_human_stats[, fraction:=count/total]

Fig5.B <- ggplot() +
  geom_bar(data=KCL_human_stats[Delta_type!="None" & exon=="exon", ] , aes(x= G4_bol, y=fraction, group=Delta_type, fill=Delta_type ) , position="dodge", stat="identity"  ) +
  xlab("") +
  ylab("Fraction of differentially inclued exons ") +
  theme_bw() +
  guides(fill=guide_legend(title="Alternative splicing event")) +
    theme(legend.position = "top", legend.direction = "horizontal") 

Fig5.B

```



```{r}


diff.up <- read_dist_table("~/Google_Drive/Results/Non_B/KCL/SS_enrichment/Control_vs_KCL.diff_upstream_prob_0_9.bed.score.All_G4.tsv.clean.txt..bed.list.out.num")
diff.down <- read_dist_table("~/Google_Drive/Results/Non_B/KCL/SS_enrichment/Control_vs_KCL.diff_downstream_prob_0_9.bed.score.All_G4.tsv.clean.txt..bed.list.out.num")
eql.up <- read_dist_table("~/Google_Drive/Results/Non_B/KCL/SS_enrichment/Control_vs_KCL.diff_upstream_less.bed.score.All_G4.tsv.clean.txt..bed.list.out.num")
eql.down <- read_dist_table("~/Google_Drive/Results/Non_B/KCL/SS_enrichment/Control_vs_KCL.diff_downstream_less.bed.score.All_G4.tsv.clean.txt..bed.list.out.num")


diff.up[, `:=`(type="diff", pos="up" )]
diff.down[, `:=`(type="diff", pos="down" )]
eql.up[, `:=`(type="eql", pos="up" )]
eql.down[, `:=`(type="eql", pos="down" )]


diff_eql <-  rbind(diff.up,  diff.down, eql.up, eql.down)


ggplot(diff_eql) +
  geom_line(aes(x=Position, y=Enrrichment, group=type)) +
  facet_grid(pos ~ .) +
  theme_bw() 

```




```{r}

plot_eql_diff <- function(eql_up_plus, eql_up_minus, eql_down_plus, eql_down_minus, diff_up_plus, diff_up_minus, diff_down_plus, diff_down_minus   ){ 


diff.up_plus <- read_dist_table(diff_up_plus)
diff.up_minus <- read_dist_table(diff_up_minus)
diff.down_plus <- read_dist_table(diff_down_plus)
diff.down_minus <- read_dist_table(diff_down_minus)
diff.up_minus[,Position:=Position*-1]
diff.down_minus[,Position:=Position*-1]


diff.TOTAL <- plot_density(diff.up_plus, diff.up_minus, diff.down_plus, diff.down_minus)


eql.up_plus <- read_dist_table(eql_up_plus)
eql.up_minus <- read_dist_table(eql_up_minus)
eql.down_plus <- read_dist_table(eql_down_plus)
eql.down_minus <- read_dist_table(eql_down_minus)
eql.up_minus[,Position:=Position*-1]
eql.down_minus[,Position:=Position*-1]


eql.TOTAL <- plot_density(eql.up_plus, eql.up_minus, eql.down_plus, eql.down_minus)

diff.TOTAL[, type:="diff"]
eql.TOTAL[, type:="eql"]

diff_eql.TOTAL <- rbind(diff.TOTAL, eql.TOTAL)


ggplot(diff_eql.TOTAL) +
  geom_line(aes(x=Position, y=Enrrichment, color=type)) +
  xlim(c(-250,250)) +
  facet_grid(. ~ exon_pos  ) +
  labs(colour = "Potassium effect") +
  theme_bw() +
  theme(legend.position = "top", legend.direction = "horizontal") 

}
```









```{r}



plot_eql_diff_binomial <- function(eql_up_plus, eql_up_minus, eql_down_plus, eql_down_minus, diff_up_plus, diff_up_minus, diff_down_plus, diff_down_minus, TOTAL.eql, TOTAL.diff, sig   ){ 


diff.up_plus <- read_dist_table(diff_up_plus)
diff.up_minus <- read_dist_table(diff_up_minus)
diff.down_plus <- read_dist_table(diff_down_plus)
diff.down_minus <- read_dist_table(diff_down_minus)
diff.up_minus[,Position:=Position*-1]
diff.down_minus[,Position:=Position*-1]


diff.TOTAL <- plot_density_binomial(diff.up_plus, diff.up_minus, diff.down_plus, diff.down_minus, TOTAL.diff, sig)


eql.up_plus <- read_dist_table(eql_up_plus)
eql.up_minus <- read_dist_table(eql_up_minus)
eql.down_plus <- read_dist_table(eql_down_plus)
eql.down_minus <- read_dist_table(eql_down_minus)
eql.up_minus[,Position:=Position*-1]
eql.down_minus[,Position:=Position*-1]


eql.TOTAL <- plot_density_binomial(eql.up_plus, eql.up_minus, eql.down_plus, eql.down_minus, TOTAL.eql, sig)

diff.TOTAL[, type:="diff"]
eql.TOTAL[, type:="eql"]

diff_eql.TOTAL <- rbind(diff.TOTAL, eql.TOTAL)


ggplot(diff_eql.TOTAL) +
  geom_line(aes(x=Position, y=Enrrichment, color=type)) +
   geom_ribbon(aes(ymin=Enrrichment_l, ymax=Enrrichment_u, x=Position, fill=type), alpha=0.3 )+
  xlim(c(-250,250)) +
  facet_grid(. ~ exon_pos  ) +
  labs(colour = "Potassium effect") +
  theme_bw() +
  theme(legend.position = "top", legend.direction = "horizontal") +
  scale_fill_manual(values=c("#669900", "grey")) +  
  scale_color_manual(values=c("#669900", "darkgrey"))




}
```



```{r}
KCL_human <- fread("../KCL/MicroExonator/Control_vs_KCL.diff", sep="\t")
KCL_human <- KCL_human[, 1:11]
colnames(KCL_human) <- c("Gene","Node", "Coord", "Strand", "Type", "Psi_A", "Psi_B", "DeltaPsi", "Probability", "Complexity", "Entropy")
human.CE.NA <- nrow(KCL_human[(abs(DeltaPsi)<0.1 | Probability<0.9)  &  Type=="CE"])
human.CE.diff <- nrow(KCL_human[abs(DeltaPsi)>=0.1 & Probability>=0.9 &  Type=="CE"])


KCL_ESC_derived_neuron_CD1 <- fread("../KCL/MicroExonator/ESC-derived_neuron_CD1.diff", sep="\t")
KCL_ESC_derived_neuron_CD1  <- KCL_ESC_derived_neuron_CD1 [, 1:11]
colnames(KCL_ESC_derived_neuron_CD1 ) <- c("Gene","Node", "Coord", "Strand", "Type", "Psi_A", "Psi_B", "DeltaPsi", "Probability", "Complexity", "Entropy")
ESC_derived_neuron_CD1.CE.NA <- nrow(KCL_ESC_derived_neuron_CD1 [(abs(DeltaPsi)<0.1 | Probability<0.9)  &  Type=="CE"])
ESC_derived_neuron_CD1.CE.diff <- nrow(KCL_ESC_derived_neuron_CD1[abs(DeltaPsi)>=0.1 & Probability>=0.9 &  Type=="CE"])




KCL_primary_cortical_neuron_DIV10_Tc1  <- fread("../KCL/MicroExonator/primary_cortical_neuron_DIV10_Tc1.diff", sep="\t")
KCL_primary_cortical_neuron_DIV10_Tc1  <- KCL_primary_cortical_neuron_DIV10_Tc1 [, 1:11]
colnames(KCL_primary_cortical_neuron_DIV10_Tc1 ) <- c("Gene","Node", "Coord", "Strand", "Type", "Psi_A", "Psi_B", "DeltaPsi", "Probability", "Complexity", "Entropy")
primary_cortical_neuron_DIV10_Tc1.CE.NA <- nrow(KCL_primary_cortical_neuron_DIV10_Tc1 [(abs(DeltaPsi)<0.1 | Probability<0.9)  &  Type=="CE"])
primary_cortical_neuron_DIV10_Tc1.CE.diff <- nrow(KCL_primary_cortical_neuron_DIV10_Tc1 [abs(DeltaPsi)>=0.1 & Probability>=0.9 &  Type=="CE"])


KCL_primary_cortical_neuron_DIV4_CD1  <- fread("../KCL/MicroExonator/primary_cortical_neuron_DIV4_CD1.diff", sep="\t")
KCL_primary_cortical_neuron_DIV4_CD1  <- KCL_primary_cortical_neuron_DIV4_CD1 [, 1:11]
colnames(KCL_primary_cortical_neuron_DIV4_CD1 ) <- c("Gene","Node", "Coord", "Strand", "Type", "Psi_A", "Psi_B", "DeltaPsi", "Probability", "Complexity", "Entropy")
primary_cortical_neuron_DIV4_CD1.CE.NA <- nrow(KCL_primary_cortical_neuron_DIV4_CD1 [(abs(DeltaPsi)<0.1 | Probability<0.9)  &  Type=="CE"])
primary_cortical_neuron_DIV4_CD1.CE.diff <- nrow(KCL_primary_cortical_neuron_DIV4_CD1 [abs(DeltaPsi)>=0.1 & Probability>=0.9 &  Type=="CE"])



KCL_primary_cortical_neuron_DIV10_CD1  <- fread("../KCL/MicroExonator/primary_cortical_neuron_DIV10_CD1.diff", sep="\t")
KCL_primary_cortical_neuron_DIV10_CD1  <- KCL_primary_cortical_neuron_DIV10_CD1 [, 1:11]
colnames(KCL_primary_cortical_neuron_DIV10_CD1 ) <- c("Gene","Node", "Coord", "Strand", "Type", "Psi_A", "Psi_B", "DeltaPsi", "Probability", "Complexity", "Entropy")
primary_cortical_neuron_DIV10_CD1.CE.NA <- nrow(KCL_primary_cortical_neuron_DIV10_CD1 [(abs(DeltaPsi)<0.1 | Probability<0.9)  &  Type=="CE"])
primary_cortical_neuron_DIV10_CD1.CE.diff <- nrow(KCL_primary_cortical_neuron_DIV10_CD1[abs(DeltaPsi)>=0.1 & Probability>=0.9 &  Type=="CE"])


```



```{r, message=FALSE, error=FALSE, warning=FALSE}

 plot_eql_diff_binomial(
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/Control_vs_KCL.diff_upstream_less.bed.plus.score.All_G4.tsv.clean.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/Control_vs_KCL.diff_upstream_less.bed.minus.score.All_G4.tsv.clean.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/Control_vs_KCL.diff_downstream_less.bed.plus.score.All_G4.tsv.clean.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/Control_vs_KCL.diff_downstream_less.bed.minus.score.All_G4.tsv.clean.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/Control_vs_KCL.diff_upstream_prob_0_9.bed.plus.score.All_G4.tsv.clean.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/Control_vs_KCL.diff_upstream_prob_0_9.bed.minus.score.All_G4.tsv.clean.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/Control_vs_KCL.diff_downstream_prob_0_9.bed.plus.score.All_G4.tsv.clean.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/Control_vs_KCL.diff_downstream_prob_0_9.bed.minus.score.All_G4.tsv.clean.txt..bed.list.out.num",
human.CE.NA,
human.CE.diff,
0.05
)


```




```{r, message=FALSE, error=FALSE, warning=FALSE}

plot_eql_diff_binomial(
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/ESC-derived_neuron_CD1.diff_upstream_less.bed.plus.score.mm10.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/ESC-derived_neuron_CD1.diff_upstream_less.bed.minus.score.mm10.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/ESC-derived_neuron_CD1.diff_downstream_less.bed.plus.score.mm10.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/ESC-derived_neuron_CD1.diff_downstream_less.bed.minus.score.mm10.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/ESC-derived_neuron_CD1.diff_upstream_prob_0_9.bed.plus.score.mm10.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/ESC-derived_neuron_CD1.diff_upstream_prob_0_9.bed.minus.score.mm10.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/ESC-derived_neuron_CD1.diff_downstream_prob_0_9.bed.plus.score.mm10.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/ESC-derived_neuron_CD1.diff_downstream_prob_0_9.bed.minus.score.mm10.txt..bed.list.out.num",
ESC_derived_neuron_CD1.CE.NA,
ESC_derived_neuron_CD1.CE.diff,
0.05)
```





```{r, message=FALSE, error=FALSE, warning=FALSE}

plot_eql_diff_binomial(
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/primary_cortical_neuron_DIV10_Tc1.diff_upstream_less.bed.plus.score.mm10.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/primary_cortical_neuron_DIV10_Tc1.diff_upstream_less.bed.minus.score.mm10.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/primary_cortical_neuron_DIV10_Tc1.diff_downstream_less.bed.plus.score.mm10.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/primary_cortical_neuron_DIV10_Tc1.diff_downstream_less.bed.minus.score.mm10.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/primary_cortical_neuron_DIV10_Tc1.diff_upstream_prob_0_9.bed.plus.score.mm10.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/primary_cortical_neuron_DIV10_Tc1.diff_upstream_prob_0_9.bed.minus.score.mm10.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/primary_cortical_neuron_DIV10_Tc1.diff_downstream_prob_0_9.bed.plus.score.mm10.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/primary_cortical_neuron_DIV10_Tc1.diff_downstream_prob_0_9.bed.minus.score.mm10.txt..bed.list.out.num",
primary_cortical_neuron_DIV10_Tc1.CE.NA,
primary_cortical_neuron_DIV10_Tc1.CE.diff,
0.05)
```










```{r, message=FALSE, error=FALSE, warning=FALSE}

plot_eql_diff_binomial(
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/primary_cortical_neuron_DIV4_CD1.diff_upstream_less.bed.plus.score.mm10.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/primary_cortical_neuron_DIV4_CD1.diff_upstream_less.bed.minus.score.mm10.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/primary_cortical_neuron_DIV4_CD1.diff_downstream_less.bed.plus.score.mm10.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/primary_cortical_neuron_DIV4_CD1.diff_downstream_less.bed.minus.score.mm10.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/primary_cortical_neuron_DIV4_CD1.diff_upstream_prob_0_9.bed.plus.score.mm10.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/primary_cortical_neuron_DIV4_CD1.diff_upstream_prob_0_9.bed.minus.score.mm10.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/primary_cortical_neuron_DIV4_CD1.diff_downstream_prob_0_9.bed.plus.score.mm10.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/primary_cortical_neuron_DIV4_CD1.diff_downstream_prob_0_9.bed.minus.score.mm10.txt..bed.list.out.num",
primary_cortical_neuron_DIV4_CD1.CE.NA,
primary_cortical_neuron_DIV4_CD1.CE.diff,
0.05
)
```










```{r, message=FALSE, error=FALSE, warning=FALSE}

plot_eql_diff_binomial(
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/primary_cortical_neuron_DIV10_CD1.diff_upstream_less.bed.plus.score.mm10.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/primary_cortical_neuron_DIV10_CD1.diff_upstream_less.bed.minus.score.mm10.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/primary_cortical_neuron_DIV10_CD1.diff_downstream_less.bed.plus.score.mm10.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/primary_cortical_neuron_DIV10_CD1.diff_downstream_less.bed.minus.score.mm10.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/primary_cortical_neuron_DIV10_CD1.diff_upstream_prob_0_9.bed.plus.score.mm10.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/primary_cortical_neuron_DIV10_CD1.diff_upstream_prob_0_9.bed.minus.score.mm10.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/primary_cortical_neuron_DIV10_CD1.diff_downstream_prob_0_9.bed.plus.score.mm10.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/SS_enrichment/plus_minus/primary_cortical_neuron_DIV10_CD1.diff_downstream_prob_0_9.bed.minus.score.mm10.txt..bed.list.out.num",
primary_cortical_neuron_DIV10_CD1.CE.NA,
primary_cortical_neuron_DIV10_CD1.CE.diff,
0.05)
```







### G4seq 





```{r, message=FALSE, error=FALSE, warning=FALSE}

plot_eql_diff(
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/Control_vs_KCL.diff_upstream_less.bed.plus.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/Control_vs_KCL.diff_upstream_less.bed.minus.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/Control_vs_KCL.diff_downstream_less.bed.plus.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/Control_vs_KCL.diff_downstream_less.bed.minus.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num", 

"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/Control_vs_KCL.diff_upstream_prob_0_9.bed.plus.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/Control_vs_KCL.diff_upstream_prob_0_9.bed.minus.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/Control_vs_KCL.diff_downstream_prob_0_9.bed.plus.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/Control_vs_KCL.diff_downstream_prob_0_9.bed.minus.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num"

)
```


```{r, message=FALSE, error=FALSE, warning=FALSE}

Fig5.C <- plot_eql_diff_binomial(
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/Control_vs_KCL.diff_upstream_less.bed.plus.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/Control_vs_KCL.diff_upstream_less.bed.minus.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/Control_vs_KCL.diff_downstream_less.bed.plus.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/Control_vs_KCL.diff_downstream_less.bed.minus.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num", 

"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/Control_vs_KCL.diff_upstream_prob_0_9.bed.plus.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/Control_vs_KCL.diff_upstream_prob_0_9.bed.minus.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/Control_vs_KCL.diff_downstream_prob_0_9.bed.plus.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/Control_vs_KCL.diff_downstream_prob_0_9.bed.minus.score.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
human.CE.NA,
human.CE.diff,
0.05)


Fig5.c <- Fig5.C +ylim( c(0.4, 2.1))

Fig5.c
```




```{r, message=FALSE, error=FALSE, warning=FALSE}



Fig5.D <- plot_eql_diff_binomial(
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/Control_vs_KCL.diff_upstream_less.bed.plus.score.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/Control_vs_KCL.diff_upstream_less.bed.minus.score.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/Control_vs_KCL.diff_downstream_less.bed.plus.score.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/Control_vs_KCL.diff_downstream_less.bed.minus.score.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/Control_vs_KCL.diff_upstream_prob_0_9.bed.plus.score.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/Control_vs_KCL.diff_upstream_prob_0_9.bed.minus.score.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/Control_vs_KCL.diff_downstream_prob_0_9.bed.plus.score.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/Control_vs_KCL.diff_downstream_prob_0_9.bed.minus.score.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
human.CE.NA,
human.CE.diff,
0.05
)


Fig5.D <- Fig5.D +ylim( c(0.4, 2.1))
Fig5.D
```




```{r, message=FALSE, error=FALSE, warning=FALSE}


sup.mus.ss_enrichment.ESC.K <- plot_eql_diff_binomial(
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/ESC-derived_neuron_CD1.diff_upstream_less.bed.plus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/ESC-derived_neuron_CD1.diff_upstream_less.bed.minus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/ESC-derived_neuron_CD1.diff_downstream_less.bed.plus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/ESC-derived_neuron_CD1.diff_downstream_less.bed.minus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/ESC-derived_neuron_CD1.diff_upstream_prob_0_9.bed.plus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/ESC-derived_neuron_CD1.diff_upstream_prob_0_9.bed.minus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/ESC-derived_neuron_CD1.diff_downstream_prob_0_9.bed.plus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/ESC-derived_neuron_CD1.diff_downstream_prob_0_9.bed.minus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",

ESC_derived_neuron_CD1.CE.NA,
ESC_derived_neuron_CD1.CE.diff,
0.05

)

```





```{r, message=FALSE, error=FALSE, warning=FALSE}


sup.mus.ss_enrichment.ESC.PDS  <- plot_eql_diff_binomial(
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/ESC-derived_neuron_CD1.diff_upstream_less.bed.plus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/ESC-derived_neuron_CD1.diff_upstream_less.bed.minus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/ESC-derived_neuron_CD1.diff_downstream_less.bed.plus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/ESC-derived_neuron_CD1.diff_downstream_less.bed.minus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",


"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/ESC-derived_neuron_CD1.diff_upstream_prob_0_9.bed.plus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/ESC-derived_neuron_CD1.diff_upstream_prob_0_9.bed.minus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/ESC-derived_neuron_CD1.diff_downstream_prob_0_9.bed.plus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/ESC-derived_neuron_CD1.diff_downstream_prob_0_9.bed.minus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",

ESC_derived_neuron_CD1.CE.NA,
ESC_derived_neuron_CD1.CE.diff,
0.05

)

```






```{r, message=FALSE, error=FALSE, warning=FALSE}

sup.mus.ss_enrichment.DIV10_CD1.K <- plot_eql_diff_binomial(
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_CD1.diff_upstream_less.bed.plus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_CD1.diff_upstream_less.bed.minus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_CD1.diff_downstream_less.bed.plus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_CD1.diff_downstream_less.bed.minus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_CD1.diff_upstream_prob_0_9.bed.plus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_CD1.diff_upstream_prob_0_9.bed.minus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_CD1.diff_downstream_prob_0_9.bed.plus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_CD1.diff_downstream_prob_0_9.bed.minus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",

primary_cortical_neuron_DIV10_CD1.CE.NA,
primary_cortical_neuron_DIV10_CD1.CE.diff,
0.05
)

```



```{r, message=FALSE, error=FALSE, warning=FALSE}


sup.mus.ss_enrichment.DIV10_CD1.PDS <- plot_eql_diff_binomial(
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_CD1.diff_upstream_less.bed.plus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_CD1.diff_upstream_less.bed.minus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_CD1.diff_downstream_less.bed.plus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_CD1.diff_downstream_less.bed.minus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",


"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_CD1.diff_upstream_prob_0_9.bed.plus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_CD1.diff_upstream_prob_0_9.bed.minus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_CD1.diff_downstream_prob_0_9.bed.plus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_CD1.diff_downstream_prob_0_9.bed.minus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",

primary_cortical_neuron_DIV10_CD1.CE.NA,
primary_cortical_neuron_DIV10_CD1.CE.diff,
0.05
)

```





```{r, message=FALSE, error=FALSE, warning=FALSE}


sup.mus.ss_enrichment.DIV10_Tc1.K <- plot_eql_diff_binomial(
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_Tc1.diff_upstream_less.bed.plus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_Tc1.diff_upstream_less.bed.minus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_Tc1.diff_downstream_less.bed.plus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_Tc1.diff_downstream_less.bed.minus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",


"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_Tc1.diff_upstream_prob_0_9.bed.plus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_Tc1.diff_upstream_prob_0_9.bed.minus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_Tc1.diff_downstream_prob_0_9.bed.plus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_Tc1.diff_downstream_prob_0_9.bed.minus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",

primary_cortical_neuron_DIV10_Tc1.CE.NA,
primary_cortical_neuron_DIV10_Tc1.CE.diff,
0.05

)

```





```{r, message=FALSE, error=FALSE, warning=FALSE}


sup.mus.ss_enrichment.DIV10_Tc1.PDS <- plot_eql_diff_binomial(
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_Tc1.diff_upstream_less.bed.plus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_Tc1.diff_upstream_less.bed.minus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_Tc1.diff_downstream_less.bed.plus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_Tc1.diff_downstream_less.bed.minus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",


"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_Tc1.diff_upstream_prob_0_9.bed.plus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_Tc1.diff_upstream_prob_0_9.bed.minus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_Tc1.diff_downstream_prob_0_9.bed.plus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV10_Tc1.diff_downstream_prob_0_9.bed.minus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
primary_cortical_neuron_DIV10_Tc1.CE.NA,
primary_cortical_neuron_DIV10_Tc1.CE.diff,
0.05

)

```





```{r, message=FALSE, error=FALSE, warning=FALSE}


sup.mus.ss_enrichment.DIV4_CD1.K <- plot_eql_diff_binomial(
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV4_CD1.diff_upstream_less.bed.plus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV4_CD1.diff_upstream_less.bed.minus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV4_CD1.diff_downstream_less.bed.plus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV4_CD1.diff_downstream_less.bed.minus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",


"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV4_CD1.diff_upstream_prob_0_9.bed.plus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV4_CD1.diff_upstream_prob_0_9.bed.minus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV4_CD1.diff_downstream_prob_0_9.bed.plus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV4_CD1.diff_downstream_prob_0_9.bed.minus.score.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed.txt..bed.list.out.num",


primary_cortical_neuron_DIV4_CD1.CE.NA,
primary_cortical_neuron_DIV4_CD1.CE.diff,
0.05

)

```





```{r, message=FALSE, error=FALSE, warning=FALSE}


sup.mus.ss_enrichment.DIV4_CD1.PDS <- plot_eql_diff_binomial(
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV4_CD1.diff_upstream_less.bed.plus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV4_CD1.diff_upstream_less.bed.minus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV4_CD1.diff_downstream_less.bed.plus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV4_CD1.diff_downstream_less.bed.minus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",


"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV4_CD1.diff_upstream_prob_0_9.bed.plus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV4_CD1.diff_upstream_prob_0_9.bed.minus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV4_CD1.diff_downstream_prob_0_9.bed.plus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",
"~/Google_Drive/Results/Non_B/KCL/G4seq_enrichment/primary_cortical_neuron_DIV4_CD1.diff_downstream_prob_0_9.bed.minus.score.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed.txt..bed.list.out.num",

primary_cortical_neuron_DIV4_CD1.CE.NA,
primary_cortical_neuron_DIV4_CD1.CE.diff,
0.05
)

```


### smooth ###




```{r}

rep(1:1999, each=4)

diff_eql.TOTAL[ , bin:=rep(1:1999, each=4) ]


diff_eql.TOTAL[, mean_Enrrichment:=mean(Enrrichment) , by=bin]


ggplot(diff_eql.TOTAL) +
  geom_line(aes(x=Position, y=mean_Enrrichment, color=type)) +
  xlim(c(-250,250)) +
  facet_grid(. ~ exon_pos  ) +
  labs(colour = "Potassium effect") +
  theme_bw() +
  theme(legend.position = "top", legend.direction = "horizontal") 
```


#Intron lenght


```{r}
intron_upstream <- fread("../Figures/Figure3/Introns.hg19_upstream_100nt_window.bed.intersect_c.G4s")
intron_downstream <- fread("../Figures/Figure3/Introns.hg19_downstream_100nt_window.bed.intersect_c.G4s")

colnames(intron_upstream) <- c("chrom", "start", "end", "strand", "ID", "intron_size", "transcript", "G4")
colnames(intron_downstream) <- c("chrom", "start", "end", "strand", "ID", "intron_size", "transcript", "G4")

intron_upstream[, pos:="Upstream intron"]
intron_downstream[, pos:="Downstream intron"]

intron_size <-  rbind(intron_upstream, intron_downstream)
intron_size$pos <- factor(intron_size$pos, level=c("Upstream intron", "Downstream intron"))


Fig3.B <- ggplot(intron_size) +
  geom_line(aes(intron_size, colour=G4>0), stat="density") +
  facet_grid(. ~ pos) +
  xlim(c(0,5000)) +
  theme(legend.position = "top", legend.direction = "horizontal")

Fig3.B
```

```{r}
ks.test(intron_size[G4>0, intron_size], intron_size[G4==0, intron_size])
```



```{r}
size_int <- c(seq(0,5000,10), c(10000, 100000, 1000000, 10000000))


intron_size$int_count <- cut(intron_size$intron_size, size_int)


int_counts_up_w_G4 <- table(intron_size[pos=="Upstream intron" & G4>0, ]$int_count)
int_counts_up_wo_G4 <- table(intron_size[pos=="Upstream intron" & G4==0, ]$int_count)
int_counts_up <- rbind(int_counts_up_w_G4, int_counts_up_wo_G4)
colnames(int_counts_up) <- c(seq(10,5000,10), c(10000, 100000, 1000000, 10000000))
int_counts_up <- melt(int_counts_up)


int_counts_down_w_G4 <- table(intron_size[pos=="Downstream intron" & G4>0, ]$int_count)
int_counts_down_wo_G4 <- table(intron_size[pos=="Downstream intron" & G4==0, ]$int_count)
int_counts_down <- rbind(int_counts_down_w_G4, int_counts_down_wo_G4)
colnames(int_counts_down) <- c(seq(10,5000,10), c(10000, 100000, 1000000, 10000000))
int_counts_down <- melt(int_counts_down)

intron_size_ints <- rbind(int_counts_up, int_counts_down)

colnames(intron_size_ints) <- c("factor", "int", "count")

intron_size_ints <- data.table(intron_size_ints)


intron_size_ints[ , total:=sum(count) , by="factor" ]
intron_size_ints[ , frac:=(count/total) ]

intron_size_ints_up <- merge(intron_size_ints[ factor=="int_counts_up_w_G4", ], intron_size_ints[ factor=="int_counts_up_wo_G4", ], by="int" )
intron_size_ints_up[ , relative_frac:=frac.x/frac.y ]

intron_size_ints_down <- merge(intron_size_ints[ factor=="int_counts_down_w_G4", ], intron_size_ints[ factor=="int_counts_down_wo_G4", ], by="int" )
intron_size_ints_down[ , relative_frac:=frac.x/frac.y ]

intron_size_ints_up[, pos:="Upstream intron"]
intron_size_ints_down[, pos:="Downstream intron"]

intron_size_ints_total <- rbind(intron_size_ints_up, intron_size_ints_down)

intron_size_ints_total$pos <- factor(intron_size_ints_total$pos, level=c("Upstream intron", "Downstream intron"))



Fig3.C <- ggplot(intron_size_ints_total, aes(int, log2(relative_frac))  ) +
  geom_hline(yintercept=0, linetype="dashed") +
  geom_point() +
  stat_smooth( method = "lm", formula = y ~ poly(x, 8)) +
  xlim(c(0, 2000)) +
  ylim(c(-2,2)) +
  facet_grid(. ~ pos)

Fig3.C

```



```{r}

intron_size_ints_total[ relative_frac %in% intron_size_ints_total[, max(relative_frac), by=pos]$V1, ]

```



```{r}
 Upstream_U2_score_len <- fread("../Figures/Figure3/Upstream_U2_score_len_matrix.txt2.old")
 Downstream_U2_score_len <- fread("../Figures/Figure3/Downstream_U2_score_len_matrix.txt2.old")
 
 
 Upstream_U2_score_len <-data.table::melt(Upstream_U2_score_len, id.vars='IL')
Downstream_U2_score_len <- data.table::melt(Downstream_U2_score_len, id.vars='IL')

Upstream_U2_score_len[, pos:="Upstream intron"]
Downstream_U2_score_len[, pos:="Downstream intron"]


 
Fig3.D1 <-  ggplot(Upstream_U2_score_len) +
   geom_tile( aes(variable, as.factor(IL), fill = value*100)) +
   scale_fill_gradient2(low="blue", high="red", mid = "grey", midpoint = 12.5, limits = c(0, 25)) +
   facet_grid(. ~ pos) +
    theme(legend.position = "bottom", legend.direction = "horizontal") + theme(axis.text.x = element_text(angle = 90)) +labs(x = NULL, y = NULL)
 
 
Fig3.D2 <-  ggplot(Downstream_U2_score_len) +
   geom_tile( aes(variable, as.factor(IL), fill = value*100)) +
   scale_fill_gradient2(low="blue", high="red", mid = "grey", midpoint = 12.5, limits = c(0, 25)) +
   facet_grid(. ~ pos) +
    theme(legend.position = "bottom", legend.direction = "horizontal") +
      theme(legend.position = "bottom", legend.direction = "horizontal") + theme(axis.text.x = element_text(angle = 90)) +labs(x = NULL, y = NULL)

Fig3.D <- plot_grid(Fig3.D1, Fig3.D2)
 Fig3.D
```




```{r}

closest_g4 <- function(dist.exon_up_plus_path, dist.exon_up_minus_path, dist.exon_down_plus_path, dist.exon_down_minus_path){

dist.exon_up_plus <- fread(dist.exon_up_plus_path)
colnames(dist.exon_up_plus) <- c("echrom", "estart", "eend", "exon", "cero", "strand", "gchrom", "gstart", "gend", "gscore", "exon_dist")

dist.exon_up_plus[, gmid:= gstart + (gend-gstart)/2]
dist.exon_up_plus[ ,  dist:=gmid - estart]


dist.exon_up_minus <- fread(dist.exon_up_minus_path)
colnames(dist.exon_up_minus) <- c("echrom", "estart", "eend", "exon", "cero", "strand", "gchrom", "gstart", "gend", "gscore", "exon_dist")

dist.exon_up_minus[, gmid:= gstart + (gend-gstart)/2]
dist.exon_up_minus[ ,  dist:= estart - gmid]


dist.exon_down_plus <- fread(dist.exon_down_plus_path)
colnames(dist.exon_down_plus) <- c("echrom", "estart", "eend", "exon", "cero", "strand", "gchrom", "gstart", "gend", "gscore", "exon_dist")

dist.exon_down_plus[, gmid:= gstart + (gend-gstart)/2]
dist.exon_down_plus[ ,  dist:=gmid - estart]


dist.exon_down_minus <- fread(dist.exon_down_minus_path)
colnames(dist.exon_down_minus) <- c("echrom", "estart", "eend", "exon", "cero", "strand", "gchrom", "gstart", "gend", "gscore", "exon_dist")

dist.exon_down_minus[, gmid:= gstart + (gend-gstart)/2]
dist.exon_down_minus[ ,  dist:= estart - gmid]


dist.exon_up <- rbind(dist.exon_up_plus, dist.exon_up_minus) 
dist.exon_down <- rbind(dist.exon_down_plus, dist.exon_down_minus)

dist.exon_up[, pos:="3'Splice site"]
dist.exon_down[, pos:="5'Splice site"]

dist.exon <- rbind(dist.exon_up, dist.exon_down)

return(dist.exon)

}



```



```{r}
exon.up.motif <- fread("../Figures/Figure1/exons.up.closest_G4_clean")
colnames(exon.up.motif) <- c("echrom", "estart", "eend", "exon", "cero", "strand", "gchrom", "gstart", "gend")

exon.up.motif[ , gmid:= gstart + (gend-gstart)/2]
exon.up.motif[  strand=="+"  ,  dist:=gmid - estart]
exon.up.motif[  strand=="-"  ,  dist:= estart - gmid]


exon.down.motif <- fread("../Figures/Figure1/exons.down.closest_G4_clean")
colnames(exon.down.motif) <- c("echrom", "estart", "eend", "exon", "cero", "strand", "gchrom", "gstart", "gend")

exon.down.motif[ , gmid:= gstart + (gend-gstart)/2]
exon.down.motif[  strand=="+"  ,  dist:=gmid - estart]
exon.down.motif[  strand=="-"  ,  dist:= estart - gmid]



exon.up.motif[, pos:="3'Splice site"]
exon.down.motif[, pos:="5'Splice site"]

exon.dist.motif <- rbind(exon.up.motif, exon.down.motif)


ggplot(exon.dist.motif) +
  geom_density(aes(dist)) +
  xlim(c(-1000, 1000)) +
  facet_grid(. ~ pos)
 
```




```{r, fig.height=7, fig.width=7}

closest_g4_KCL <- closest_g4("../Figures/Figure1/exon_up_plus.hg19.closest.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed",
                             "Figures/Figure1/exon_up_minus.hg19.closest.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed",
                             "Figures/Figure1/exon_down_plus.hg19.closest.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed",
                             "Figures/Figure1/exon_down_minus.hg19.closest.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed")



closest_g4_PDS <- closest_g4("../Figures/Figure1/exon_up_plus.hg19.closest.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed",
                             "Figures/Figure1/exon_up_minus.hg19.closest.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed",
                             "Figures/Figure1/exon_down_plus.hg19.closest.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed",
                             "Figures/Figure1/exon_down_minus.hg19.closest.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed")


closest_g4_KCL[, Type:="G4-seq K+"]
closest_g4_PDS[, Type:="G4-seq PDS"]
exon.dist.motif[, Type:="Consensus G4 motif"]


closest_g4_KCL <- unique(closest_g4_KCL[ , c("echrom", "estart", "eend", "dist", "pos", "Type")])
closest_g4_PDS <- unique(closest_g4_PDS[ , c("echrom", "estart", "eend", "dist", "pos",  "Type")])
exon.dist.motif <- unique(exon.dist.motif[ , c("echrom", "estart", "eend", "dist", "pos", "Type")])


closest_g4_KCL_PDS_motif <- rbind(closest_g4_KCL, closest_g4_PDS, exon.dist.motif)

Fig1.B <- ggplot(closest_g4_KCL_PDS_motif) +
  geom_density(aes(dist, colour=Type), bw=10) +
  xlim(c(-1000, 1000)) +
  facet_grid(Type ~ pos) +
  xlab("Distance") +
  ylab("Density") +
  theme_bw() +
  theme(legend.position = "top", legend.direction = "horizontal")  + 
    theme(axis.text.x = element_text( angle = 45)) 

Fig1.B + xlim(c(-300, 300))

```






```{r}

#closest_g4_KCL_PDS_motif


closest_g4_KCL_PDS_motif.1kb <- closest_g4_KCL_PDS_motif[abs(dist)<=1000 , ]

closest_g4_KCL_PDS_motif.1kb[, bin:=cut(dist, seq(-1000, 1000, 5), labels= seq(-995, 1000, 5), include.lowest=TRUE ) ]


closest_g4_KCL_PDS_motif.1kb.enrichment <- closest_g4_KCL_PDS_motif.1kb[ , .(Occurrences=.N) , by=c("dist", "pos", "Type")]


closest_g4_KCL_PDS_motif.1kb.enrichment.median <-  closest_g4_KCL_PDS_motif.1kb.enrichment[, .(median=median(Occurrences)), by=c( "pos", "Type") ]


closest_g4_KCL_PDS_motif.1kb.enrichment <- merge(closest_g4_KCL_PDS_motif.1kb.enrichment, closest_g4_KCL_PDS_motif.1kb.enrichment.median, by=c( "pos", "Type")) 

closest_g4_KCL_PDS_motif.1kb.enrichment[, Enrrichment:=Occurrences/median]

closest_g4_KCL_PDS_motif.1kb.enrichment[Type=="Consensus G4 motif", Type:="G4 motif" ]


#closest_g4_KCL_PDS_motif.1kb.enrichment[ , bin:=as.numeric(as.character(bin))]


ggplot(closest_g4_KCL_PDS_motif.1kb.enrichment) +
  geom_line(aes(dist, Enrrichment, colour=Type)) +
  xlim(c(-1000, 1000)) +
  facet_grid(Type ~ pos) +
  xlab("Distance") +
  ylab("Enrichment") +
  theme_bw() +
  theme(legend.position = "top", legend.direction = "horizontal")  + 
    theme(axis.text.x = element_text( angle = 45)) 






```




```{r}

closest_g4_2015 <- function(dist.exon_up_plus_path, dist.exon_up_minus_path, dist.exon_down_plus_path, dist.exon_down_minus_path){

dist.exon_up_plus <- fread(dist.exon_up_plus_path)
colnames(dist.exon_up_plus) <- c("echrom", "estart", "eend", "exon", "cero", "strand", "gchrom", "gstart", "gend", "exon_dist")

dist.exon_up_plus[, gmid:= gstart + (gend-gstart)/2]
dist.exon_up_plus[ ,  dist:=gmid - estart]


dist.exon_up_minus <- fread(dist.exon_up_minus_path)
colnames(dist.exon_up_minus) <- c("echrom", "estart", "eend", "exon", "cero", "strand", "gchrom", "gstart", "gend",  "exon_dist")

dist.exon_up_minus[, gmid:= gstart + (gend-gstart)/2]
dist.exon_up_minus[ ,  dist:= estart - gmid]


dist.exon_down_plus <- fread(dist.exon_down_plus_path)
colnames(dist.exon_down_plus) <- c("echrom", "estart", "eend", "exon", "cero", "strand", "gchrom", "gstart", "gend",  "exon_dist")

dist.exon_down_plus[, gmid:= gstart + (gend-gstart)/2]
dist.exon_down_plus[ ,  dist:=gmid - estart]


dist.exon_down_minus <- fread(dist.exon_down_minus_path)
colnames(dist.exon_down_minus) <- c("echrom", "estart", "eend", "exon", "cero", "strand", "gchrom", "gstart", "gend",  "exon_dist")

dist.exon_down_minus[, gmid:= gstart + (gend-gstart)/2]
dist.exon_down_minus[ ,  dist:= estart - gmid]


dist.exon_up <- rbind(dist.exon_up_plus, dist.exon_up_minus) 
dist.exon_down <- rbind(dist.exon_down_plus, dist.exon_down_minus)

dist.exon_up[, pos:="3'Splice site"]
dist.exon_down[, pos:="5'Splice site"]

dist.exon <- rbind(dist.exon_up, dist.exon_down)

return(dist.exon)

}



```




```{r, fig.height=7, fig.width=7}

closest_g4_KCL_2015 <- closest_g4_2015("../Figures/Figure1/exon_up_plus.hg19.closest.GSE63874_Na_K_plus_minus_hits_intersect.bed",
                             "Figures/Figure1/exon_up_minus.hg19.closest.GSE63874_Na_K_plus_minus_hits_intersect.bed",
                             "Figures/Figure1/exon_down_plus.hg19.closest.GSE63874_Na_K_plus_minus_hits_intersect.bed",
                             "Figures/Figure1/exon_down_minus.hg19.closest.GSE63874_Na_K_plus_minus_hits_intersect.bed")



closest_g4_PDS_2015 <- closest_g4_2015("../Figures/Figure1/exon_up_plus.hg19.closest.GSE63874_Na_PDS_plus_minus_hits_intersect.bed",
                             "Figures/Figure1/exon_up_minus.hg19.closest.GSE63874_Na_PDS_plus_minus_hits_intersect.bed",
                             "Figures/Figure1/exon_down_plus.hg19.closest.GSE63874_Na_PDS_plus_minus_hits_intersect.bed",
                             "Figures/Figure1/exon_down_minus.hg19.closest.GSE63874_Na_PDS_plus_minus_hits_intersect.bed")


closest_g4_KCL_2015[, Type:="G4-seq Na+ K+"]
closest_g4_PDS_2015[, Type:="G4-seq Na+ PDS"]



closest_g4_KCL_2015 <- unique(closest_g4_KCL_2015[ , c("echrom", "estart", "eend", "dist", "pos", "Type")])
closest_g4_PDS_2015 <- unique(closest_g4_PDS_2015[ , c("echrom", "estart", "eend", "dist", "pos",  "Type")])



closest_g4_KCL_PDS_2015 <- rbind(closest_g4_KCL_2015, closest_g4_PDS_2015)

ggplot(closest_g4_KCL_PDS_2015) +
  geom_line(aes(dist, colour=Type), bw=10, stat="density") +
  xlim(c(-1000, 1000)) +
  facet_grid(Type ~ pos) +
  xlab("Distance") +
  ylab("Density") +
  theme_bw() +
  theme(legend.position = "top", legend.direction = "horizontal")  + 
    theme(axis.text.x = element_text( angle = 45)) 



```





```{r, fig.height=5, fig.width=7}
closest_g4_KCL.sacCer3 <- closest_g4("../Figures/Figure1/exon_up_plus.bed.sacCer3.closest.GSM3003553_Saccaromyces_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed",
                             "Figures/Figure1/exon_up_minus.bed.sacCer3.closest.GSM3003553_Saccaromyces_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed",
                             "Figures/Figure1/exon_down_plus.bed.sacCer3.closest.GSM3003553_Saccaromyces_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed",
                             "Figures/Figure1/exon_down_minus.bed.sacCer3.closest.GSM3003553_Saccaromyces_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed")

closest_g4_PDS.sacCer3 <- closest_g4("../Figures/Figure1/exon_up_plus.bed.sacCer3.closest.GSM3003554_Saccaromyces_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed",
                             "Figures/Figure1/exon_up_minus.bed.sacCer3.closest.GSM3003554_Saccaromyces_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed",
                             "Figures/Figure1/exon_down_plus.bed.sacCer3.closest.GSM3003554_Saccaromyces_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed",
                             "Figures/Figure1/exon_down_minus.bed.sacCer3.closest.GSM3003554_Saccaromyces_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed")


closest_g4_KCL.tair10 <- closest_g4("../Figures/Figure1/exon_up_plus.bed.tair10.closest.GSM3003535_Arabidopsis_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed",
                             "Figures/Figure1/exon_up_minus.bed.tair10.closest.GSM3003535_Arabidopsis_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed",
                             "Figures/Figure1/exon_down_plus.bed.tair10.closest.GSM3003535_Arabidopsis_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed",
                             "Figures/Figure1/exon_down_minus.bed.tair10.closest.GSM3003535_Arabidopsis_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed")

closest_g4_PDS.tair10 <- closest_g4("../Figures/Figure1/exon_up_plus.bed.tair10.closest.GSM3003536_Arabidopsis_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed",
                             "Figures/Figure1/exon_up_minus.bed.tair10.closest.GSM3003536_Arabidopsis_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed",
                             "Figures/Figure1/exon_down_plus.bed.tair10.closest.GSM3003536_Arabidopsis_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed",
                             "Figures/Figure1/exon_down_minus.bed.tair10.closest.GSM3003536_Arabidopsis_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed")


closest_g4_KCL.danRer10 <- closest_g4("../Figures/Figure1/exon_up_plus.danRer10.closest.GSM3003557_Zebrafish_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed",
                             "Figures/Figure1/exon_up_minus.danRer10.closest.GSM3003557_Zebrafish_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed",
                             "Figures/Figure1/exon_down_plus.danRer10.closest.GSM3003557_Zebrafish_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed",
                             "Figures/Figure1/exon_down_minus.danRer10.closest.GSM3003557_Zebrafish_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed")


closest_g4_PDS.danRer10 <- closest_g4("../Figures/Figure1/exon_up_plus.danRer10.closest.GSM3003558_Zebrafish_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed",
                             "Figures/Figure1/exon_up_minus.danRer10.closest.GSM3003558_Zebrafish_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed",
                             "Figures/Figure1/exon_down_plus.danRer10.closest.GSM3003558_Zebrafish_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed",
                             "Figures/Figure1/exon_down_minus.danRer10.closest.GSM3003558_Zebrafish_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed")


closest_g4_KCL.dm6 <- closest_g4("../Figures/Figure1/exon_up_plus.dm6.closest.GSM3003541_Drosophila_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed",
                             "Figures/Figure1/exon_up_minus.dm6.closest.GSM3003541_Drosophila_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed",
                             "Figures/Figure1/exon_down_plus.dm6.closest.GSM3003541_Drosophila_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed",
                             "Figures/Figure1/exon_down_minus.dm6.closest.GSM3003541_Drosophila_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed")


closest_g4_PDS.dm6 <- closest_g4("../Figures/Figure1/exon_up_plus.dm6.closest.GSM3003542_Drosophila_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed",
                             "Figures/Figure1/exon_up_minus.dm6.closest.GSM3003542_Drosophila_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed",
                             "Figures/Figure1/exon_down_plus.dm6.closest.GSM3003542_Drosophila_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed",
                             "Figures/Figure1/exon_down_minus.dm6.closest.GSM3003542_Drosophila_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed")


closest_g4_KCL.mm10 <- closest_g4("../Figures/Figure1/exon_up_plus.mm10.closest.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed",
                             "Figures/Figure1/exon_up_minus.mm10.closest.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed",
                             "Figures/Figure1/exon_down_plus.mm10.closest.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed",
                             "Figures/Figure1/exon_down_minus.mm10.closest.GSM3003547_Mouse_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed")

closest_g4_PDS.mm10 <- closest_g4("../Figures/Figure1/exon_up_plus.mm10.closest.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed",
                             "Figures/Figure1/exon_up_minus.mm10.closest.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed",
                             "Figures/Figure1/exon_down_plus.mm10.closest.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed",
                             "Figures/Figure1/exon_down_minus.mm10.closest.GSM3003548_Mouse_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed")


closest_g4_KCL.hg19 <- closest_g4("../Figures/Figure1/exon_up_plus.hg19.closest.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed",
                             "Figures/Figure1/exon_up_minus.hg19.closest.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed",
                             "Figures/Figure1/exon_down_plus.hg19.closest.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed",
                             "Figures/Figure1/exon_down_minus.hg19.closest.GSM3003539_Homo_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed")

closest_g4_PDS.hg19 <- closest_g4("../Figures/Figure1/exon_up_plus.hg19.closest.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed",
                             "Figures/Figure1/exon_up_minus.hg19.closest.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed",
                             "Figures/Figure1/exon_down_plus.hg19.closest.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed",
                             "Figures/Figure1/exon_down_minus.hg19.closest.GSM3003540_Homo_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed")



closest_g4_KCL.ce10 <- closest_g4("../Figures/Figure1/exon_up_plus.ce10.closest.GSM3003538_Celegans_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed",
                             "Figures/Figure1/exon_up_minus.ce10.closest.GSM3003537_Celegans_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed",
                             "Figures/Figure1/exon_down_plus.ce10.closest.GSM3003537_Celegans_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed",
                             "Figures/Figure1/exon_down_minus.ce10.closest.GSM3003537_Celegans_all_w15_th-1_plus_minus.hits.max.K.w50.25.bed")


closest_g4_PDS.ce10 <- closest_g4("../Figures/Figure1/exon_up_plus.ce10.closest.GSM3003538_Celegans_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed",
                             "Figures/Figure1/exon_up_minus.ce10.closest.GSM3003538_Celegans_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed",
                             "Figures/Figure1/exon_down_plus.ce10.closest.GSM3003538_Celegans_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed",
                             "Figures/Figure1/exon_down_minus.ce10.closest.GSM3003538_Celegans_all_w15_th-1_plus_minus.hits.max.PDS.w50.35.bed")



  


closest_g4_KCL.sacCer3[, `:=`(Treatment="KCl", Species="S. cerevisiae") ]
closest_g4_PDS.sacCer3[, `:=`(Treatment="PDS", Species="S. cerevisiae") ]
closest_g4_KCL.tair10[, `:=`(Treatment="KCl", Species="A. thaliana" ) ]
closest_g4_PDS.tair10[, `:=`(Treatment="PDS", Species="A. thaliana" ) ]
closest_g4_KCL.danRer10[, `:=`(Treatment="KCl", Species="D. rerio"  ) ]
closest_g4_PDS.danRer10[, `:=`(Treatment="PDS", Species="D. rerio"  ) ]
closest_g4_KCL.dm6[, `:=`(Treatment="KCl", Species= "D. melanogaster" ) ]
closest_g4_PDS.dm6[, `:=`(Treatment="PDS", Species=  "D. melanogaster" ) ]
closest_g4_KCL.mm10[, `:=`(Treatment="KCl", Species="M. musculus"  ) ]
closest_g4_PDS.mm10[, `:=`(Treatment="PDS", Species="M. musculus"  ) ]
closest_g4_KCL.hg19[, `:=`(Treatment="KCl", Species="H. sapiens"    ) ]
closest_g4_PDS.hg19[, `:=`(Treatment="PDS", Species="H. sapiens"    ) ]
closest_g4_KCL.ce10[, `:=`(Treatment="KCl", Species="C. elegans"    ) ]
closest_g4_PDS.ce10[, `:=`(Treatment="PDS", Species="C. elegans"   ) ]


closest_g4.TOTAL <- rbind(closest_g4_KCL.sacCer3,
closest_g4_PDS.sacCer3,
closest_g4_KCL.tair10,
closest_g4_PDS.tair10,
closest_g4_KCL.danRer10,
closest_g4_PDS.danRer10,
closest_g4_KCL.dm6,
closest_g4_PDS.dm6,
closest_g4_KCL.mm10,
closest_g4_PDS.mm10,
closest_g4_KCL.hg19,
closest_g4_PDS.hg19,
closest_g4_KCL.hg19,
closest_g4_PDS.hg19)


ggplot(closest_g4.TOTAL) +
  geom_line(aes(dist, colour=Species), bw=10, stat="density") +
  xlim(c(-1000, 1000)) +
  facet_grid(Treatment ~ pos) +
  xlab("Distance") +
  ylab("Density") +
  theme_bw() +
  theme(legend.position = "top", legend.direction = "horizontal")  + 
  theme(axis.text.x = element_text(vjust = 0.7, 
    angle = 45)) 




```



```{r}
closest_g4.TOTAL.1kb <- closest_g4.TOTAL[abs(dist)<=1000 , ]

closest_g4.TOTAL.1kb[, bin:=cut(dist, seq(-1000, 1000, 5), labels= seq(-995, 1000, 5) ) ]


closest_g4.TOTAL.1kb.enrichment <- closest_g4.TOTAL.1kb[ , .(Occurrences=.N) , by=c("bin", "pos", "Treatment", "Species")]


closest_g4.TOTAL.1kb.enrichment.median <-  closest_g4.TOTAL.1kb.enrichment[, .(median=median(Occurrences)), by=c( "pos", "Treatment", "Species") ]


closest_g4.TOTAL.1kb.enrichment <- merge(closest_g4.TOTAL.1kb.enrichment, closest_g4.TOTAL.1kb.enrichment.median, by=c( "pos", "Treatment", "Species")) 

closest_g4.TOTAL.1kb.enrichment[, Enrrichment:=Occurrences/median]


closest_g4.TOTAL.1kb.enrichment[ , bin:=as.numeric(as.character(bin))]


Fig6.D <- ggplot(closest_g4.TOTAL.1kb.enrichment) +
  geom_line(aes(bin, Enrrichment, colour=Species)) +
  xlim(c(-1000, 1000)) +
  facet_grid(Treatment ~ pos) +
  xlab("Distance") +
  ylab("Enrichment") +
  theme_bw() +
  theme(legend.position = "top", legend.direction = "horizontal")  + 
  theme(axis.text.x = element_text(vjust = 0.7, 
    angle = 45)) 

Fig6.D

```





```{r}



read_dist_table_plus_minus <- function(path.plus, path.minus ){
  
dist_table.plus <- data.table(read_delim(path.plus, 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE))

dist_table.plus <- dist_table.plus[, 2:2001]
dist_table.plus <- data.table(as.data.frame(t(dist_table.plus)))
colnames(dist_table.plus) <- c("Position", "Occurrences")


dist_table.minus <- data.table(read_delim(path.minus, 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE))

dist_table.minus <- dist_table.minus[, 2:2001]
dist_table.minus <- data.table(as.data.frame(t(dist_table.minus)))
colnames(dist_table.minus) <- c("Position", "Occurrences")

 
dist_table <-  merge(dist_table.plus, dist_table.minus, by="Position")

dist_table[, Occurrences:=(Occurrences.x + Occurrences.y)]

dist_table[, Occurrences:=(Occurrences.x + Occurrences.y)]
dist_table <- dist_table[, c("Position", "Occurrences")]


dist_table[,median:=median(Occurrences)]
dist_table[, Enrrichment:=Occurrences/median]
dist_table[, Position:=Position-1]

return(dist_table)  
}


```







```{r}

enrichment_G4_seq_plot <- function(path.up_plus.plus, path.up_plus.minus, 
                                   path.down_plus.plus, path.down_plus.minus, 
                                   path.up_minus.plus, path.up_minus.minus,
                                   path.down_minus.plus, path.down_minus.minus ){


x.up_plus <- read_dist_table_plus_minus(path.up_plus.plus,
                           path.up_plus.minus)


x.down_plus <- read_dist_table_plus_minus(path.down_plus.plus,
                           path.down_plus.minus)


x.up_minus <- read_dist_table_plus_minus(path.up_minus.plus,
                           path.up_minus.minus)


x.down_minus <- read_dist_table_plus_minus(path.down_minus.plus,
                          path.down_minus.minus)


x.up_minus[,Position:=Position*-1]
x.down_minus[,Position:=Position*-1]

x.TOTAL <- plot_density(x.up_plus, x.up_minus, x.down_plus, x.down_minus)

return(x.TOTAL)

}

```


```{r, message=FALSE, error=FALSE, warning=FALSE}

hg19.TOTAL.KCL <- enrichment_G4_seq_plot("../G4_seq_2019/Evolution/exon_up_plus.hg19.score.GSM3003539_Homo_all_w15_th-1_plus.hits.max.K.w50.25.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_up_plus.hg19.score.GSM3003539_Homo_all_w15_th-1_minus.hits.max.K.w50.25.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_down_plus.hg19.score.GSM3003539_Homo_all_w15_th-1_plus.hits.max.K.w50.25.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_down_plus.hg19.score.GSM3003539_Homo_all_w15_th-1_minus.hits.max.K.w50.25.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_up_minus.hg19.score.GSM3003539_Homo_all_w15_th-1_plus.hits.max.K.w50.25.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_up_minus.hg19.score.GSM3003539_Homo_all_w15_th-1_minus.hits.max.K.w50.25.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_down_minus.hg19.score.GSM3003539_Homo_all_w15_th-1_plus.hits.max.K.w50.25.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_down_minus.hg19.score.GSM3003539_Homo_all_w15_th-1_minus.hits.max.K.w50.25.bed.txt..num")


hg19.TOTAL.PDS <- enrichment_G4_seq_plot("./G4_seq_2019/Evolution/exon_up_plus.hg19.score.GSM3003540_Homo_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_up_plus.hg19.score.GSM3003540_Homo_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_down_plus.hg19.score.GSM3003540_Homo_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_down_plus.hg19.score.GSM3003540_Homo_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_up_minus.hg19.score.GSM3003540_Homo_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_up_minus.hg19.score.GSM3003540_Homo_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_down_minus.hg19.score.GSM3003540_Homo_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_down_minus.hg19.score.GSM3003540_Homo_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.txt..num")



sacCer3.TOTAL.KCL <- enrichment_G4_seq_plot("./G4_seq_2019/Evolution/exon_up_plus.bed.sacCer3.score.GSM3003553_Saccaromyces_all_w15_th-1_plus.hits.max.K.w50.25.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_up_plus.bed.sacCer3.score.GSM3003553_Saccaromyces_all_w15_th-1_minus.hits.max.K.w50.25.bed.txt..num",
                       
                       "./G4_seq_2019/Evolution/exon_down_plus.bed.sacCer3.score.GSM3003554_Saccaromyces_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_down_plus.bed.sacCer3.score.GSM3003553_Saccaromyces_all_w15_th-1_minus.hits.max.K.w50.25.bed.txt..num",
                       
                       "./G4_seq_2019/Evolution/exon_up_minus.bed.sacCer3.score.GSM3003553_Saccaromyces_all_w15_th-1_plus.hits.max.K.w50.25.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_up_minus.bed.sacCer3.score.GSM3003553_Saccaromyces_all_w15_th-1_minus.hits.max.K.w50.25.bed.txt..num",
                       
                       "./G4_seq_2019/Evolution/exon_down_minus.bed.sacCer3.score.GSM3003553_Saccaromyces_all_w15_th-1_plus.hits.max.K.w50.25.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_down_minus.bed.sacCer3.score.GSM3003553_Saccaromyces_all_w15_th-1_minus.hits.max.K.w50.25.bed.txt..num")


sacCer3.TOTAL.PDS <- enrichment_G4_seq_plot("./G4_seq_2019/Evolution/exon_up_plus.bed.sacCer3.score.GSM3003554_Saccaromyces_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_up_plus.bed.sacCer3.score.GSM3003554_Saccaromyces_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.txt..num",
                       
                       "./G4_seq_2019/Evolution/exon_down_plus.bed.sacCer3.score.GSM3003554_Saccaromyces_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_down_plus.bed.sacCer3.score.GSM3003554_Saccaromyces_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.txt..num",
                       
                       "./G4_seq_2019/Evolution/exon_up_minus.bed.sacCer3.score.GSM3003554_Saccaromyces_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_up_minus.bed.sacCer3.score.GSM3003554_Saccaromyces_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.txt..num",
                       
                       "./G4_seq_2019/Evolution/exon_down_minus.bed.sacCer3.score.GSM3003554_Saccaromyces_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_down_minus.bed.sacCer3.score.GSM3003554_Saccaromyces_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.txt..num")


mouse.TOTAL.KCL <- enrichment_G4_seq_plot("./G4_seq_2019/Evolution/exon_up_plus.mm10.score.GSM3003547_Mouse_all_w15_th-1_plus.hits.max.K.w50.25.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_up_plus.mm10.score.GSM3003547_Mouse_all_w15_th-1_minus.hits.max.K.w50.25.bed.txt..num",
                       
                       "./G4_seq_2019/Evolution/exon_down_plus.mm10.score.GSM3003547_Mouse_all_w15_th-1_plus.hits.max.K.w50.25.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_down_plus.mm10.score.GSM3003547_Mouse_all_w15_th-1_minus.hits.max.K.w50.25.bed.txt..num",
                       
                       "./G4_seq_2019/Evolution/exon_up_minus.mm10.score.GSM3003547_Mouse_all_w15_th-1_plus.hits.max.K.w50.25.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_up_minus.mm10.score.GSM3003547_Mouse_all_w15_th-1_minus.hits.max.K.w50.25.bed.txt..num",
                       
                       "./G4_seq_2019/Evolution/exon_down_minus.mm10.score.GSM3003547_Mouse_all_w15_th-1_plus.hits.max.K.w50.25.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_down_minus.mm10.score.GSM3003547_Mouse_all_w15_th-1_minus.hits.max.K.w50.25.bed.txt..num")


mouse.TOTAL.PDS <- enrichment_G4_seq_plot("./G4_seq_2019/Evolution/exon_up_plus.mm10.score.GSM3003548_Mouse_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_up_plus.mm10.score.GSM3003548_Mouse_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.txt..num",
                       
                       "./G4_seq_2019/Evolution/exon_down_plus.mm10.score.GSM3003548_Mouse_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_down_plus.mm10.score.GSM3003548_Mouse_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.txt..num",
                       
                       "./G4_seq_2019/Evolution/exon_up_minus.mm10.score.GSM3003548_Mouse_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_up_minus.mm10.score.GSM3003548_Mouse_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.txt..num",
                       
                       "./G4_seq_2019/Evolution/exon_down_minus.mm10.score.GSM3003548_Mouse_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_down_minus.mm10.score.GSM3003548_Mouse_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.txt..num")

 


dm6.TOTAL.KCL <- enrichment_G4_seq_plot("./G4_seq_2019/Evolution/exon_up_plus.dm6.score.GSM3003541_Drosophila_all_w15_th-1_plus.hits.max.K.w50.25.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_up_plus.dm6.score.GSM3003541_Drosophila_all_w15_th-1_minus.hits.max.K.w50.25.bed.txt..num",
                       
                       "./G4_seq_2019/Evolution/exon_down_plus.dm6.score.GSM3003541_Drosophila_all_w15_th-1_plus.hits.max.K.w50.25.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_down_plus.dm6.score.GSM3003541_Drosophila_all_w15_th-1_minus.hits.max.K.w50.25.bed.txt..num",
                       
                       "./G4_seq_2019/Evolution/exon_up_minus.dm6.score.GSM3003541_Drosophila_all_w15_th-1_plus.hits.max.K.w50.25.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_up_minus.dm6.score.GSM3003541_Drosophila_all_w15_th-1_minus.hits.max.K.w50.25.bed.txt..num",
                       
                       "./G4_seq_2019/Evolution/exon_down_minus.dm6.score.GSM3003541_Drosophila_all_w15_th-1_plus.hits.max.K.w50.25.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_down_minus.dm6.score.GSM3003541_Drosophila_all_w15_th-1_minus.hits.max.K.w50.25.bed.txt..num")


dm6.TOTAL.PDS <- enrichment_G4_seq_plot("./G4_seq_2019/Evolution/exon_up_plus.dm6.score.GSM3003542_Drosophila_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_up_plus.dm6.score.GSM3003542_Drosophila_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.txt..num",
                       
                       "./G4_seq_2019/Evolution/exon_down_plus.dm6.score.GSM3003542_Drosophila_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_down_plus.dm6.score.GSM3003542_Drosophila_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.txt..num",
                       
                       "./G4_seq_2019/Evolution/exon_up_minus.dm6.score.GSM3003542_Drosophila_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_up_minus.dm6.score.GSM3003542_Drosophila_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.txt..num",
                       
                       "./G4_seq_2019/Evolution/exon_down_minus.dm6.score.GSM3003542_Drosophila_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_down_minus.dm6.score.GSM3003542_Drosophila_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.txt..num")




tair10.TOTAL.KCL <- enrichment_G4_seq_plot("./G4_seq_2019/Evolution/exon_up_plus.bed.tair10.score.GSM3003535_Arabidopsis_all_w15_th-1_plus.hits.max.K.w50.25.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_up_plus.bed.tair10.score.GSM3003535_Arabidopsis_all_w15_th-1_minus.hits.max.K.w50.25.bed.txt..num",
                       
                       "./G4_seq_2019/Evolution/exon_down_plus.bed_tair10.score.GSM3003535_Arabidopsis_all_w15_th-1_plus.hits.max.K.w50.25.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_down_plus.bed_tair10.score.GSM3003535_Arabidopsis_all_w15_th-1_minus.hits.max.K.w50.25.bed.txt..num",
                       
                       "./G4_seq_2019/Evolution/exon_up_minus.bed.tair10.score.GSM3003535_Arabidopsis_all_w15_th-1_plus.hits.max.K.w50.25.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_up_minus.bed.tair10.score.GSM3003535_Arabidopsis_all_w15_th-1_minus.hits.max.K.w50.25.bed.txt..num",
                       
                       "./G4_seq_2019/Evolution/exon_down_minus.bed.tair10.score.GSM3003535_Arabidopsis_all_w15_th-1_plus.hits.max.K.w50.25.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_down_minus.bed.tair10.score.GSM3003535_Arabidopsis_all_w15_th-1_minus.hits.max.K.w50.25.bed.txt..num")


tair10.TOTAL.PDS <- enrichment_G4_seq_plot("./G4_seq_2019/Evolution/exon_up_plus.bed.tair10.score.GSM3003536_Arabidopsis_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_up_plus.bed.tair10.score.GSM3003536_Arabidopsis_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.txt..num",
                       
                       "./G4_seq_2019/Evolution/exon_down_plus.bed_tair10.score.GSM3003536_Arabidopsis_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_down_plus.bed_tair10.score.GSM3003536_Arabidopsis_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.txt..num",
                       
                       "./G4_seq_2019/Evolution/exon_up_minus.bed.tair10.score.GSM3003536_Arabidopsis_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_up_minus.bed.tair10.score.GSM3003536_Arabidopsis_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.txt..num",
                       
                       "./G4_seq_2019/Evolution/exon_down_minus.bed.tair10.score.GSM3003536_Arabidopsis_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_down_minus.bed.tair10.score.GSM3003536_Arabidopsis_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.txt..num")



danRer10.TOTAL.KCL <- enrichment_G4_seq_plot("./G4_seq_2019/Evolution/exon_up_plus.danRer10.score.GSM3003557_Zebrafish_all_w15_th-1_plus.hits.max.K.w50.25.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_up_plus.danRer10.score.GSM3003557_Zebrafish_all_w15_th-1_minus.hits.max.K.w50.25.bed.txt..num",
                       
                       "./G4_seq_2019/Evolution/exon_down_plus.danRer10.score.GSM3003557_Zebrafish_all_w15_th-1_plus.hits.max.K.w50.25.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_down_plus.danRer10.score.GSM3003557_Zebrafish_all_w15_th-1_minus.hits.max.K.w50.25.bed.txt..num",
                       
                       "./G4_seq_2019/Evolution/exon_up_minus.danRer10.score.GSM3003557_Zebrafish_all_w15_th-1_plus.hits.max.K.w50.25.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_up_minus.danRer10.score.GSM3003557_Zebrafish_all_w15_th-1_minus.hits.max.K.w50.25.bed.txt..num",
                       
                       "./G4_seq_2019/Evolution/exon_down_minus.danRer10.score.GSM3003557_Zebrafish_all_w15_th-1_plus.hits.max.K.w50.25.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_down_minus.danRer10.score.GSM3003557_Zebrafish_all_w15_th-1_minus.hits.max.K.w50.25.bed.txt..num")



danRer10.TOTAL.PDS <- enrichment_G4_seq_plot("./G4_seq_2019/Evolution/exon_up_plus.danRer10.score.GSM3003558_Zebrafish_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_up_plus.danRer10.score.GSM3003558_Zebrafish_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.txt..num",
                       
                       "./G4_seq_2019/Evolution/exon_down_plus.danRer10.score.GSM3003558_Zebrafish_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_down_plus.danRer10.score.GSM3003558_Zebrafish_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.txt..num",
                       
                       "./G4_seq_2019/Evolution/exon_up_minus.danRer10.score.GSM3003558_Zebrafish_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_up_minus.danRer10.score.GSM3003558_Zebrafish_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.txt..num",
                       
                       "./G4_seq_2019/Evolution/exon_down_minus.danRer10.score.GSM3003558_Zebrafish_all_w15_th-1_plus.hits.max.PDS.w50.35.bed.txt..num",
                       "./G4_seq_2019/Evolution/exon_down_minus.danRer10.score.GSM3003558_Zebrafish_all_w15_th-1_minus.hits.max.PDS.w50.35.bed.txt..num")

```


```{r, message=FALSE, error=FALSE, warning=FALSE}

hg19.TOTAL.KCL[, Treatment:="KCl"] 
hg19.TOTAL.PDS[, Treatment:="PDS"]
hg19.TOTAL.KCL[, Species:="H. sapiens"] 
hg19.TOTAL.PDS[, Species:="H. sapiens"]  

sacCer3.TOTAL.KCL[, Treatment:="KCl"]  
sacCer3.TOTAL.PDS[, Treatment:="PDS"]
sacCer3.TOTAL.KCL[, Species:="S. cerevisiae"]  
sacCer3.TOTAL.PDS[, Species:="S. cerevisiae"]  

mouse.TOTAL.KCL[, Treatment:="KCl"]  
mouse.TOTAL.PDS[, Treatment:="PDS"] 
mouse.TOTAL.KCL[, Species:="M. musculus"]  
mouse.TOTAL.PDS[, Species:="M. musculus"]  

dm6.TOTAL.KCL[, Treatment:="KCl"]  
dm6.TOTAL.PDS[, Treatment:="PDS"]
dm6.TOTAL.KCL[, Species:="D. melanogaster"]  
dm6.TOTAL.PDS[, Species:="D. melanogaster"]  

tair10.TOTAL.KCL[, Treatment:="KCl"] 
tair10.TOTAL.PDS[, Treatment:="PDS"] 
tair10.TOTAL.KCL[, Species:="A. thaliana"] 
tair10.TOTAL.PDS[, Species:="A. thaliana"] 

danRer10.TOTAL.KCL[, Treatment:="KCl"]  
danRer10.TOTAL.PDS[, Treatment:="PDS"]
danRer10.TOTAL.KCL[, Species:="D. rerio"]  
danRer10.TOTAL.PDS[, Species:="D. rerio"]


TOTAL.KCL_PDS <-  rbind(hg19.TOTAL.KCL, hg19.TOTAL.PDS, sacCer3.TOTAL.KCL, sacCer3.TOTAL.PDS, mouse.TOTAL.KCL, mouse.TOTAL.PDS, dm6.TOTAL.KCL, dm6.TOTAL.PDS, tair10.TOTAL.KCL, tair10.TOTAL.PDS, danRer10.TOTAL.KCL, danRer10.TOTAL.PDS)


ggplot(TOTAL.KCL_PDS) +
  geom_line(aes(Position, Enrrichment, colour=Species)) +
  xlim(c(-1000, 1000)) +
  facet_grid(Treatment ~ exon_pos) +
  xlab("Distance") +
  ylab("Enrichment") +
  theme_bw() +
  theme(legend.position = "top", legend.direction = "horizontal")  + 
  theme(axis.text.x = element_text(vjust = 0.7, 
    angle = 45)) 

```







```{r}
closest_g4_KCL.hg19.2015 <- closest_g4("../Figures/Supplementary/exon_up_plus.hg19.closest.GSE63874_Na_K_plus_minus_hits_intersect.bed",
                             "Figures/Supplementary/exon_up_minus.hg19.closest.GSE63874_Na_K_plus_minus_hits_intersect.bed",
                             "Figures/Supplementary/exon_down_plus.hg19.closest.GSE63874_Na_K_plus_minus_hits_intersect.bed",
                             "Figures/Supplementary/exon_down_minus.hg19.closest.GSE63874_Na_K_plus_minus_hits_intersect.bed")


closest_g4_KCL.hg19.2015 <- closest_g4("../Figures/Supplementary/exon_up_plus.hg19.closest.GSE63874_Na_K_plus_minus_hits_intersect.bed",
                             "Figures/Supplementary/exon_up_minus.hg19.closest.GSE63874_Na_K_plus_minus_hits_intersect.bed",
                             "Figures/Supplementary/exon_down_plus.hg19.closest.GSE63874_Na_K_plus_minus_hits_intersect.bed",
                             "Figures/Supplementary/exon_down_minus.hg19.closest.GSE63874_Na_K_plus_minus_hits_intersect.bed")
```



```{r}
KCL_up <- unique(closest_g4_KCL[abs(dist)>=100 & pos=="3'Splice site", paste(echrom, estart, eend, sep = "_")])


```








```{r}
gg_color_hue <- function(n) {
  hues = seq(15, 375, length = n + 1)
  hcl(h = hues, l = 65, c = 100)[1:n]
}

gg_color_hue(3)
```


```{r}


 library(eulerr)




VennDiag.up <- euler(c("Motif" = 8807+9325,  
                    "K" = 9653+10029,  
                    "PDS" = 34880+36909, 
                    "K&PDS" = 9890+9539,
                    "PDS&Motif" = 8331+8855,
                    "K&Motif" = 4509+4749,
                    "K&PDS&Motif" = 4503+4747))
Fig1.C1 <- plot(VennDiag.up, counts = TRUE, font=2, cex=1, alpha=0.5,
     fill=gg_color_hue(3), quantities = list(fontsize = 10))


VennDiag.down <- euler(c("Motif" = 9507+10142,  
                    "K" = 10120+9365,  
                    "PDS" = 34844+36737, 
                    "K&PDS" = 9236+9985,
                    "PDS&Motif" = 8973+9593,
                    "K&Motif" = 4806+5221,
                    "K&PDS&Motif" = 4802+5211))
Fig1.C2 <- plot(VennDiag.down, counts = TRUE, font=1, cex=1, alpha=0.5,
     fill=gg_color_hue(3),  quantities = list(fontsize = 10))


Fig1.C <- plot_grid(Fig1.C1, Fig1.C2, nrow=1)
Fig1.C
```


```{r}


VennDiag.up_2015 <- euler(c("Motif" = 18132,  
                    "K" = 34761,  
                    "PDS" = 50894, 
                    "K&PDS" = 31711,
                    "PDS&Motif" = 14786,
                    "K&Motif" = 11794,
                    "K&PDS&Motif" = 11250))
venn_supp.up <- plot(VennDiag.up_2015, counts = TRUE, font=2, cex=1, alpha=0.5,
     fill=gg_color_hue(3), quantities = list(fontsize = 10))


VennDiag.down_2015 <- euler(c("Motif" = 19649,  
                    "K" = 33908,  
                    "PDS" = 49806, 
                    "K&PDS" = 30920,
                    "PDS&Motif" = 15951,
                    "K&Motif" = 12615,
                    "K&PDS&Motif" = 12027))
venn_supp.down <- plot(VennDiag.down_2015, counts = TRUE, font=1, cex=1, alpha=0.5,
     fill=gg_color_hue(3),  quantities = list(fontsize = 10))

plot_grid(venn_supp.up, venn_supp.down, labels = c("3' Splice site", "5' Splice site"))

```





```{r}
dataframe_G4_consensus_motif_enrichment_species <- fread("../Figures/Figure1/dataframe_G4_consensus_motif_enrichment_species")

dataframe_G4_consensus_motif_enrichment_species_matrix <- t(dataframe_G4_consensus_motif_enrichment_species)



header <- as.character(head(dataframe_G4_consensus_motif_enrichment_species_matrix, 1))

dataframe_G4_consensus_motif_enrichment_species_df <- as.data.frame(tail(dataframe_G4_consensus_motif_enrichment_species_matrix, -1))

colnames(dataframe_G4_consensus_motif_enrichment_species_df) <- header

dataframe_G4_consensus_motif_enrichment_species_df$specie <- rownames(dataframe_G4_consensus_motif_enrichment_species_df)


dataframe_G4_consensus_motif_enrichment_species_dt <- data.table(dataframe_G4_consensus_motif_enrichment_species_df)

dataframe_G4_consensus_motif_enrichment_species_dt$median <- as.numeric(as.character(dataframe_G4_consensus_motif_enrichment_species_dt$median))
dataframe_G4_consensus_motif_enrichment_species_dt$st.dev <- as.numeric(as.character(dataframe_G4_consensus_motif_enrichment_species_dt$st.dev))

dataframe_G4_consensus_motif_enrichment_species_dt$specie <- factor(dataframe_G4_consensus_motif_enrichment_species_dt$specie , levels=dataframe_G4_consensus_motif_enrichment_species_dt[order(-median)]$specie)



```

```{r}
Fig6.A <- ggplot(dataframe_G4_consensus_motif_enrichment_species_dt, aes(x = specie, y=median) ) +
  geom_bar( stat="identity", colour="black", fill="white") +
  geom_errorbar(aes(ymin = (median - st.dev), ymax = (as.numeric(median) + as.numeric(st.dev))), colour="red" , width=0.2) +
  theme_bw()+
  xlab("Species") +
  ylab("Density (G4s / kB)") +
  theme(axis.text.x = element_text(vjust = 0.7, 
    angle = 45))

Fig6.A

```

```{r}

dataframe_evo_PDS_K <- fread("../Figures/Figure1/dataframe_evo_PDS_K")




dataframe_evo_PDS_K_matrix <- t(dataframe_evo_PDS_K)



header <- as.character(head(dataframe_evo_PDS_K_matrix, 1))

dataframe_evo_PDS_K_df <- as.data.frame(tail(dataframe_evo_PDS_K_matrix, -1))

colnames(dataframe_evo_PDS_K_df) <- header


dataframe_evo_PDS_K_df$Species <- rownames(dataframe_evo_PDS_K_df)

dataframe_evo_PDS_K_dt <- data.table(dataframe_evo_PDS_K_df)

dataframe_evo_PDS_K_dt_nice <- rbind(dataframe_evo_PDS_K_dt[ , .(Species=Species, Treatment="K+", median=K_median, st.dev=K_st.dev  )  ],
dataframe_evo_PDS_K_dt[ , .(Species=Species, Treatment="PDS", median=PDS_median, st.dev=PDS_st.dev  )  ])


dataframe_evo_PDS_K_dt_nice$median <- as.numeric(as.character(dataframe_evo_PDS_K_dt_nice$median))
dataframe_evo_PDS_K_dt_nice$st.dev <- as.numeric(as.character(dataframe_evo_PDS_K_dt_nice$st.dev))


ggplot(dataframe_evo_PDS_K_dt_nice, aes(x = Species, y=median, group=Treatment)) +
  geom_bar( aes(colour=Treatment ), stat="identity", fill="white", position="dodge") +
  geom_errorbar(aes(ymin = (median - st.dev), ymax = (as.numeric(median) + as.numeric(st.dev))), colour="black" , width=0.2,  position = position_dodge()) +
  theme_bw()+
  xlab("Species") +
  ylab("Density (G4s / kB)") +
  theme(axis.text.x = element_text(vjust = 0.7, 
    angle = 45))



dataframe_evo_PDS_K_dt_nice$Species <- factor(dataframe_evo_PDS_K_dt_nice$Species , levels=dataframe_evo_PDS_K_dt_nice[Treatment=="PDS"][order(-median)]$Species)


Fig6.C <-  ggplot(dataframe_evo_PDS_K_dt_nice, aes(x=Species, weight=median, ymin=median-st.dev, ymax=median+st.dev, group=Treatment)) +
     geom_bar(position=position_dodge(), aes(y=median, color=Treatment), fill="white", stat="identity") +
     geom_errorbar (position=position_dodge(width=0.9), colour="black", width=0.2) +
    theme_bw()+
  xlab("Species") +
  ylab("Density (G4s / kB)") +
  theme(axis.text.x = element_text(vjust = 0.7, 
    angle = 45))


Fig6.C

```

```{r}
dataframe_evo_PDS_K_dt_nice
```


###### Figures  #####



```{r, fig.height=8, fig.width=12}

Fig1.BC <- plot_grid(Fig1.B, Fig1.C, ncol=1, labels = c("B", "C"), rel_heights = c(2, 1))

Fig1 <- plot_grid(Fig1.A, Fig1.BC, nrow=1, labels = c("A", ""))

Fig1
```


```{r, fig.height=10, fig.width=7}

Fig2 <- plot_grid(Fig2.A, Fig2.B,ncol = 1 , labels ="AUTO", rel_heights = c(1.5, 1))
Fig2
```


```{r, fig.height=10, fig.width=14}
Fig3.old <- plot_grid(Fig3.A, Fig3.B, Fig3.C, Fig3.D, labels = "AUTO", ncol=2)


plot_grid( plot_grid(Fig3.C, Fig3.D), Fig3.C.new, Fig3.D.new, ncol = 1)


Fig3.top <-  plot_grid( Fig3.B,  Fig3.C, Fig3.D, Fig3.C.new, labels = "AUTO", ncol=2)

Fig3 <-plot_grid(Fig3.top, Fig3.D.new, labels = c("", "E"), ncol = 1, rel_heights = c(1.5, 1) )




```

```{r, fig.height=8, fig.width=14}
Fig3.sup.top <-  plot_grid(Fig3.A, Fig3.sup.B, labels="AUTO", rel_widths = c(1, 3) )
Fig3.sup <- plot_grid(Fig3.sup.top, Fig3.sup.C, ncol=1, labels=c("", "C"))
Fig3.sup
```



```{r, fig.height=10, fig.width=13.35}



Fig5.bottom <- plot_grid(Fig5.A, Fig5.B, Fig5.C, Fig5.D, nrow = 2, ncol = 2 , labels = c("B", "C", "D", "E"), rel_heights = c(1, 1, 1, 1))

Fig5 <- plot_grid(Fig5.A1, Fig5.bottom, nrow=2, labels = c("A", ""),  rel_heights = c(2, 4))

Fig5
```



```{r, fig.height=8, fig.width=10}


Fig6.top <- plot_grid(Fig6.A, Fig6.C, labels = c("A", "C"))
Fig6.bottom <- plot_grid(Fig6.B, Fig6.D, labels = c("B", "D"))

Fig6 <- plot_grid(Fig6.A, Fig6.C, Fig6.B , Fig6.D, labels = c("A", "C", "B", "D") , ncol=2, rel_heights = c(1, 1, 3, 3))


Fig6 <- plot_grid(Fig6.top, Fig6.bottom, rel_heights = c(1, 1.5), nrow=2)
Fig6
```



